<!DOCTYPE html>
<html lang="en-us" dir="ltr">
    <head><meta charset='utf-8'>
<meta name='viewport' content='width=device-width, initial-scale=1'><meta name='description' content='ÂÆèÂü∫Âõ†ÁªÑÔºàMetagenomeÔºâÊòØÊåáÂØπ‰∏Ä‰∏™ÁîüÊÄÅÁ≥ªÁªü‰∏≠ÁöÑÊâÄÊúâÂæÆÁîüÁâ©ËøõË°åDNAÂàÜÊûêÁöÑËøáÁ®ãÔºåÂèØ‰ª•Â∏ÆÂä©Á†îÁ©∂‰∫∫Âëò‰∫ÜËß£ÂæÆÁîüÁâ©ÁöÑÂ§öÊ†∑ÊÄß„ÄÅÂäüËÉΩÂíå‰∫í‰ΩúÂÖ≥Á≥ª„ÄÇËøôÈáå‰ªãÁªçÂ∏∏Áî®ÂàÜÊûêÊµÅÁ®ã„ÄÇ'>
<title>Metagenomic workflow</title>

<link rel='canonical' href='/p/metagenomic-workflow/'>

<link rel="stylesheet" href="/scss/style.min.8191399262444ab68b72a18c97392f5349be20a1615d77445be51e974c144cff.css"><meta property='og:title' content='Metagenomic workflow'>
<meta property='og:description' content='ÂÆèÂü∫Âõ†ÁªÑÔºàMetagenomeÔºâÊòØÊåáÂØπ‰∏Ä‰∏™ÁîüÊÄÅÁ≥ªÁªü‰∏≠ÁöÑÊâÄÊúâÂæÆÁîüÁâ©ËøõË°åDNAÂàÜÊûêÁöÑËøáÁ®ãÔºåÂèØ‰ª•Â∏ÆÂä©Á†îÁ©∂‰∫∫Âëò‰∫ÜËß£ÂæÆÁîüÁâ©ÁöÑÂ§öÊ†∑ÊÄß„ÄÅÂäüËÉΩÂíå‰∫í‰ΩúÂÖ≥Á≥ª„ÄÇËøôÈáå‰ªãÁªçÂ∏∏Áî®ÂàÜÊûêÊµÅÁ®ã„ÄÇ'>
<meta property='og:url' content='/p/metagenomic-workflow/'>
<meta property='og:site_name' content='Asa&#39;s blog'>
<meta property='og:type' content='article'><meta property='article:section' content='Post' /><meta property='article:tag' content='shell' /><meta property='article:tag' content='ÁºñÁ®ã' /><meta property='article:published_time' content='2023-03-26T00:00:00&#43;00:00'/><meta property='article:modified_time' content='2023-03-26T00:00:00&#43;00:00'/><meta property='og:image' content='/' />
<meta name="twitter:title" content="Metagenomic workflow">
<meta name="twitter:description" content="ÂÆèÂü∫Âõ†ÁªÑÔºàMetagenomeÔºâÊòØÊåáÂØπ‰∏Ä‰∏™ÁîüÊÄÅÁ≥ªÁªü‰∏≠ÁöÑÊâÄÊúâÂæÆÁîüÁâ©ËøõË°åDNAÂàÜÊûêÁöÑËøáÁ®ãÔºåÂèØ‰ª•Â∏ÆÂä©Á†îÁ©∂‰∫∫Âëò‰∫ÜËß£ÂæÆÁîüÁâ©ÁöÑÂ§öÊ†∑ÊÄß„ÄÅÂäüËÉΩÂíå‰∫í‰ΩúÂÖ≥Á≥ª„ÄÇËøôÈáå‰ªãÁªçÂ∏∏Áî®ÂàÜÊûêÊµÅÁ®ã„ÄÇ"><meta name="twitter:card" content="summary_large_image">
    <meta name="twitter:image" content='/' /><link rel="apple-touch-icon" sizes="180x180" href="/apple-touch-icon.png">
<link rel="icon" type="image/png" sizes="32x32" href="/favicon-32x32.png">
<link rel="icon" type="image/png" sizes="16x16" href="/favicon-16x16.png">
<link rel="manifest" href="/site.webmanifest">
<link rel="shortcut icon" href="/favicon-32x32.png" />
    </head>
    <body class="
    article-page
    ">
    <script>
        (function() {
            const colorSchemeKey = 'StackColorScheme';
            if(!localStorage.getItem(colorSchemeKey)){
                localStorage.setItem(colorSchemeKey, "auto");
            }
        })();
    </script><script>
    (function() {
        const colorSchemeKey = 'StackColorScheme';
        const colorSchemeItem = localStorage.getItem(colorSchemeKey);
        const supportDarkMode = window.matchMedia('(prefers-color-scheme: dark)').matches === true;

        if (colorSchemeItem == 'dark' || colorSchemeItem === 'auto' && supportDarkMode) {
            

            document.documentElement.dataset.scheme = 'dark';
        } else {
            document.documentElement.dataset.scheme = 'light';
        }
    })();
</script>
<div class="container main-container flex on-phone--column extended"><aside class="sidebar left-sidebar sticky ">
    <button class="hamburger hamburger--spin" type="button" id="toggle-menu" aria-label="Toggle Menu">
        <span class="hamburger-box">
            <span class="hamburger-inner"></span>
        </span>
    </button>

    <header>
        
            
            <figure class="site-avatar">
                <a href="/">
                
                    
                    
                    
                        
                        <img src="/img/guodong1_hu19ef19e107d2cae4138b147ef6582252_202975_300x0_resize_q75_box.jpg" width="300"
                            height="302" class="site-logo" loading="lazy" alt="Avatar">
                    
                
                </a>
                
                    <span class="emoji">üêõüê∂</span>
                
            </figure>
            
        
        
        <div class="site-meta">
            <h1 class="site-name"><a href="/">Asa&#39;s blog</a></h1>
            <h2 class="site-description">work hard, live happily.</h2>
        </div>
    </header><ol class="social-menu">
            
                <li>
                    <a 
                        href='https://github.com/Asa12138'
                        target="_blank"
                        title="GitHub"
                        rel="me"
                    >
                        
                        
                            <svg xmlns="http://www.w3.org/2000/svg" class="icon icon-tabler icon-tabler-brand-github" width="24" height="24" viewBox="0 0 24 24" stroke-width="2" stroke="currentColor" fill="none" stroke-linecap="round" stroke-linejoin="round">
  <path stroke="none" d="M0 0h24v24H0z" fill="none"/>
  <path d="M9 19c-4.3 1.4 -4.3 -2.5 -6 -3m12 5v-3.5c0 -1 .1 -1.4 -.5 -2c2.8 -.3 5.5 -1.4 5.5 -6a4.6 4.6 0 0 0 -1.3 -3.2a4.2 4.2 0 0 0 -.1 -3.2s-1.1 -.3 -3.5 1.3a12.3 12.3 0 0 0 -6.2 0c-2.4 -1.6 -3.5 -1.3 -3.5 -1.3a4.2 4.2 0 0 0 -.1 3.2a4.6 4.6 0 0 0 -1.3 3.2c0 4.6 2.7 5.7 5.5 6c-.6 .6 -.6 1.2 -.5 2v3.5" />
</svg>



                        
                    </a>
                </li>
            
                <li>
                    <a 
                        href='https://twitter.com'
                        target="_blank"
                        title="Twitter"
                        rel="me"
                    >
                        
                        
                            <svg xmlns="http://www.w3.org/2000/svg" class="icon icon-tabler icon-tabler-brand-twitter" width="24" height="24" viewBox="0 0 24 24" stroke-width="2" stroke="currentColor" fill="none" stroke-linecap="round" stroke-linejoin="round">
  <path stroke="none" d="M0 0h24v24H0z" fill="none"/>
  <path d="M22 4.01c-1 .49 -1.98 .689 -3 .99c-1.121 -1.265 -2.783 -1.335 -4.38 -.737s-2.643 2.06 -2.62 3.737v1c-3.245 .083 -6.135 -1.395 -8 -4c0 0 -4.182 7.433 4 11c-1.872 1.247 -3.739 2.088 -6 2c3.308 1.803 6.913 2.423 10.034 1.517c3.58 -1.04 6.522 -3.723 7.651 -7.742a13.84 13.84 0 0 0 .497 -3.753c-.002 -.249 1.51 -2.772 1.818 -4.013z" />
</svg>



                        
                    </a>
                </li>
            
        </ol><ol class="menu" id="main-menu">
        
        
        
        <li >
            <a href='/' >
                
                
                
                    <svg xmlns="http://www.w3.org/2000/svg" class="icon icon-tabler icon-tabler-home" width="24" height="24" viewBox="0 0 24 24" stroke-width="2" stroke="currentColor" fill="none" stroke-linecap="round" stroke-linejoin="round">
  <path stroke="none" d="M0 0h24v24H0z"/>
  <polyline points="5 12 3 12 12 3 21 12 19 12" />
  <path d="M5 12v7a2 2 0 0 0 2 2h10a2 2 0 0 0 2 -2v-7" />
  <path d="M9 21v-6a2 2 0 0 1 2 -2h2a2 2 0 0 1 2 2v6" />
</svg>



                
                <span>Home</span>
            </a>
        </li>
        
        
        <li >
            <a href='/about/' >
                
                
                
                    <svg xmlns="http://www.w3.org/2000/svg" class="icon icon-tabler icon-tabler-user" width="24" height="24" viewBox="0 0 24 24" stroke-width="2" stroke="currentColor" fill="none" stroke-linecap="round" stroke-linejoin="round">
  <path stroke="none" d="M0 0h24v24H0z"/>
  <circle cx="12" cy="7" r="4" />
  <path d="M6 21v-2a4 4 0 0 1 4 -4h4a4 4 0 0 1 4 4v2" />
</svg>



                
                <span>About</span>
            </a>
        </li>
        
        
        <li >
            <a href='/archives/' >
                
                
                
                    <svg xmlns="http://www.w3.org/2000/svg" class="icon icon-tabler icon-tabler-archive" width="24" height="24" viewBox="0 0 24 24" stroke-width="2" stroke="currentColor" fill="none" stroke-linecap="round" stroke-linejoin="round">
  <path stroke="none" d="M0 0h24v24H0z"/>
  <rect x="3" y="4" width="18" height="4" rx="2" />
  <path d="M5 8v10a2 2 0 0 0 2 2h10a2 2 0 0 0 2 -2v-10" />
  <line x1="10" y1="12" x2="14" y2="12" />
</svg>



                
                <span>Archives</span>
            </a>
        </li>
        
        
        <li >
            <a href='/search/' >
                
                
                
                    <svg xmlns="http://www.w3.org/2000/svg" class="icon icon-tabler icon-tabler-search" width="24" height="24" viewBox="0 0 24 24" stroke-width="2" stroke="currentColor" fill="none" stroke-linecap="round" stroke-linejoin="round">
  <path stroke="none" d="M0 0h24v24H0z"/>
  <circle cx="10" cy="10" r="7" />
  <line x1="21" y1="21" x2="15" y2="15" />
</svg>



                
                <span>Search</span>
            </a>
        </li>
        
        
        <li >
            <a href='/links/' >
                
                
                
                    <svg xmlns="http://www.w3.org/2000/svg" class="icon icon-tabler icon-tabler-link" width="24" height="24" viewBox="0 0 24 24" stroke-width="2" stroke="currentColor" fill="none" stroke-linecap="round" stroke-linejoin="round">
  <path stroke="none" d="M0 0h24v24H0z"/>
  <path d="M10 14a3.5 3.5 0 0 0 5 0l4 -4a3.5 3.5 0 0 0 -5 -5l-.5 .5" />
  <path d="M14 10a3.5 3.5 0 0 0 -5 0l-4 4a3.5 3.5 0 0 0 5 5l.5 -.5" />
</svg>



                
                <span>Links</span>
            </a>
        </li>
        

        <div class="menu-bottom-section">
                <li id="i18n-switch">  
                    <svg xmlns="http://www.w3.org/2000/svg" class="icon icon-tabler icon-tabler-language" width="24" height="24" viewBox="0 0 24 24" stroke-width="2" stroke="currentColor" fill="none" stroke-linecap="round" stroke-linejoin="round">
  <path stroke="none" d="M0 0h24v24H0z" fill="none"/>
  <path d="M4 5h7" />
  <path d="M9 3v2c0 4.418 -2.239 8 -5 8" />
  <path d="M5 9c-.003 2.144 2.952 3.908 6.7 4" />
  <path d="M12 20l4 -9l4 9" />
  <path d="M19.1 18h-6.2" />
</svg>



                    <select name="language" onchange="window.location.href = this.selectedOptions[0].value">
                        
                            <option value="/" selected>English</option>
                        
                            <option value="/zh-cn/" >‰∏≠Êñá</option>
                        
                    </select>
                </li>
            
            
            
                <li id="dark-mode-toggle">
                    <svg xmlns="http://www.w3.org/2000/svg" class="icon icon-tabler icon-tabler-toggle-left" width="24" height="24" viewBox="0 0 24 24" stroke-width="2" stroke="currentColor" fill="none" stroke-linecap="round" stroke-linejoin="round">
  <path stroke="none" d="M0 0h24v24H0z"/>
  <circle cx="8" cy="12" r="2" />
  <rect x="2" y="6" width="20" height="12" rx="6" />
</svg>



                    <svg xmlns="http://www.w3.org/2000/svg" class="icon icon-tabler icon-tabler-toggle-right" width="24" height="24" viewBox="0 0 24 24" stroke-width="2" stroke="currentColor" fill="none" stroke-linecap="round" stroke-linejoin="round">
  <path stroke="none" d="M0 0h24v24H0z"/>
  <circle cx="16" cy="12" r="2" />
  <rect x="2" y="6" width="20" height="12" rx="6" />
</svg>



                    <span>Dark Mode</span>
                </li>
            
        </div>
    </ol>
</aside>

    <aside class="sidebar right-sidebar sticky">
        
            
                
    <section class="widget archives">
        <div class="widget-icon">
            <svg xmlns="http://www.w3.org/2000/svg" class="icon icon-tabler icon-tabler-hash" width="24" height="24" viewBox="0 0 24 24" stroke-width="2" stroke="currentColor" fill="none" stroke-linecap="round" stroke-linejoin="round">
  <path stroke="none" d="M0 0h24v24H0z"/>
  <line x1="5" y1="9" x2="19" y2="9" />
  <line x1="5" y1="15" x2="19" y2="15" />
  <line x1="11" y1="4" x2="7" y2="20" />
  <line x1="17" y1="4" x2="13" y2="20" />
</svg>



        </div>
        <h2 class="widget-title section-title">Table of contents</h2>
        
        <div class="widget--toc">
            <nav id="TableOfContents">
  <ol>
    <li><a href="#introduction">Introduction</a></li>
    <li><a href="#preprocess">preprocess</a>
      <ol>
        <li><a href="#Ë¥®Êéßfastp">Ë¥®ÊéßÔºöfastp</a></li>
        <li><a href="#ÂéªÂÆø‰∏ªbowtie2">ÂéªÂÆø‰∏ªÔºöbowtie2</a></li>
        <li><a href="#Âü∫Êú¨‰ø°ÊÅØÁªüËÆ°">Âü∫Êú¨‰ø°ÊÅØÁªüËÆ°</a></li>
      </ol>
    </li>
    <li><a href="#reads-based">reads-based</a>
      <ol>
        <li><a href="#Áâ©ÁßçÊ≥®Èáäkraken2">Áâ©ÁßçÊ≥®ÈáäÔºökraken2</a>
          <ol>
            <li><a href="#ËæìÂá∫Êñá‰ª∂Ê†ºÂºè">ËæìÂá∫Êñá‰ª∂Ê†ºÂºè</a></li>
          </ol>
        </li>
        <li><a href="#humann">HUMAnN</a></li>
      </ol>
    </li>
    <li><a href="#contigs-based">contigs-based</a>
      <ol>
        <li><a href="#ÁªÑË£Ömegahit">ÁªÑË£ÖÔºömegahit</a></li>
        <li><a href="#ÁªÑË£ÖËØÑ‰º∞quast">ÁªÑË£ÖËØÑ‰º∞ÔºöQUAST</a></li>
        <li><a href="#Âü∫Âõ†È¢ÑÊµãprodigal">Âü∫Âõ†È¢ÑÊµãÔºöProdigal</a></li>
        <li><a href="#ÂéªÂÜó‰Ωô">ÂéªÂÜó‰Ωô</a>
          <ol>
            <li><a href="#cd-hit">Cd-hit</a></li>
            <li><a href="#mmseq2">mmseq2</a></li>
          </ol>
        </li>
        <li><a href="#Âü∫Âõ†ÂÆöÈáèsalmon">Âü∫Âõ†ÂÆöÈáèÔºösalmon</a></li>
        <li><a href="#ÂäüËÉΩÂü∫Âõ†Ê≥®Èáä">ÂäüËÉΩÂü∫Âõ†Ê≥®Èáä</a>
          <ol>
            <li><a href="#1-eggnogcogkeggcazy">1 eggNOG(COG/KEGG/CAZy)</a></li>
            <li><a href="#2-Á¢≥Ê∞¥ÂåñÂêàÁâ©dbcan2">2 Á¢≥Ê∞¥ÂåñÂêàÁâ©dbCAN2</a></li>
            <li><a href="#3-argscard">3 ARGsÔºàCARDÔºâ</a></li>
            <li><a href="#4-ÊØíÂäõÂõ†Â≠êvfdb">4 ÊØíÂäõÂõ†Â≠êVFDB</a></li>
            <li><a href="#5-ÂÖ∂‰ªñÂêÑÁßçÊï∞ÊçÆÂ∫ì">5 ÂÖ∂‰ªñÂêÑÁßçÊï∞ÊçÆÂ∫ì</a></li>
          </ol>
        </li>
        <li><a href="#ÂäüËÉΩÊ≥®ÈáäÂêàÂπ∂">ÂäüËÉΩÊ≥®ÈáäÂêàÂπ∂</a></li>
      </ol>
    </li>
    <li><a href="#binning">binning</a></li>
    <li><a href="#reference">Reference</a></li>
  </ol>
</nav>
        </div>
    </section>

            
        
    </aside>


            <main class="main full-width">
    <article class="main-article">
    <header class="article-header">

    <div class="article-details">
    
    <header class="article-category">
        
            <a href="/categories/metagenomic/" style="background-color: #8dd3c7; color: #fff;">
                metagenomic
            </a>
        
    </header>
    

    <div class="article-title-wrapper">
        <h2 class="article-title">
            <a href="/p/metagenomic-workflow/">Metagenomic workflow</a>
        </h2>
    
        
        <h3 class="article-subtitle">
            ÂÆèÂü∫Âõ†ÁªÑÔºàMetagenomeÔºâÊòØÊåáÂØπ‰∏Ä‰∏™ÁîüÊÄÅÁ≥ªÁªü‰∏≠ÁöÑÊâÄÊúâÂæÆÁîüÁâ©ËøõË°åDNAÂàÜÊûêÁöÑËøáÁ®ãÔºåÂèØ‰ª•Â∏ÆÂä©Á†îÁ©∂‰∫∫Âëò‰∫ÜËß£ÂæÆÁîüÁâ©ÁöÑÂ§öÊ†∑ÊÄß„ÄÅÂäüËÉΩÂíå‰∫í‰ΩúÂÖ≥Á≥ª„ÄÇËøôÈáå‰ªãÁªçÂ∏∏Áî®ÂàÜÊûêÊµÅÁ®ã„ÄÇ
        </h3>
        
    </div>

    
    
    
    
    <footer class="article-time">
        
            <div>
                <svg xmlns="http://www.w3.org/2000/svg" class="icon icon-tabler icon-tabler-calendar-time" width="56" height="56" viewBox="0 0 24 24" stroke-width="2" stroke="currentColor" fill="none" stroke-linecap="round" stroke-linejoin="round">
  <path stroke="none" d="M0 0h24v24H0z"/>
  <path d="M11.795 21h-6.795a2 2 0 0 1 -2 -2v-12a2 2 0 0 1 2 -2h12a2 2 0 0 1 2 2v4" />
  <circle cx="18" cy="18" r="4" />
  <path d="M15 3v4" />
  <path d="M7 3v4" />
  <path d="M3 11h16" />
  <path d="M18 16.496v1.504l1 1" />
</svg>
                <time class="article-time--published">Mar 26, 2023</time>
            </div>
        

        
            <div>
                <svg xmlns="http://www.w3.org/2000/svg" class="icon icon-tabler icon-tabler-clock" width="24" height="24" viewBox="0 0 24 24" stroke-width="2" stroke="currentColor" fill="none" stroke-linecap="round" stroke-linejoin="round">
  <path stroke="none" d="M0 0h24v24H0z"/>
  <circle cx="12" cy="12" r="9" />
  <polyline points="12 7 12 12 15 15" />
</svg>



                <time class="article-time--reading">
                    10 minute read
                </time>
            </div>
        
    </footer>
    

    
</div>

</header>

    <section class="article-content">
    
    
    <h2 id="introduction">Introduction</h2>
<p>ÂÆèÂü∫Âõ†ÁªÑÔºàMetagenomeÔºâÊòØÊåáÂØπ‰∏Ä‰∏™ÁîüÊÄÅÁ≥ªÁªü‰∏≠ÁöÑÊâÄÊúâÂæÆÁîüÁâ©ËøõË°åDNAÂàÜÊûêÁöÑËøáÁ®ãÔºåÂèØ‰ª•Â∏ÆÂä©Á†îÁ©∂‰∫∫Âëò‰∫ÜËß£ÂæÆÁîüÁâ©ÁöÑÂ§öÊ†∑ÊÄß„ÄÅÂäüËÉΩÂíå‰∫í‰ΩúÂÖ≥Á≥ª„ÄÇ</p>
<p>ÂÆèÂü∫Âõ†ÁªÑÁöÑÂ∫îÁî®ÈùûÂ∏∏ÂπøÊ≥õÔºåÂåÖÊã¨Ôºö</p>
<ul>
<li>
<p>ÁîüÁâ©Â§öÊ†∑ÊÄßÁ†îÁ©∂ÔºöÈÄöËøáÂØπÂÆèÂü∫Âõ†ÁªÑËøõË°åÂàÜÊûêÔºåÂèØ‰ª•‰∫ÜËß£‰∏çÂêåÁîüÊÄÅÁ≥ªÁªü‰∏≠ÂæÆÁîüÁâ©ÁöÑÂ§öÊ†∑ÊÄßÂíåÂàÜÂ∏ÉÊÉÖÂÜµ„ÄÇ</p>
</li>
<li>
<p>ÁîüÊÄÅÂ≠¶Á†îÁ©∂ÔºöÂÆèÂü∫Âõ†ÁªÑÂèØ‰ª•Â∏ÆÂä©Á†îÁ©∂‰∫∫Âëò‰∫ÜËß£ÂæÆÁîüÁâ©Âú®ÁîüÊÄÅÁ≥ªÁªü‰∏≠ÁöÑÂäüËÉΩ„ÄÅ‰∫í‰ΩúÂÖ≥Á≥ªÂíåÁîüÊÄÅ‰ΩçÁ≠â„ÄÇ</p>
</li>
<li>
<p>ÁîüÁâ©ÊäÄÊúØÔºöÂÆèÂü∫Âõ†ÁªÑÂèØ‰ª•Áî®‰∫éÁ≠õÈÄâÂÖ∑ÊúâÁâπÂÆöÂäüËÉΩÁöÑÂæÆÁîüÁâ©Ôºå‰æãÂ¶ÇÔºåÂØªÊâæËÉΩÂ§üÈôçËß£ÊúâÂÆ≥Áâ©Ë¥®ÁöÑÂæÆÁîüÁâ©„ÄÇ</p>
</li>
</ul>
<p>ÂÆèÂü∫Âõ†ÁªÑÁöÑÂàÜÊûê‰∏ÄËà¨ÂåÖÊã¨‰ª•‰∏ãÊ≠•È™§Ôºö</p>
<ol>
<li>
<p>DNAÊèêÂèñ‰∏éÂª∫Â∫ì„ÄÇ</p>
</li>
<li>
<p>È´òÈÄöÈáèÊµãÂ∫èÔºö‰ΩøÁî®È´òÈÄöÈáèÊµãÂ∫èÊäÄÊúØÂØπÊâ©Â¢ûÂêéÁöÑDNAËøõË°åÊµãÂ∫èÔºåÂæóÂà∞ÂéüÂßãÂ∫èÂàóÊï∞ÊçÆ„ÄÇ</p>
</li>
<li>
<p>Êï∞ÊçÆÊ∏ÖÊ¥óÂíåÁªÑË£ÖÔºöÂØπÂéüÂßãÊï∞ÊçÆËøõË°åË¥®ÈáèÊéßÂà∂„ÄÅÂéªÈô§‰ΩéË¥®ÈáèÂ∫èÂàóÂíåÂÜó‰ΩôÂ∫èÂàóÔºåÂ∞ÜÂ∫èÂàóÊãºÊé•ÊàêËæÉÈïøÁöÑËøûÁª≠Â∫èÂàóÔºàcontigsÔºâ„ÄÇ</p>
</li>
<li>
<p>Âü∫Âõ†Ê≥®ÈáäÔºöÂ∞Ücontigs‰∏≠ÁöÑÂü∫Âõ†ËøõË°åÊ≥®ÈáäÔºåÂæóÂà∞Âü∫Âõ†ÂäüËÉΩ‰ø°ÊÅØ„ÄÇ</p>
</li>
<li>
<p>Êï∞ÊçÆÂàÜÊûêÔºö‰∫ÜËß£ÂæÆÁîüÁâ©Â§öÊ†∑ÊÄß„ÄÅÁæ§ËêΩÁªìÊûÑ„ÄÅÂäüËÉΩÁâπÂæÅÁ≠â‰ø°ÊÅØÔºàÊõ¥Â§öÊòØÊåáËé∑Âèñ‰∫ÜÁâ©Áßç‰∏∞Â∫¶Ë°®ÊàñÂäüËÉΩ‰∏∞Â∫¶Ë°®‰πãÂêéÁöÑËøõ‰∏ÄÊ≠•ÂàÜÊûêÔºâ„ÄÇ</p>
</li>
<li>
<p>MAGs binningÔºå ËøõÂåñÂä®ÊÄÅÁ≠âËøõ‰∏ÄÊ≠•ÂàÜÊûê</p>
</li>
</ol>
<p>ËøôÊòØÊàëÂ∏∏Áî®ÁöÑ‰∏ÄÂ•óÂü∫Êú¨ÊµÅÁ®ã(Figure <a href="#fig:1-work">1</a>)ÔºåÂΩìÁÑ∂Âú®Èù¢ÂØπ‰∏çÂêåÈ°πÁõÆÊó∂Â∫îËØ•Êúâ‰∏çÂêåÁöÑ‰æßÈáçÁÇπÂíåÈÄÇÁî®ÁöÑÂàÜÊûêÊñπÊ≥ïÔºåÂèØ‰ª•Âú®Ê≠§Âü∫Á°Ä‰∏äÊ∑ªÂä†Êàñ‰øÆÊîπ„ÄÇ</p>
<p>ÊúÄÊó©ËøôÊñπÈù¢ÁöÑÂàÜÊûêÊàëÈÉΩÊòØÂèÇËÄÉÂàòÊ∞∏Èë´ËÄÅÂ∏àÁöÑ<a class="link" href="https://github.com/YongxinLiu/EasyMetagenome"  target="_blank" rel="noopener"
    ><strong>EasyMetagenome</strong></a>,Áé∞Âú®ËøôÂ•óÊµÅÁ®ã‰πüÂèëÊñáÁ´†‰∫Ü (<a class="link" href="#ref-liuPracticalGuideAmplicon2021" ><em>1</em></a>)ÔºåÂÄºÂæóÂèÇËÄÉÔºåÂØπ‰∏äÊâã16SÊµãÂ∫èÊï∞ÊçÆÊàñÂÆèÂü∫Âõ†ÁªÑÊï∞ÊçÆÈÉΩÂæàÊúâÂ∏ÆÂä©„ÄÇ</p>
<div class="figure">
<img src="images/workflow.png" alt="Basic workflow" width="80%" />
<p class="caption">
Figure 1: Basic workflow
</p>
</div>
<h2 id="preprocess">preprocess</h2>
<p>ÁªùÂ§ßÂ§öÊï∞ËøôÈáå‰ªãÁªçÁöÑËΩØ‰ª∂ÈÉΩÊòØ‰ªÖÊîØÊåÅlinuxÂπ≥Âè∞ÁöÑÔºåÊàë‰ª¨ÂÅöÊµãÂ∫èÊñá‰ª∂ÁöÑ‰∏äÊ∏∏ÂàÜÊûê‰πüËÇØÂÆöÊòØÂú®ÊúçÂä°Âô®‰∏äÂÅöÔºå‰∏™‰∫∫PC‰∏ÄËà¨ÂæàÈöæÊª°Ë∂≥ÈúÄÊ±ÇÔºåÊâÄ‰ª•Âú®ÂÅöËøô‰∫õÂàÜÊûêÂâçÂøÖÈ°ªÂÖàÂ≠¶‰π†linuxÂü∫Á°ÄÁü•ËØÜÂ¶ÇÊñá‰ª∂Á≥ªÁªüÔºåshellËÑöÊú¨ÁºñÂÜôÔºåËΩØ‰ª∂ÂÆâË£ÖÁ≠â„ÄÇ</p>
<p>ÂÆâË£ÖËΩØ‰ª∂Âª∫ËÆÆ‰ΩøÁî®condaÊàñmambaÔºàÊñ∞Âª∫ÁéØÂ¢ÉÂíåÁÆ°ÁêÜÔºâÔºåÊúâÂæàÂ§öÂèÇËÄÉÊñπÊ≥ï„ÄÇ</p>
<p>Êàë‰ª¨ÊúçÂä°Âô®‰ΩøÁî®ÁöÑÊòØslurm‰Ωú‰∏öÁÆ°ÁêÜÁ≥ªÁªüÔºåÂ∞ΩÈáèÂÖàÂ≠¶‰π†‰∏Ä‰∏ã<a class="link" href="https://docs.hpc.sjtu.edu.cn/job/slurm.html"  target="_blank" rel="noopener"
    >slurmÁöÑ‰ΩøÁî®</a>ÂÜçÂ∞ùËØïÊèê‰∫§‰Ωú‰∏ö„ÄÇ</p>
<p>‰∏ÄËà¨ÊääÊâÄÊúâÊ†∑Êú¨ÁöÑÊµãÂ∫èÂèåÁ´ØÊñá‰ª∂ÊîæÂú®‰∏Ä‰∏™Êñá‰ª∂Â§π‰∏ã</p>
<h3 id="Ë¥®Êéßfastp">Ë¥®ÊéßÔºöfastp</h3>
<div class="highlight"><div class="chroma">
<table class="lntable"><tr><td class="lntd">
<pre tabindex="0" class="chroma"><code><span class="lnt"> 1
</span><span class="lnt"> 2
</span><span class="lnt"> 3
</span><span class="lnt"> 4
</span><span class="lnt"> 5
</span><span class="lnt"> 6
</span><span class="lnt"> 7
</span><span class="lnt"> 8
</span><span class="lnt"> 9
</span><span class="lnt">10
</span><span class="lnt">11
</span><span class="lnt">12
</span><span class="lnt">13
</span><span class="lnt">14
</span><span class="lnt">15
</span><span class="lnt">16
</span><span class="lnt">17
</span><span class="lnt">18
</span><span class="lnt">19
</span><span class="lnt">20
</span><span class="lnt">21
</span><span class="lnt">22
</span><span class="lnt">23
</span><span class="lnt">24
</span></code></pre></td>
<td class="lntd">
<pre tabindex="0" class="chroma"><code class="language-bash" data-lang="bash"><span class="line"><span class="cl"><span class="cp">#!/bin/bash
</span></span></span><span class="line"><span class="cl"><span class="cp"></span><span class="c1">#SBATCH --job-name=fastp</span>
</span></span><span class="line"><span class="cl"><span class="c1">#SBATCH --output=/share/home/jianglab/pengchen/work/asthma/fastp/log/%x_%a.out</span>
</span></span><span class="line"><span class="cl"><span class="c1">#SBATCH --error=/share/home/jianglab/pengchen/work/asthma/fastp/log/%x_%a.err</span>
</span></span><span class="line"><span class="cl"><span class="c1">#SBATCH --array=1-33</span>
</span></span><span class="line"><span class="cl"><span class="c1">#SBATCH --partition=short</span>
</span></span><span class="line"><span class="cl"><span class="c1">#SBATCH --cpus-per-task=8</span>
</span></span><span class="line"><span class="cl">
</span></span><span class="line"><span class="cl">
</span></span><span class="line"><span class="cl"><span class="nb">echo</span> start: <span class="sb">`</span>date +<span class="s1">&#39;%Y-%m-%d %T&#39;</span><span class="sb">`</span>
</span></span><span class="line"><span class="cl"><span class="nv">start</span><span class="o">=</span><span class="sb">`</span>date +%s<span class="sb">`</span>
</span></span><span class="line"><span class="cl"><span class="nb">echo</span> <span class="s2">&#34;SLURM_ARRAY_TASK_ID: &#34;</span> <span class="nv">$SLURM_ARRAY_TASK_ID</span>
</span></span><span class="line"><span class="cl"><span class="nv">sample</span><span class="o">=</span><span class="k">$(</span>head -n <span class="nv">$SLURM_ARRAY_TASK_ID</span> ~/work/asthma/data/namelist <span class="p">|</span> tail -1<span class="k">)</span>
</span></span><span class="line"><span class="cl"><span class="c1">#sample=$(head -n 1 namelist | tail -1)</span>
</span></span><span class="line"><span class="cl"><span class="nb">echo</span> handling: <span class="nv">$sample</span>
</span></span><span class="line"><span class="cl"><span class="c1">####################</span>
</span></span><span class="line"><span class="cl">
</span></span><span class="line"><span class="cl">fastp -w <span class="m">8</span> -i ~/work/asthma/data/<span class="nv">$sample</span>/<span class="nv">$sample</span><span class="s1">&#39;_f1.fastq&#39;</span> -o <span class="si">${</span><span class="nv">sample</span><span class="si">}</span>_1 <span class="se">\
</span></span></span><span class="line"><span class="cl"><span class="se"></span>-I ~/work/asthma/data/<span class="nv">$sample</span>/<span class="nv">$sample</span><span class="s1">&#39;_r2.fastq&#39;</span> -O <span class="si">${</span><span class="nv">sample</span><span class="si">}</span>_2 -j ~/work/asthma/fastp/<span class="si">${</span><span class="nv">i</span><span class="si">}</span>.json
</span></span><span class="line"><span class="cl">
</span></span><span class="line"><span class="cl"><span class="c1">####################</span>
</span></span><span class="line"><span class="cl"><span class="nb">echo</span> end: <span class="sb">`</span>date +<span class="s1">&#39;%Y-%m-%d %T&#39;</span><span class="sb">`</span>
</span></span><span class="line"><span class="cl"><span class="nv">end</span><span class="o">=</span><span class="sb">`</span>date +%s<span class="sb">`</span>
</span></span><span class="line"><span class="cl"><span class="nb">echo</span> TIME:<span class="sb">`</span>expr <span class="nv">$end</span> - <span class="nv">$start</span><span class="sb">`</span>s
</span></span></code></pre></td></tr></table>
</div>
</div><p>ÂêéÈù¢Êé•‰∏Ä‰∏™pythonËÑöÊú¨Â∞±ÂèØ‰ª•ÁªüËÆ°Â∏∏Áî®ÊåáÊ†á‰∫Ü„ÄÇ</p>
<p>ÊääÊâÄÊúâÁöÑ.jsonÊñá‰ª∂ÁßªÂà∞‰∏Ä‰∏™Êñá‰ª∂Â§πÈáåÔºåreport/‰∏ãÔºåÂ∞±ÂèØ‰ª•ÁªüËÆ°‰∫Ü„ÄÇ</p>
<h3 id="ÂéªÂÆø‰∏ªbowtie2">ÂéªÂÆø‰∏ªÔºöbowtie2</h3>
<p>ÂÖ∂ÂÆûÂ∞±ÊòØÂ∞ÜÂ∫èÂàóÊØîÂØπÂà∞‰∫∫Á±ªÂü∫Âõ†ÁªÑ‰∏äÔºåÊ≤°ÊúâÊØîÂØπÂà∞ÁöÑÂ∫èÂàóÊï¥ÂêàÊàêÊñ∞Êñá‰ª∂Â∞±ÊòØÂéªÂÆø‰∏ªÂêéÁöÑ‰∫Ü„ÄÇ</p>
<div class="highlight"><div class="chroma">
<table class="lntable"><tr><td class="lntd">
<pre tabindex="0" class="chroma"><code><span class="lnt"> 1
</span><span class="lnt"> 2
</span><span class="lnt"> 3
</span><span class="lnt"> 4
</span><span class="lnt"> 5
</span><span class="lnt"> 6
</span><span class="lnt"> 7
</span><span class="lnt"> 8
</span><span class="lnt"> 9
</span><span class="lnt">10
</span><span class="lnt">11
</span><span class="lnt">12
</span><span class="lnt">13
</span><span class="lnt">14
</span><span class="lnt">15
</span><span class="lnt">16
</span><span class="lnt">17
</span><span class="lnt">18
</span><span class="lnt">19
</span></code></pre></td>
<td class="lntd">
<pre tabindex="0" class="chroma"><code class="language-bash" data-lang="bash"><span class="line"><span class="cl"><span class="cp">#!/bin/bash
</span></span></span><span class="line"><span class="cl"><span class="cp"></span><span class="c1">#SBATCH --job-name=rm_human</span>
</span></span><span class="line"><span class="cl"><span class="c1">#SBATCH --output=/share/home/jianglab/pengchen/work/meta/%x_%a.out</span>
</span></span><span class="line"><span class="cl"><span class="c1">#SBATCH --error=/share/home/jianglab/pengchen/work/meta/%x_%a.err</span>
</span></span><span class="line"><span class="cl"><span class="c1">#SBATCH --cpus-per-task=32</span>
</span></span><span class="line"><span class="cl"><span class="c1">#SBATCH --partition=short</span>
</span></span><span class="line"><span class="cl">
</span></span><span class="line"><span class="cl"><span class="nb">echo</span> start: <span class="sb">`</span>date +<span class="s1">&#39;%Y-%m-%d %T&#39;</span><span class="sb">`</span>
</span></span><span class="line"><span class="cl"><span class="nv">start</span><span class="o">=</span><span class="sb">`</span>date +%s<span class="sb">`</span>
</span></span><span class="line"><span class="cl"><span class="c1">#############</span>
</span></span><span class="line"><span class="cl"><span class="k">for</span> i in C1 C2
</span></span><span class="line"><span class="cl"><span class="k">do</span>
</span></span><span class="line"><span class="cl">bowtie2 -p <span class="m">32</span> -x ~/db/humangenome/hg38 -1 seq/<span class="si">${</span><span class="nv">i</span><span class="si">}</span>_1.fq.gz <span class="se">\
</span></span></span><span class="line"><span class="cl"><span class="se"></span> -2 seq/<span class="si">${</span><span class="nv">i</span><span class="si">}</span>_2.fq.gz -S <span class="si">${</span><span class="nv">i</span><span class="si">}</span>.sam --un-conc <span class="si">${</span><span class="nv">i</span><span class="si">}</span>.fq --very-sensitive
</span></span><span class="line"><span class="cl"><span class="k">done</span>
</span></span><span class="line"><span class="cl"><span class="c1">##############</span>
</span></span><span class="line"><span class="cl"><span class="nb">echo</span> end: <span class="sb">`</span>date +<span class="s1">&#39;%Y-%m-%d %T&#39;</span><span class="sb">`</span>
</span></span><span class="line"><span class="cl"><span class="nv">end</span><span class="o">=</span><span class="sb">`</span>date +%s<span class="sb">`</span>
</span></span><span class="line"><span class="cl"><span class="nb">echo</span> TIME:<span class="sb">`</span>expr <span class="nv">$end</span> - <span class="nv">$start</span><span class="sb">`</span>s
</span></span></code></pre></td></tr></table>
</div>
</div><h3 id="Âü∫Êú¨‰ø°ÊÅØÁªüËÆ°">Âü∫Êú¨‰ø°ÊÅØÁªüËÆ°</h3>
<p>ÂèØ‰ª•Áî®FastqCountÔºö</p>
<div class="highlight"><div class="chroma">
<table class="lntable"><tr><td class="lntd">
<pre tabindex="0" class="chroma"><code><span class="lnt">1
</span><span class="lnt">2
</span><span class="lnt">3
</span><span class="lnt">4
</span></code></pre></td>
<td class="lntd">
<pre tabindex="0" class="chroma"><code class="language-bash" data-lang="bash"><span class="line"><span class="cl">~/biosoft/FastqCount-master/FastqCount_v0.5 xx.fastq.gz
</span></span><span class="line"><span class="cl">
</span></span><span class="line"><span class="cl">Total Reads     Total Bases     N Bases Q20     Q30     GC
</span></span><span class="line"><span class="cl"><span class="m">11568822</span> <span class="o">(</span>11.57 M<span class="o">)</span>      <span class="m">1702829127</span> <span class="o">(</span>1.70 G<span class="o">)</span>     0.00%   98.00%  94.00%  54.00%
</span></span></code></pre></td></tr></table>
</div>
</div><h2 id="reads-based">reads-based</h2>
<h3 id="Áâ©ÁßçÊ≥®Èáäkraken2">Áâ©ÁßçÊ≥®ÈáäÔºökraken2</h3>
<p>Kraken 2ÊòØ‰∏Ä‰∏™Áî®‰∫éÂØπÈ´òÈÄöÈáèÊµãÂ∫èÊï∞ÊçÆËøõË°åÂàÜÁ±ªÂíåÊ†áËØÜÁâ©ÁßçÁöÑËΩØ‰ª∂„ÄÇÂÆÉ‰ΩøÁî®ÂèÇËÄÉÊï∞ÊçÆÂ∫ì‰∏≠ÁöÑÂü∫Âõ†ÁªÑÂ∫èÂàóÊù•ËøõË°åÂàÜÁ±ªÔºåÂπ∂‰ΩøÁî®k-merÊñπÊ≥ïÊù•ÂÆûÁé∞Âø´ÈÄüÂíåÂáÜÁ°ÆÁöÑÂàÜÁ±ª„ÄÇ</p>
<p>‰ΩøÁî®Kraken 2ËøõË°åÂü∫Êú¨ÂàÜÁ±ªÁöÑÁÆÄÂçïÊ≠•È™§Ôºö</p>
<ol>
<li>
<p>ÂáÜÂ§áÂèÇËÄÉÊï∞ÊçÆÂ∫ìÔºöKraken 2ÈúÄË¶Å‰∏Ä‰∏™ÂèÇËÄÉÊï∞ÊçÆÂ∫ìÔºå‰ª•‰æøÂØπÊµãÂ∫èÊï∞ÊçÆËøõË°åÂàÜÁ±ª„ÄÇÂèØ‰ª•‰ªéNCBI„ÄÅEnsemblÊàñÂÖ∂‰ªñÊï∞ÊçÆÂ∫ì‰∏ãËΩΩÁõ∏Â∫îÁöÑÂü∫Âõ†ÁªÑÂ∫èÂàóÔºåÂπ∂‰ΩøÁî®Kraken 2ÂÜÖÁΩÆÁöÑÂ∑•ÂÖ∑Êù•ÊûÑÂª∫Êï∞ÊçÆÂ∫ì„ÄÇ</p>
</li>
<li>
<p>ÂÆâË£ÖKraken 2ÔºöÂèØ‰ª•‰ªéKraken 2ÂÆòÊñπÁΩëÁ´ô‰∏ãËΩΩÂπ∂ÂÆâË£ÖKraken 2ËΩØ‰ª∂„ÄÇ</p>
</li>
<li>
<p>ËøêË°åKraken 2Ôºö‰ΩøÁî®Kraken 2ÂØπÊµãÂ∫èÊï∞ÊçÆËøõË°åÂàÜÁ±ªÈúÄË¶Å‰ΩøÁî®‰ª•‰∏ãÂëΩ‰ª§Ôºö</p>
</li>
</ol>
<p><code>kraken2 \--db \&lt;path_to_database\&gt; \&lt;input_file\&gt; \--output \&lt;output_file\&gt;</code></p>
<p>ËøôÈáåÔºå**<code>&lt;path_to_database&gt;</code><strong>ÊòØÂèÇËÄÉÊï∞ÊçÆÂ∫ìÁöÑË∑ØÂæÑÔºå</strong><code>&lt;input_file&gt;</code><strong>ÊòØÈúÄË¶ÅËøõË°åÂàÜÁ±ªÁöÑËæìÂÖ•Êñá‰ª∂Ôºå</strong><code>&lt;output_file&gt;</code>**ÊòØËæìÂá∫Êñá‰ª∂ÁöÑÂêçÁß∞„ÄÇKraken 2Â∞ÜËæìÂá∫‰∏Ä‰∏™ÂàÜÁ±ªÊä•ÂëäÊñá‰ª∂Âíå‰∏Ä‰∏™Â∫èÂàóÊñá‰ª∂„ÄÇ</p>
<p><code>kraken2-build --standard --threads 24 --db ./</code></p>
<p>&ndash;standardÊ†áÂáÜÊ®°Âºè‰∏ãÂè™‰∏ãËΩΩ5ÁßçÊï∞ÊçÆÂ∫ìÔºöÂè§Ëèåarchaea„ÄÅÁªÜËèåbacteria„ÄÅ‰∫∫Á±ªhuman„ÄÅËΩΩ‰ΩìUniVec_Core„ÄÅÁóÖÊØíviral„ÄÇ‰πüÂèØÈÄâÁõ¥Êé•‰∏ãËΩΩ‰ΩúËÄÖÊûÑÂª∫ÁöÑÁ¥¢ÂºïÔºåËøòÂåÖÊã¨brackenÁöÑÁ¥¢Âºï„ÄÇ</p>
<p>Ëøô‰∏™krakenÊï∞ÊçÆÂ∫ìÊòØÂèØ‰ª•Ëá™Â∑±ÊûÑÂª∫ÁöÑÔºåÊâÄ‰ª•ÈÄÇÁî®‰∫éÂêÑÁßçÈ°πÁõÆÁöÑÁâ©ÁßçÊ≥®ÈáäÔºåÊàëÂÅöÁöÑÊØîËæÉÂ§öÁöÑÊòØÁéØÂ¢ÉÊ†∑Êú¨ÁöÑÂÆèÂü∫Âõ†ÁªÑÔºåÂ∞±ÂèØËÉΩÈúÄË¶ÅÊõ¥ÂÖ®Èù¢ÁöÑÁâ©ÁßçÊï∞ÊçÆÂ∫ìÔºàÁîöËá≥Èô§‰∫ÜÂêÑÁßçÂæÆÁîüÁâ©ÔºåËøòË¶ÅÂä®Ê§çÁâ©Êï∞ÊçÆÁ≠âÔºâÔºåÂÆûÈ™åÂÆ§ÁöÑWXÂ∏àÂßêÊî∂ÈõÜÊûÑÂª∫‰∫Ü‰∏Ä‰∏™Ë∂ÖÂ§ßÁöÑÁâ©ÁßçÂ∫ì„ÄÇ</p>
<p>ÈúÄË¶ÅÊ≥®ÊÑèÁöÑÊòØkrakenËøêË°åËá≥Â∞ëË¶ÅÊèê‰æõÊï∞ÊçÆÂ∫ìÂ§ßÂ∞èÁöÑÂÜÖÂ≠òÂ§ßÂ∞èÔºàËøêË°åÂÜÖÂ≠òÔºâÔºåÂõ†‰∏∫ÂÆÉ‰ºöÊääÊï¥‰∏™Êï∞ÊçÆÂ∫ìËΩΩÂÖ•ÂÜÖÂ≠òÂêéËøõË°åÂ∫èÂàóÁöÑÊ≥®ÈáäÔºåÊâÄ‰ª•Â¶ÇÊûúÂèëÁé∞Êó†Ê≥ïËΩΩÂÖ•Êï∞ÊçÆÂ∫ìÁöÑÊä•ÈîôÔºåÂèØ‰ª•Â∞ùËØïË∞ÉÂ§ßÂÜÖÂ≠òËµÑÊ∫ê„ÄÇ</p>
<p>krakenËΩØ‰ª∂ËøêË°åÊó∂ËΩΩÂÖ•Êï∞ÊçÆÂ∫ìÊòØ‰∏Ä‰∏™ÂçÅÂàÜËÄóÊó∂ÁöÑÊ≠•È™§ÔºåËÄåÊØèÊù°Â∫èÂàóÁöÑÈâ¥ÂÆöÊó∂Èó¥Â∑Æ‰∏çÂ§öÔºåÊâÄ‰ª•Êàë‰ª¨ÂèØ‰ª•Â∞ÜÂæàÂ§öÊ†∑Êú¨ÁöÑfastqÊñá‰ª∂ÂêàÂπ∂Êàê‰∏Ä‰∏™Â§ßÊñá‰ª∂ÂêéËæìÂÖ•krakenÊ≥®ÈáäÔºå‰πãÂêéÂÜçÊåâÁÖßÂ∫èÂàóÁöÑÊï∞ÈáèÊãÜÂàÜÁªìÊûúÊñá‰ª∂ÔºåËøôÊ†∑Â§ö‰∏™Ê†∑Êú¨‰πüÂè™ÈúÄË¶ÅËΩΩÂÖ•‰∏ÄÊ¨°Êï∞ÊçÆÂ∫ìÔºåËäÇÁúÅÊó∂Èó¥„ÄÇ</p>
<div class="highlight"><div class="chroma">
<table class="lntable"><tr><td class="lntd">
<pre tabindex="0" class="chroma"><code><span class="lnt"> 1
</span><span class="lnt"> 2
</span><span class="lnt"> 3
</span><span class="lnt"> 4
</span><span class="lnt"> 5
</span><span class="lnt"> 6
</span><span class="lnt"> 7
</span><span class="lnt"> 8
</span><span class="lnt"> 9
</span><span class="lnt">10
</span><span class="lnt">11
</span><span class="lnt">12
</span><span class="lnt">13
</span><span class="lnt">14
</span><span class="lnt">15
</span><span class="lnt">16
</span><span class="lnt">17
</span><span class="lnt">18
</span></code></pre></td>
<td class="lntd">
<pre tabindex="0" class="chroma"><code class="language-bash" data-lang="bash"><span class="line"><span class="cl"><span class="cp">#!/bin/bash
</span></span></span><span class="line"><span class="cl"><span class="cp"></span><span class="c1">#SBATCH --job-name=kraken2M</span>
</span></span><span class="line"><span class="cl"><span class="c1">#SBATCH --output=/share/home/jianglab/pengchen/work/asthma/kraken/%x_%a.out</span>
</span></span><span class="line"><span class="cl"><span class="c1">#SBATCH --error=/share/home/jianglab/pengchen/work/asthma/kraken/%x_%a.err</span>
</span></span><span class="line"><span class="cl"><span class="c1">#SBATCH --time=14-00:00:00</span>
</span></span><span class="line"><span class="cl"><span class="c1">#SBATCH --partition=mem</span>
</span></span><span class="line"><span class="cl"><span class="c1">#SBATCH --cpus-per-task=32</span>
</span></span><span class="line"><span class="cl"><span class="c1">#SBATCH --mem-per-cpu=100G</span>
</span></span><span class="line"><span class="cl">
</span></span><span class="line"><span class="cl"><span class="nv">fqp</span><span class="o">=</span>~/work/asthma/data/CRR205159/
</span></span><span class="line"><span class="cl">python /share/home/jianglab/shared/krakenDB/K2ols/kraken2M.py -t <span class="m">32</span> <span class="se">\
</span></span></span><span class="line"><span class="cl"><span class="se"></span>    -i <span class="si">${</span><span class="nv">fqp</span><span class="si">}</span> <span class="se">\
</span></span></span><span class="line"><span class="cl"><span class="se"></span>    -c 0.05 <span class="se">\
</span></span></span><span class="line"><span class="cl"><span class="se"></span>    -s _f1.fastq,_r2.fastq <span class="se">\
</span></span></span><span class="line"><span class="cl"><span class="se"></span>    -o ~/work/asthma/kraken/ <span class="se">\
</span></span></span><span class="line"><span class="cl"><span class="se"></span>    -d /share/home/jianglab/shared/krakenDB/mydb2 <span class="se">\
</span></span></span><span class="line"><span class="cl"><span class="se"></span>    -k ~/miniconda3/envs/waste/bin/kraken2 <span class="se">\
</span></span></span><span class="line"><span class="cl"><span class="se"></span>    -kt /share/home/jianglab/shared/krakenDB/K2ols/KrakenTools 
</span></span></code></pre></td></tr></table>
</div>
</div><h4 id="ËæìÂá∫Êñá‰ª∂Ê†ºÂºè">ËæìÂá∫Êñá‰ª∂Ê†ºÂºè</h4>
<p>KrakenÊ†áÂáÜËæìÂá∫Ê†ºÂºè</p>
<p>‰∫îÂàóË°® output</p>
<ul>
<li>
<p>C/U‰ª£Ë°®ÂàÜÁ±ªclassifiedÊàñÈùûÂàÜÁ±ªunclassifed</p>
</li>
<li>
<p>Â∫èÂàóID</p>
</li>
<li>
<p>Áâ©ÁßçÊ≥®Èáä</p>
</li>
<li>
<p>ÊØîÂ∫èÂàóÊ≥®ÈáäÁöÑÂå∫ÂüüÔºåÂ¶Ç98|94‰ª£Ë°®Â∑¶Á´Ø98bpÔºåÂè≥Á´Ø94bpÊØîÂØπËá≥Êï∞ÊçÆÂ∫ì</p>
</li>
<li>
<p>LCAÊØîÂØπÁªìÊûúÔºåÂ¶Ç‚Äù562:13 561:4‚Äù‰ª£Ë°®13 k-merÊØîÂØπËá≥Áâ©Áßç#562Ôºå4 k-merÊØîÂØπËá≥#561Áâ©Áßç</p>
</li>
</ul>
<p>Êä•ÂëäËæìÂá∫Ê†ºÂºè report</p>
<p>ÂåÖÊã¨6ÂàóÔºåÊñπ‰æøÊï¥ÁêÜ‰∏ãÊ∏∏ÂàÜÊûê„ÄÇ</p>
<ol>
<li>
<p>ÁôæÂàÜÊØî</p>
</li>
<li>
<p>count</p>
</li>
<li>
<p>countÊúÄ‰ºò</p>
</li>
<li>
<p>(U)nclassified, (R)oot, (D)omain, (K)ingdom, (P)hylum, (C)lass, (O)rder, (F)amily, (G)enus, or (S)pecies. ‚ÄúG2‚Äù‰ª£Ë°®‰Ωç‰∫éÂ±û‰∏ÄÁßçÈó¥</p>
</li>
<li>
<p>NCBIÁâ©ÁßçID</p>
</li>
<li>
<p>ÁßëÂ≠¶Áâ©ÁßçÂêç</p>
</li>
</ol>
<p>Â∏∏Áî®ÁöÑÁâ©Áßç‰∏∞Â∫¶Ë°®Ê†ºÂºèÈô§‰∫Ükraken reportÔºåËøòÊúâmpaÔºåspfÔºåkronaÁ≠âÊ†ºÂºèÔºåÂÖ≥‰∫ékrakenÁªìÊûúÁöÑÊï¥ÁêÜ‰ª•ÂèäÊ†ºÂºèËΩ¨Êç¢ÊñπÂºèÔºåÊúâ‰∏Ä‰∫õÁé∞ÊàêÁöÑËÑöÊú¨ÊàñËÄÖËá™Â∑±ÂÜô„ÄÇ</p>
<p><a class="link" href="https://ccb.jhu.edu/software/krakentools/"  target="_blank" rel="noopener"
    >KrakenTools (jhu.edu)</a> Â∞±ÊòØ‰∏ÄÂ•óÂæàÂ•ΩÁî®ÁöÑkrakenÂ∑•ÂÖ∑ÂåÖÔºåÂÖ∂‰∏≠Â∏∏Áî®ÁöÑÊúâÔºö</p>
<ol>
<li>extract_kraken_reads.py</li>
</ol>
<p>Ê≠§Á®ãÂ∫èÊèêÂèñËØªÂèñÂú®‰ªª‰ΩïÁî®Êà∑ÊåáÂÆöÁöÑÂàÜÁ±ªidÂ§ÑÂàÜÁ±ªÁöÑÂÜÖÂÆπ„ÄÇÁî®Êà∑ÂøÖÈ°ªÊåáÂÆöKrakenËæìÂá∫Êñá‰ª∂„ÄÅÂ∫èÂàóÊñá‰ª∂ÂíåËá≥Â∞ë‰∏Ä‰∏™ÂàÜÁ±ªÊ≥ïID„ÄÇ‰∏ãÈù¢ÊåáÂÆö‰∫ÜÂÖ∂‰ªñÈÄâÈ°π„ÄÇÊà™Ëá≥2021Âπ¥4Êúà19Êó•ÔºåÊ≠§ËÑöÊú¨‰∏éKrakenUniq/Kraken2UniqÊä•ÂëäÂÖºÂÆπ„ÄÇ</p>
<ol start="2">
<li>combine_kreports.py</li>
</ol>
<p>This script combines multiple Kraken reports into a combined report file.</p>
<p><code>python combine_kreports.py</code></p>
<ul>
<li>
<p>-r 1.KREPORT 2.KREPORT&hellip;&hellip;&hellip;&hellip;&hellip;&hellip;&hellip;&hellip;Kraken-style reports to combine</p>
</li>
<li>
<p>-o COMBINED.KREPORT&hellip;&hellip;&hellip;&hellip;&hellip;&hellip;&hellip;&hellip;&hellip;Output file</p>
</li>
</ul>
<ol start="3">
<li>kreport2krona.py</li>
</ol>
<p>This program takes a Kraken report file and prints out a krona-compatible TEXT file</p>
<p>Êç¢ÊàêkronaÊñá‰ª∂Â•ΩÁîªÂõæ„ÄÇÂòøÂòø</p>
<p>kronaË£Ö‰∫Ü‰∏Ä‰∏™excelÁöÑÊèí‰ª∂ÂèØ‰ª•ÂæàÂÆπÊòìÁîªÂõæ<br>
<code>python kreport2krona.py</code></p>
<ul>
<li>
<p>-r/&ndash;report MYFILE.KREPORT&hellip;.‚Ä¶.Kraken report file</p>
</li>
<li>
<p>-o/&ndash;output MYFILE.KRONA&hellip;&hellip;.‚Ä¶Output Krona text file</p>
</li>
</ul>
<p>then, <code>ktImportText MYSAMPLE.krona -o MYSAMPLE.krona.html</code></p>
<p>Â•ΩÁúãÁöÑÁΩëÈ°µÂ∞±Âá∫Êù•‰∫Ü„ÄÇ</p>
<ol start="4">
<li>kreport2mpa.py</li>
</ol>
<p>This program takes a Kraken report file and prints out a mpa (MetaPhlAn) -style TEXT file</p>
<p><code>python kreport2mpa.py</code></p>
<ul>
<li>
<p>-r/&ndash;report MYFILE.KREPORT&hellip;.‚Ä¶.Kraken report file</p>
</li>
<li>
<p>-o/&ndash;output MYFILE.MPA.TXT&hellip;.‚Ä¶.Output MPA-STYLE text file</p>
</li>
</ul>
<ol start="5">
<li>combine_mpa.py</li>
</ol>
<p><code>python combine_mpa.py</code></p>
<ul>
<li>
<p>-i/&ndash;input MYFILE1.MPA MYFILE2.MPA&hellip;.‚Ä¶Multiple MPA-STYLE text files (separated by spaces)</p>
</li>
<li>
<p>-o/&ndash;output MYFILE.COMBINED.MPA&hellip;&hellip;.‚Ä¶Output MPA-STYLE text file</p>
</li>
</ul>
<p>This program combines multiple outputs from <a class="link" href="#kreport2mpapy" >kreport2mpa.py</a>. Files to be combined must have been generated using the same kreport2mpa.py options.</p>
<p><code>python combine_mpa.py -i -o --intermediate-ranks</code></p>
<h3 id="humann">HUMAnN</h3>
<p>HUMAnN2ÔºàThe HMP Unified Metabolic Analysis Network 2ÔºâÊòØ‰∏ÄÊ¨æÁî®‰∫éÂàÜÊûê‰∫∫Á±ªÂæÆÁîüÁâ©ÁªÑÁöÑÂäüËÉΩÂíå‰ª£Ë∞¢ËÉΩÂäõÁöÑÂ∑•ÂÖ∑„ÄÇÂÆÉÈÄöËøáÂ∞ÜÂÆèÂü∫Âõ†ÁªÑÂ∫èÂàó‰∏éÂèÇËÄÉÂü∫Âõ†ÁªÑÊï∞ÊçÆÂ∫ìÊØîÂØπÔºåÂà©Áî®MetaCyc‰ª£Ë∞¢ÈÄöË∑ØÊï∞ÊçÆÂ∫ìÂíåUniRefËõãÁôΩË¥®Â∫èÂàóÊï∞ÊçÆÂ∫ìÔºåÂàÜÊûêÂæÆÁîüÁâ©ÁªÑÂú®ÂäüËÉΩÂíå‰ª£Ë∞¢ÈÄöË∑ØÊ∞¥Âπ≥‰∏äÁöÑÁªÑÊàêÂíåÊ¥ªÊÄß„ÄÇHUMAnN2ËøòÊèê‰æõ‰∫ÜÂ§öÊ†∑ÊÄßÂàÜÊûê„ÄÅÂÖ≥ËÅîÂàÜÊûêÂíåÂèØËßÜÂåñÂ∑•ÂÖ∑ÔºåÂèØÁî®‰∫éÊ∑±ÂÖ•Á†îÁ©∂‰∫∫Á±ªÂæÆÁîüÁâ©ÁªÑÂØπÂÆø‰∏ªÂÅ•Â∫∑ÁöÑÂΩ±ÂìçÂíåÊ≤ªÁñóÁ≠ñÁï•ÁöÑÂà∂ÂÆöÁ≠âÊñπÈù¢„ÄÇ</p>
<p>HUMAnN2ÊòØÁî±ÁæéÂõΩÂõΩÂÆ∂‰∫∫Á±ªÂæÆÁîüÁâ©ÁªÑËÆ°ÂàíÔºàHMPÔºâÂºÄÂèëÁöÑÔºåÁõÆÂâçÊúÄÊñ∞ÁâàÊú¨‰∏∫<a class="link" href="https://github.com/biobakery/humann"  target="_blank" rel="noopener"
    >HUMAnN3</a>Ôºå‰∫é2020Âπ¥ÂèëÂ∏É„ÄÇ‰∏éHUMAnN2Áõ∏ÊØîÔºåHUMAnN3ÊîπËøõ‰∫ÜÂü∫Âõ†ÂÆ∂ÊóèÊ≥®ÈáäÁöÑÊñπÊ≥ïÔºåÊèêÈ´ò‰∫ÜÊ≥®ÈáäÁ≤æÂ∫¶ÂíåÈÄüÂ∫¶ÔºåÂπ∂Êèê‰æõ‰∫ÜÊñ∞ÁöÑÂäüËÉΩÂíåÂ∑•ÂÖ∑ÔºåÂ¶ÇÂäüËÉΩÈüßÂ∫¶ÂàÜÊûê„ÄÅ‰ª£Ë∞¢ÊåáÁ∫πËØÜÂà´ÂíåÂ§öÊ†∑ÊÄßÂàÜÊûêÁ≠â„ÄÇ</p>
<p>‰ΩÜÊòØHUMAnN2ÁöÑÊï∞ÊçÆÂ∫ìÂü∫Êú¨ÈÉΩÊòØ‰∏é‰∫∫Áõ∏ÂÖ≥ÁöÑÂæÆÁîüÁâ©ÔºåÊØîËæÉÈÄÇÂêàÂÅöÂêÑÁßç‰∫∫‰ΩìÂæÆÁîüÁâ©ÁªÑÔºàËÇ†ÈÅìÔºåËÇ∫ÈÉ®ÔºåÂè£ËÖîÔºåÁöÆËÇ§Á≠âÁ≠âÔºâÔºåÂØπ‰∫éÁéØÂ¢ÉÊ†∑Êú¨ÂèØËÉΩunclassifiedÊØîËæÉÂ§ö„ÄÇ</p>
<p><strong>HUMAnN2Ë¶ÅÊ±ÇÂèåÁ´ØÂ∫èÂàóÂêàÂπ∂ÁöÑÊñá‰ª∂‰Ωú‰∏∫ËæìÂÖ•</strong>ÔºåforÂæ™ÁéØÊ†πÊçÆÂÆûÈ™åËÆæËÆ°Ê†∑Êú¨ÂêçÊâπÈáèÂèåÁ´ØÂ∫èÂàóÂêàÂπ∂„ÄÇ</p>
<ul>
<li>
<p><strong>Áâ©ÁßçÁªÑÊàêË∞ÉÁî®MetaPhlAn2, bowtie2ÊØîÂØπËá≥Ê†∏ÈÖ∏Â∫èÂàó</strong>ÔºåËß£ÂÜ≥ÊúâÂì™‰∫õÂæÆÁîüÁâ©Â≠òÂú®ÁöÑÈóÆÈ¢òÔºõ</p>
</li>
<li>
<p><strong>ÂäüËÉΩÁªÑÊàê‰∏∫humann2Ë∞ÉÁî®diamondÊØîÂØπËá≥ËõãÁôΩÂ∫ì11Gb</strong>ÔºåËß£ÂÜ≥Ëøô‰∫õÂæÆÁîüÁâ©ÂèÇ‰∏éÂì™‰∫õÂäüËÉΩÈÄöË∑ØÁöÑÈóÆÈ¢òÔºõ</p>
</li>
</ul>
<div class="highlight"><div class="chroma">
<table class="lntable"><tr><td class="lntd">
<pre tabindex="0" class="chroma"><code><span class="lnt"> 1
</span><span class="lnt"> 2
</span><span class="lnt"> 3
</span><span class="lnt"> 4
</span><span class="lnt"> 5
</span><span class="lnt"> 6
</span><span class="lnt"> 7
</span><span class="lnt"> 8
</span><span class="lnt"> 9
</span><span class="lnt">10
</span><span class="lnt">11
</span><span class="lnt">12
</span><span class="lnt">13
</span><span class="lnt">14
</span><span class="lnt">15
</span><span class="lnt">16
</span><span class="lnt">17
</span><span class="lnt">18
</span><span class="lnt">19
</span><span class="lnt">20
</span><span class="lnt">21
</span><span class="lnt">22
</span><span class="lnt">23
</span><span class="lnt">24
</span><span class="lnt">25
</span><span class="lnt">26
</span><span class="lnt">27
</span><span class="lnt">28
</span><span class="lnt">29
</span><span class="lnt">30
</span><span class="lnt">31
</span><span class="lnt">32
</span><span class="lnt">33
</span><span class="lnt">34
</span><span class="lnt">35
</span><span class="lnt">36
</span><span class="lnt">37
</span><span class="lnt">38
</span><span class="lnt">39
</span><span class="lnt">40
</span><span class="lnt">41
</span></code></pre></td>
<td class="lntd">
<pre tabindex="0" class="chroma"><code class="language-bash" data-lang="bash"><span class="line"><span class="cl"><span class="nb">cd</span> alldata
</span></span><span class="line"><span class="cl"><span class="k">for</span> i in <span class="sb">`</span>cat ~/work/asthma/data/namelist<span class="sb">`</span>
</span></span><span class="line"><span class="cl"><span class="k">do</span>
</span></span><span class="line"><span class="cl"><span class="nb">echo</span> <span class="nv">$i</span>
</span></span><span class="line"><span class="cl">cat <span class="si">${</span><span class="nv">i</span><span class="si">}</span>_f1.fastq <span class="si">${</span><span class="nv">i</span><span class="si">}</span>_r2.fastq &gt;<span class="si">${</span><span class="nv">i</span><span class="si">}</span>_paired.fastq
</span></span><span class="line"><span class="cl"><span class="k">done</span>
</span></span><span class="line"><span class="cl">
</span></span><span class="line"><span class="cl"><span class="c1">#!/bin/bash</span>
</span></span><span class="line"><span class="cl"><span class="c1">#SBATCH --job-name=humann2</span>
</span></span><span class="line"><span class="cl"><span class="c1">#SBATCH --output=/share/home/jianglab/pengchen/work/asthma/humann/%x_%a.out</span>
</span></span><span class="line"><span class="cl"><span class="c1">#SBATCH --error=/share/home/jianglab/pengchen/work/asthma/humann/%x_%a.err</span>
</span></span><span class="line"><span class="cl"><span class="c1">#SBATCH --array=1-32</span>
</span></span><span class="line"><span class="cl"><span class="c1">#SBATCH --cpus-per-task=24</span>
</span></span><span class="line"><span class="cl"><span class="c1">#SBATCH --partition=cpu</span>
</span></span><span class="line"><span class="cl">
</span></span><span class="line"><span class="cl"><span class="nb">echo</span> start: <span class="sb">`</span>date +<span class="s1">&#39;%Y-%m-%d %T&#39;</span><span class="sb">`</span>
</span></span><span class="line"><span class="cl"><span class="nv">start</span><span class="o">=</span><span class="sb">`</span>date +%s<span class="sb">`</span>
</span></span><span class="line"><span class="cl"><span class="c1">##############</span>
</span></span><span class="line"><span class="cl"><span class="nv">myarray</span><span class="o">=(</span><span class="sb">`</span>cat ~/work/asthma/data/namelist<span class="sb">`</span><span class="o">)</span>
</span></span><span class="line"><span class="cl"><span class="nb">echo</span> <span class="nv">$SLURM_ARRAY_TASK_ID</span>
</span></span><span class="line"><span class="cl"><span class="c1">#this is your single file name</span>
</span></span><span class="line"><span class="cl"><span class="nv">sample</span><span class="o">=</span><span class="si">${</span><span class="nv">myarray</span><span class="p">[</span><span class="si">${</span><span class="nv">SLURM_ARRAY_TASK_ID</span><span class="si">}</span><span class="p">]</span><span class="si">}</span>
</span></span><span class="line"><span class="cl"><span class="nb">echo</span> handling: <span class="nv">$sample</span>
</span></span><span class="line"><span class="cl">humann2 --input data/alldata/<span class="si">${</span><span class="nv">sample</span><span class="si">}</span>_paired.fastq  <span class="se">\
</span></span></span><span class="line"><span class="cl"><span class="se"></span>  --output temp/humann2/ --threads <span class="m">24</span>
</span></span><span class="line"><span class="cl">  
</span></span><span class="line"><span class="cl"><span class="c1">## ÈìæÊé•ÈáçË¶ÅÊñá‰ª∂Ëá≥humann2ÁõÆÂΩï</span>
</span></span><span class="line"><span class="cl">ln temp/humann2/<span class="si">${</span><span class="nv">sample</span><span class="si">}</span>_paired_humann2_temp/<span class="si">${</span><span class="nv">sample</span><span class="si">}</span>_paired_metaphlan_bugs_list.tsv temp/humann2/
</span></span><span class="line"><span class="cl"><span class="c1">## Âà†Èô§‰∏¥Êó∂Êñá‰ª∂</span>
</span></span><span class="line"><span class="cl">rm -rf temp/humann2/<span class="si">${</span><span class="nv">sample</span><span class="si">}</span>_paired_humann2_temp
</span></span><span class="line"><span class="cl"><span class="c1">##############</span>
</span></span><span class="line"><span class="cl"><span class="nb">echo</span> end: <span class="sb">`</span>date +<span class="s1">&#39;%Y-%m-%d %T&#39;</span><span class="sb">`</span>
</span></span><span class="line"><span class="cl"><span class="nv">end</span><span class="o">=</span><span class="sb">`</span>date +%s<span class="sb">`</span>
</span></span><span class="line"><span class="cl"><span class="nb">echo</span> TIME:<span class="sb">`</span>expr <span class="nv">$end</span> - <span class="nv">$start</span><span class="sb">`</span>s
</span></span><span class="line"><span class="cl">
</span></span><span class="line"><span class="cl">
</span></span><span class="line"><span class="cl"><span class="c1">## ÂêàÂπ∂„ÄÅ‰øÆÊ≠£Ê†∑Êú¨Âêç„ÄÅÈ¢ÑËßà</span>
</span></span><span class="line"><span class="cl">merge_metaphlan_tables2.py <span class="se">\
</span></span></span><span class="line"><span class="cl"><span class="se"></span>  temp/humann2/*_metaphlan_bugs_list.tsv <span class="p">|</span> <span class="se">\
</span></span></span><span class="line"><span class="cl"><span class="se"></span>  sed <span class="s1">&#39;s/_metaphlan_bugs_list//g&#39;</span> <span class="se">\
</span></span></span><span class="line"><span class="cl"><span class="se"></span>  &gt; metaphlan2/taxonomy.tsv
</span></span></code></pre></td></tr></table>
</div>
</div><h2 id="contigs-based">contigs-based</h2>
<h3 id="ÁªÑË£Ömegahit">ÁªÑË£ÖÔºömegahit</h3>
<p>MegaHitÊòØ‰∏Ä‰∏™Áî®‰∫éÂØπÈ´òÈÄöÈáèÊµãÂ∫èÊï∞ÊçÆËøõË°åde novoÁªÑË£ÖÁöÑËΩØ‰ª∂„ÄÇÂÆÉ‰ΩøÁî®‰∫Ü‰∏ÄÁßçÂü∫‰∫éÁü≠ËØªÊØîÂØπÂíåÂõæÂΩ¢ÊûÑÂª∫ÁöÑÁÆóÊ≥ïÊù•ÁªÑË£ÖÂü∫Âõ†ÁªÑÔºåÂèØ‰ª•È´òÊïàÂú∞Â§ÑÁêÜÂ§ßËßÑÊ®°ÁöÑÊï∞ÊçÆÈõÜ„ÄÇ‰ª•‰∏ãÊòØMegaHitÁöÑ‰∏Ä‰∫õ‰ºòÁÇπÂíåÈÄÇÁî®ÊÉÖÂÜµÔºö</p>
<ol>
<li>
<p>ÈÄüÂ∫¶Âø´ÔºöMegaHitÁöÑÁÆóÊ≥ïÈùûÂ∏∏È´òÊïàÔºåÂèØ‰ª•Â§ÑÁêÜÂ§ßËßÑÊ®°ÁöÑÊï∞ÊçÆÈõÜÔºåÈÄöÂ∏∏ÊØîÂÖ∂‰ªñde novoÁªÑË£ÖÂ∑•ÂÖ∑Êõ¥Âø´„ÄÇ</p>
</li>
<li>
<p>È´òË¥®ÈáèÁöÑÁªÑË£ÖÔºöMegaHitÂú®ÁªÑË£ÖÁªìÊûúÁöÑËøûÈÄöÊÄßÂíåÂáÜÁ°ÆÊÄßÊñπÈù¢Ë°®Áé∞‰ºòÂºÇÔºåÂ∞§ÂÖ∂Âú®Â§ÑÁêÜÈ´òGCÂê´ÈáèÂü∫Âõ†ÁªÑÊó∂ÊïàÊûúÊòæËëó„ÄÇ</p>
</li>
<li>
<p>ÈÄÇÁî®‰∫é‰∏çÂêåÁ±ªÂûãÁöÑÊµãÂ∫èÊï∞ÊçÆÔºöMegaHitÊîØÊåÅÂ§öÁßç‰∏çÂêåÁ±ªÂûãÁöÑÊµãÂ∫èÊï∞ÊçÆÔºåÂåÖÊã¨ Illumina HiSeq/MiSeq„ÄÅIonTorrentÂíåPacBioÁ≠âÂπ≥Âè∞„ÄÇ</p>
</li>
<li>
<p>Êòì‰∫é‰ΩøÁî®ÔºöMegaHitÂÖ∑ÊúâÁÆÄÂçïÁöÑÂëΩ‰ª§Ë°åËØ≠Ê≥ïÔºåÊñπ‰æøÁî®Êà∑ËøõË°åÁªÑË£ÖÊìç‰ΩúÔºå‰∏îÂÖ∑Êúâ‰∏≠Êñ≠ÁÇπÔºåÈÅøÂÖçÂ§±Ë¥•ÂêéÂÖ®ÈÉ®ÈáçË∑ë„ÄÇ</p>
</li>
</ol>
<div class="highlight"><div class="chroma">
<table class="lntable"><tr><td class="lntd">
<pre tabindex="0" class="chroma"><code><span class="lnt"> 1
</span><span class="lnt"> 2
</span><span class="lnt"> 3
</span><span class="lnt"> 4
</span><span class="lnt"> 5
</span><span class="lnt"> 6
</span><span class="lnt"> 7
</span><span class="lnt"> 8
</span><span class="lnt"> 9
</span><span class="lnt">10
</span><span class="lnt">11
</span><span class="lnt">12
</span><span class="lnt">13
</span><span class="lnt">14
</span><span class="lnt">15
</span><span class="lnt">16
</span><span class="lnt">17
</span><span class="lnt">18
</span><span class="lnt">19
</span><span class="lnt">20
</span><span class="lnt">21
</span></code></pre></td>
<td class="lntd">
<pre tabindex="0" class="chroma"><code class="language-bash" data-lang="bash"><span class="line"><span class="cl"><span class="cp">#!/bin/bash
</span></span></span><span class="line"><span class="cl"><span class="cp"></span><span class="c1">#SBATCH --job-name=asthma_megahit</span>
</span></span><span class="line"><span class="cl"><span class="c1">#SBATCH --output=/share/home/jianglab/pengchen/work/asthma/megahit/log/%x_%a.out</span>
</span></span><span class="line"><span class="cl"><span class="c1">#SBATCH --error=/share/home/jianglab/pengchen/work/asthma/megahit/log/%x_%a.err</span>
</span></span><span class="line"><span class="cl"><span class="c1">#SBATCH --array=1-33</span>
</span></span><span class="line"><span class="cl"><span class="c1">#SBATCH --partition=cpu</span>
</span></span><span class="line"><span class="cl"><span class="c1">#SBATCH --cpus-per-task=32</span>
</span></span><span class="line"><span class="cl">
</span></span><span class="line"><span class="cl"><span class="nb">echo</span> start: <span class="sb">`</span>date +<span class="s1">&#39;%Y-%m-%d %T&#39;</span><span class="sb">`</span>
</span></span><span class="line"><span class="cl"><span class="nv">start</span><span class="o">=</span><span class="sb">`</span>date +%s<span class="sb">`</span>
</span></span><span class="line"><span class="cl"><span class="nb">echo</span> <span class="s2">&#34;SLURM_ARRAY_TASK_ID: &#34;</span> <span class="nv">$SLURM_ARRAY_TASK_ID</span>
</span></span><span class="line"><span class="cl"><span class="nv">sample</span><span class="o">=</span><span class="k">$(</span>head -n <span class="nv">$SLURM_ARRAY_TASK_ID</span> ~/work/asthma/data/namelist <span class="p">|</span> tail -1<span class="k">)</span>
</span></span><span class="line"><span class="cl"><span class="c1">#sample=$(head -n 1 namelist | tail -1)</span>
</span></span><span class="line"><span class="cl"><span class="nb">echo</span> handling: <span class="nv">$sample</span>	
</span></span><span class="line"><span class="cl"><span class="c1">####################</span>
</span></span><span class="line"><span class="cl">megahit -t <span class="m">32</span> -1 ~/work/asthma/data/<span class="nv">$sample</span>/<span class="nv">$sample</span><span class="s1">&#39;_f1.fastq&#39;</span> <span class="se">\
</span></span></span><span class="line"><span class="cl"><span class="se"></span>-2 ~/work/asthma/data/<span class="nv">$sample</span>/<span class="nv">$sample</span><span class="s1">&#39;_r2.fastq&#39;</span> -o ~/work/asthma/megahit/<span class="nv">$sample</span> --out-prefix <span class="nv">$sample</span>
</span></span><span class="line"><span class="cl"><span class="c1">####################</span>
</span></span><span class="line"><span class="cl"><span class="nb">echo</span> end: <span class="sb">`</span>date +<span class="s1">&#39;%Y-%m-%d %T&#39;</span><span class="sb">`</span>
</span></span><span class="line"><span class="cl"><span class="nv">end</span><span class="o">=</span><span class="sb">`</span>date +%s<span class="sb">`</span>
</span></span><span class="line"><span class="cl"><span class="nb">echo</span> TIME:<span class="sb">`</span>expr <span class="nv">$end</span> - <span class="nv">$start</span><span class="sb">`</span>s
</span></span></code></pre></td></tr></table>
</div>
</div><h3 id="ÁªÑË£ÖËØÑ‰º∞quast">ÁªÑË£ÖËØÑ‰º∞ÔºöQUAST</h3>
<p><strong>QUAST</strong>‰ª£Ë°®Ë¥®ÈáèËØÑ‰º∞Â∑•ÂÖ∑„ÄÇ QUASTÂèØ‰ª•‰ΩøÁî®ÂèÇËÄÉÂü∫Âõ†ÁªÑ‰ª•Âèä‰∏ç‰ΩøÁî®ÂèÇËÄÉÂü∫Âõ†ÁªÑÊù•ËØÑ‰º∞Ë£ÖÈÖç„ÄÇ QUASTÁîüÊàêËØ¶ÁªÜÁöÑÊä•ÂëäÔºåË°®Ê†ºÂíåÂõæËß£Ôºå‰ª•ÊòæÁ§∫Ë£ÖÈÖçÁöÑ‰∏çÂêåÊñπÈù¢„ÄÇ</p>
<h3 id="Âü∫Âõ†È¢ÑÊµãprodigal">Âü∫Âõ†È¢ÑÊµãÔºöProdigal</h3>
<p>ËæìÂÖ•Êñá‰ª∂ÔºöÊãºË£ÖÂ•ΩÁöÑÂ∫èÂàóÊñá‰ª∂ megahit/final.contigs.fa</p>
<p>ËæìÂá∫Êñá‰ª∂ÔºöprodigalÈ¢ÑÊµãÁöÑÂü∫Âõ†Â∫èÂàó prodigal/gene.fa</p>
<p>prodigal‰∏çÊîØÊåÅÂ§öÁ∫øÁ®ãËøêË°åÔºåÊâÄ‰ª•Êàë‰ª¨ÂèØ‰ª•Ëá™Ë°åÂàÜÂâ≤Â∫èÂàóÊñá‰ª∂Ë∞ÉÁî®Â§ö‰∏™prodigalÁ®ãÂ∫èÂàÜÂà´Ë∑ëÂÆûÁé∞‰º™Â§öÁ∫øÁ®ã„ÄÇ</p>
<div class="highlight"><div class="chroma">
<table class="lntable"><tr><td class="lntd">
<pre tabindex="0" class="chroma"><code><span class="lnt"> 1
</span><span class="lnt"> 2
</span><span class="lnt"> 3
</span><span class="lnt"> 4
</span><span class="lnt"> 5
</span><span class="lnt"> 6
</span><span class="lnt"> 7
</span><span class="lnt"> 8
</span><span class="lnt"> 9
</span><span class="lnt">10
</span><span class="lnt">11
</span><span class="lnt">12
</span><span class="lnt">13
</span><span class="lnt">14
</span><span class="lnt">15
</span><span class="lnt">16
</span><span class="lnt">17
</span><span class="lnt">18
</span><span class="lnt">19
</span><span class="lnt">20
</span><span class="lnt">21
</span><span class="lnt">22
</span><span class="lnt">23
</span><span class="lnt">24
</span><span class="lnt">25
</span><span class="lnt">26
</span><span class="lnt">27
</span><span class="lnt">28
</span><span class="lnt">29
</span><span class="lnt">30
</span><span class="lnt">31
</span><span class="lnt">32
</span><span class="lnt">33
</span><span class="lnt">34
</span><span class="lnt">35
</span><span class="lnt">36
</span><span class="lnt">37
</span></code></pre></td>
<td class="lntd">
<pre tabindex="0" class="chroma"><code class="language-bash" data-lang="bash"><span class="line"><span class="cl"><span class="cp">#!/bin/bash
</span></span></span><span class="line"><span class="cl"><span class="cp"></span><span class="c1">#SBATCH --job-name=prodigal</span>
</span></span><span class="line"><span class="cl"><span class="c1">#SBATCH --output=/share/home/jianglab/pengchen/work/asthma/prodigal/log/%x_%a.out</span>
</span></span><span class="line"><span class="cl"><span class="c1">#SBATCH --error=/share/home/jianglab/pengchen/work/asthma/prodigal/log/%x_%a.err</span>
</span></span><span class="line"><span class="cl"><span class="c1">#SBATCH --array=1-33</span>
</span></span><span class="line"><span class="cl"><span class="c1">#SBATCH --partition=cpu</span>
</span></span><span class="line"><span class="cl"><span class="c1">#SBATCH --cpus-per-task=1</span>
</span></span><span class="line"><span class="cl">
</span></span><span class="line"><span class="cl"><span class="nb">echo</span> start: <span class="sb">`</span>date +<span class="s1">&#39;%Y-%m-%d %T&#39;</span><span class="sb">`</span>
</span></span><span class="line"><span class="cl"><span class="nv">start</span><span class="o">=</span><span class="sb">`</span>date +%s<span class="sb">`</span>
</span></span><span class="line"><span class="cl"><span class="nb">echo</span> <span class="s2">&#34;SLURM_ARRAY_TASK_ID: &#34;</span> <span class="nv">$SLURM_ARRAY_TASK_ID</span>
</span></span><span class="line"><span class="cl"><span class="nv">sample</span><span class="o">=</span><span class="k">$(</span>head -n <span class="nv">$SLURM_ARRAY_TASK_ID</span> ~/work/asthma/data/namelist <span class="p">|</span> tail -1<span class="k">)</span>
</span></span><span class="line"><span class="cl"><span class="c1">#sample=$(head -n 1 namelist | tail -1)</span>
</span></span><span class="line"><span class="cl"><span class="nb">echo</span> handling: <span class="nv">$sample</span>
</span></span><span class="line"><span class="cl"><span class="c1">####################</span>
</span></span><span class="line"><span class="cl">prodigal -i ~/work/asthma/megahit/contigs/<span class="nv">$sample</span>.fa <span class="se">\
</span></span></span><span class="line"><span class="cl"><span class="se"></span>    -d ~/work/asthma/prodigal/<span class="nv">$sample</span>.gene.fa <span class="se">\
</span></span></span><span class="line"><span class="cl"><span class="se"></span>    -o ~/work/asthma/prodigal/<span class="nv">$sample</span>.gene.gff <span class="se">\
</span></span></span><span class="line"><span class="cl"><span class="se"></span>    -p meta -f gff 
</span></span><span class="line"><span class="cl">    
</span></span><span class="line"><span class="cl">grep <span class="s1">&#39;partial=00&#39;</span> ~/work/asthma/prodigal/<span class="nv">$sample</span>.gene.fa <span class="p">|</span> cut -f1 -d <span class="s1">&#39; &#39;</span><span class="p">|</span> sed <span class="s1">&#39;s/&gt;//&#39;</span> &gt; ~/work/asthma/prodigal/<span class="nv">$sample</span>.fullid
</span></span><span class="line"><span class="cl">seqkit grep -f ~/work/asthma/prodigal/<span class="nv">$sample</span>.fullid ~/work/asthma/prodigal/<span class="nv">$sample</span>.gene.fa &gt; ~/work/asthma/prodigal/fullgene/<span class="nv">$sample</span>.gene.fa
</span></span><span class="line"><span class="cl">
</span></span><span class="line"><span class="cl"><span class="c1">####################</span>
</span></span><span class="line"><span class="cl"><span class="nb">echo</span> end: <span class="sb">`</span>date +<span class="s1">&#39;%Y-%m-%d %T&#39;</span><span class="sb">`</span>
</span></span><span class="line"><span class="cl"><span class="nv">end</span><span class="o">=</span><span class="sb">`</span>date +%s<span class="sb">`</span>
</span></span><span class="line"><span class="cl"><span class="nb">echo</span> TIME:<span class="sb">`</span>expr <span class="nv">$end</span> - <span class="nv">$start</span><span class="sb">`</span><span class="nv">s</span>
</span></span><span class="line"><span class="cl">
</span></span><span class="line"><span class="cl"><span class="o">============================================================================</span>
</span></span><span class="line"><span class="cl"><span class="c1">## ÁªüËÆ°Âü∫Âõ†Êï∞Èáè</span>
</span></span><span class="line"><span class="cl">grep -c <span class="s1">&#39;&gt;&#39;</span> temp/prodigal/gene.fa 
</span></span><span class="line"><span class="cl"><span class="c1">## ÁªüËÆ°ÂÆåÊï¥Âü∫Âõ†Êï∞ÈáèÔºåÊï∞ÊçÆÈáèÂ§ßÂèØÂè™Áî®ÂÆåÊï¥Âü∫Âõ†ÈÉ®ÂàÜ</span>
</span></span><span class="line"><span class="cl">grep -c <span class="s1">&#39;partial=00&#39;</span> temp/prodigal/gene.fa 
</span></span><span class="line"><span class="cl"><span class="c1">## ÊèêÂèñÂÆåÊï¥Âü∫Âõ†(ÂÆåÊï¥ÁâáÊÆµËé∑ÂæóÁöÑÂü∫Âõ†ÂÖ®‰∏∫ÂÆåÊï¥ÔºåÂ¶ÇÊàêÁéØÁöÑÁªÜËèåÂü∫Âõ†ÁªÑ)</span>
</span></span><span class="line"><span class="cl">grep <span class="s1">&#39;partial=00&#39;</span> temp/prodigal/gene.fa <span class="p">|</span> cut -f1 -d <span class="s1">&#39; &#39;</span><span class="p">|</span> sed <span class="s1">&#39;s/&gt;//&#39;</span> &gt; temp/prodigal/full_length.id
</span></span><span class="line"><span class="cl">seqkit grep -f temp/prodigal/full_length.id temp/prodigal/gene.fa &gt; temp/prodigal/full_length.fa
</span></span><span class="line"><span class="cl">seqkit stat temp/prodigal/full_length.fa
</span></span></code></pre></td></tr></table>
</div>
</div><h3 id="ÂéªÂÜó‰Ωô">ÂéªÂÜó‰Ωô</h3>
<p>‰∏äÈù¢‰∫ßÁîü‰∫Ün‰∏™Ê†∑Êú¨ÁöÑÂü∫Âõ†È¢ÑÊµãÁªìÊûúÊñá‰ª∂Ôºågene.faÊñá‰ª∂Ë¶ÅÊÉ≥ÂäûÊ≥ïÊï¥Âêà‰∏∫‰∏Ä‰∏™Êñá‰ª∂ÂÜçÂéªÂéªÂÜó‰Ωô„ÄÇ</p>
<div class="highlight"><div class="chroma">
<table class="lntable"><tr><td class="lntd">
<pre tabindex="0" class="chroma"><code><span class="lnt"> 1
</span><span class="lnt"> 2
</span><span class="lnt"> 3
</span><span class="lnt"> 4
</span><span class="lnt"> 5
</span><span class="lnt"> 6
</span><span class="lnt"> 7
</span><span class="lnt"> 8
</span><span class="lnt"> 9
</span><span class="lnt">10
</span></code></pre></td>
<td class="lntd">
<pre tabindex="0" class="chroma"><code class="language-bash" data-lang="bash"><span class="line"><span class="cl"><span class="cp">#!/bin/bash
</span></span></span><span class="line"><span class="cl"><span class="cp"></span><span class="c1">#‰øÆÊîπÊØèÊù°Â∫èÂàóÁöÑÂêçÁß∞ÔºåÂä†‰∏äÊ†∑Êú¨Âêç</span>
</span></span><span class="line"><span class="cl"><span class="k">for</span> i in <span class="sb">`</span>cat ~/work/asthma/data/namelist<span class="sb">`</span>
</span></span><span class="line"><span class="cl"><span class="k">do</span>
</span></span><span class="line"><span class="cl"><span class="nb">echo</span> <span class="nv">$i</span>
</span></span><span class="line"><span class="cl">sed -i <span class="s2">&#34;/&gt;/s/&gt;/&gt;</span><span class="si">${</span><span class="nv">i</span><span class="si">}</span><span class="s2">_/&#34;</span> <span class="nv">$i</span>.gene.fa 
</span></span><span class="line"><span class="cl"><span class="k">done</span>
</span></span><span class="line"><span class="cl"><span class="nb">echo</span> <span class="s1">&#39;start merge&#39;</span>
</span></span><span class="line"><span class="cl">cat *.gene.fa&gt;all.fullgene.fa
</span></span><span class="line"><span class="cl"><span class="nb">echo</span> <span class="s1">&#39;done&#39;</span>
</span></span></code></pre></td></tr></table>
</div>
</div><h4 id="cd-hit">Cd-hit</h4>
<div class="highlight"><div class="chroma">
<table class="lntable"><tr><td class="lntd">
<pre tabindex="0" class="chroma"><code><span class="lnt"> 1
</span><span class="lnt"> 2
</span><span class="lnt"> 3
</span><span class="lnt"> 4
</span><span class="lnt"> 5
</span><span class="lnt"> 6
</span><span class="lnt"> 7
</span><span class="lnt"> 8
</span><span class="lnt"> 9
</span><span class="lnt">10
</span><span class="lnt">11
</span><span class="lnt">12
</span><span class="lnt">13
</span><span class="lnt">14
</span><span class="lnt">15
</span><span class="lnt">16
</span><span class="lnt">17
</span><span class="lnt">18
</span><span class="lnt">19
</span><span class="lnt">20
</span><span class="lnt">21
</span><span class="lnt">22
</span><span class="lnt">23
</span><span class="lnt">24
</span><span class="lnt">25
</span><span class="lnt">26
</span><span class="lnt">27
</span><span class="lnt">28
</span><span class="lnt">29
</span><span class="lnt">30
</span><span class="lnt">31
</span><span class="lnt">32
</span><span class="lnt">33
</span><span class="lnt">34
</span></code></pre></td>
<td class="lntd">
<pre tabindex="0" class="chroma"><code class="language-bash" data-lang="bash"><span class="line"><span class="cl"><span class="cp">#!/bin/bash
</span></span></span><span class="line"><span class="cl"><span class="cp"></span><span class="c1">#SBATCH --job-name=cdhit</span>
</span></span><span class="line"><span class="cl"><span class="c1">#SBATCH --output=/share/home/jianglab/pengchen/work/asthma/%x_%a.out</span>
</span></span><span class="line"><span class="cl"><span class="c1">#SBATCH --error=/share/home/jianglab/pengchen/work/asthma/%x_%a.err</span>
</span></span><span class="line"><span class="cl"><span class="c1">#SBATCH --cpus-per-task=32</span>
</span></span><span class="line"><span class="cl"><span class="c1">#SBATCH --partition=short</span>
</span></span><span class="line"><span class="cl">
</span></span><span class="line"><span class="cl"><span class="nb">echo</span> start: <span class="sb">`</span>date +<span class="s1">&#39;%Y-%m-%d %T&#39;</span><span class="sb">`</span>
</span></span><span class="line"><span class="cl"><span class="nv">start</span><span class="o">=</span><span class="sb">`</span>date +%s<span class="sb">`</span>
</span></span><span class="line"><span class="cl"><span class="c1">##############</span>
</span></span><span class="line"><span class="cl"><span class="c1">## ËæìÂÖ•Êñá‰ª∂ÔºöprodigalÈ¢ÑÊµãÁöÑÂü∫Âõ†Â∫èÂàó all.fullgene.fa</span>
</span></span><span class="line"><span class="cl"><span class="c1">## ËæìÂá∫Êñá‰ª∂ÔºöÂéªÂÜó‰ΩôÂêéÁöÑÂü∫Âõ†ÂíåËõãÁôΩÂ∫èÂàóÔºöNR/nucleotide.fa;NR/protein.fa</span>
</span></span><span class="line"><span class="cl">mkdir NR
</span></span><span class="line"><span class="cl"><span class="c1">## aSË¶ÜÁõñÂ∫¶ÔºåcÁõ∏‰ººÂ∫¶ÔºåGÂ±ÄÈÉ®ÊØîÂØπÔºågÊúÄ‰ºòËß£ÔºåTÂ§öÁ∫øÁ®ãÔºåMÂÜÖÂ≠ò0‰∏çÈôêÂà∂</span>
</span></span><span class="line"><span class="cl"><span class="c1">## 2‰∏áÂü∫Âõ†2mÔºå2ÂçÉ‰∏áÈúÄË¶Å2000hÔºåÂ§öÁ∫øÁ®ãÂèØÂä†ÈÄü	</span>
</span></span><span class="line"><span class="cl">cd-hit-est -i prodigal/fullgene/all.fullgene.fa <span class="se">\
</span></span></span><span class="line"><span class="cl"><span class="se"></span>    -o NR/nucleotide.fa <span class="se">\
</span></span></span><span class="line"><span class="cl"><span class="se"></span>    -aS 0.9 -c 0.9 -G <span class="m">0</span> -g <span class="m">0</span> -T <span class="m">0</span> -M <span class="m">0</span>
</span></span><span class="line"><span class="cl"><span class="c1">## ÁªüËÆ°ÈùûÂÜó‰ΩôÂü∫Âõ†Êï∞ÈáèÔºåÂçïÊ¨°ÊãºÊé•ÁªìÊûúÊï∞Èáè‰∏ãÈôç‰∏çÂ§ßÔºåÂ§öÊâπÊãºÊé•ÂÜó‰ΩôÂ∫¶È´ò</span>
</span></span><span class="line"><span class="cl">
</span></span><span class="line"><span class="cl"><span class="nb">echo</span> <span class="s1">&#39;after remove, the number of genes: &#39;</span>
</span></span><span class="line"><span class="cl">grep -c <span class="s1">&#39;&gt;&#39;</span> NR/nucleotide.fa
</span></span><span class="line"><span class="cl"><span class="c1">## ÁøªËØëÊ†∏ÈÖ∏‰∏∫ÂØπÂ∫îËõãÁôΩÂ∫èÂàóÔºåemboss</span>
</span></span><span class="line"><span class="cl"><span class="c1">## emboss transeqÂ∑•ÂÖ∑Ôºå93.9 MB</span>
</span></span><span class="line"><span class="cl">conda install emboss -y
</span></span><span class="line"><span class="cl">
</span></span><span class="line"><span class="cl">transeq -sequence NR/nucleotide.fa <span class="se">\
</span></span></span><span class="line"><span class="cl"><span class="se"></span>  -outseq NR/protein.fa -trim Y 
</span></span><span class="line"><span class="cl"><span class="c1">## Â∫èÂàóÂêçËá™Âä®Ê∑ªÂä†‰∫Ü_1Ôºå‰∏∫‰∏éÊ†∏ÈÖ∏ÂØπÂ∫îË¶ÅÂéªÈô§</span>
</span></span><span class="line"><span class="cl">sed -i <span class="s1">&#39;s/_1 / /&#39;</span> NR/protein.fa
</span></span><span class="line"><span class="cl"><span class="c1">##############</span>
</span></span><span class="line"><span class="cl"><span class="nb">echo</span> end: <span class="sb">`</span>date +<span class="s1">&#39;%Y-%m-%d %T&#39;</span><span class="sb">`</span>
</span></span><span class="line"><span class="cl"><span class="nv">end</span><span class="o">=</span><span class="sb">`</span>date +%s<span class="sb">`</span>
</span></span><span class="line"><span class="cl"><span class="nb">echo</span> TIME:<span class="sb">`</span>expr <span class="nv">$end</span> - <span class="nv">$start</span><span class="sb">`</span>s
</span></span></code></pre></td></tr></table>
</div>
</div><h4 id="mmseq2">mmseq2</h4>
<p>Ëøô‰∏™Ë¶ÅÊØîcd-hitÂø´ÈùûÂ∏∏Â§ö</p>
<div class="highlight"><div class="chroma">
<table class="lntable"><tr><td class="lntd">
<pre tabindex="0" class="chroma"><code><span class="lnt"> 1
</span><span class="lnt"> 2
</span><span class="lnt"> 3
</span><span class="lnt"> 4
</span><span class="lnt"> 5
</span><span class="lnt"> 6
</span><span class="lnt"> 7
</span><span class="lnt"> 8
</span><span class="lnt"> 9
</span><span class="lnt">10
</span><span class="lnt">11
</span><span class="lnt">12
</span><span class="lnt">13
</span><span class="lnt">14
</span><span class="lnt">15
</span><span class="lnt">16
</span><span class="lnt">17
</span><span class="lnt">18
</span><span class="lnt">19
</span><span class="lnt">20
</span><span class="lnt">21
</span><span class="lnt">22
</span><span class="lnt">23
</span><span class="lnt">24
</span><span class="lnt">25
</span><span class="lnt">26
</span><span class="lnt">27
</span><span class="lnt">28
</span><span class="lnt">29
</span><span class="lnt">30
</span></code></pre></td>
<td class="lntd">
<pre tabindex="0" class="chroma"><code class="language-bash" data-lang="bash"><span class="line"><span class="cl"><span class="c1">#mmseqs2</span>
</span></span><span class="line"><span class="cl"><span class="c1">##min-seq-id:identityÔºåcË¶ÜÁõñÂ∫¶Ôºårescore-mode3: global alignment</span>
</span></span><span class="line"><span class="cl">
</span></span><span class="line"><span class="cl"><span class="c1">#ËÅöÁ±ª</span>
</span></span><span class="line"><span class="cl"><span class="nv">input_fa</span><span class="o">=</span>tmp_com.gene
</span></span><span class="line"><span class="cl">mmseqs easy-linclust <span class="nv">$input_fa</span> lin_res tmp <span class="se">\
</span></span></span><span class="line"><span class="cl"><span class="se"></span>    --min-seq-id 0.9 -c 0.9 --cov-mode <span class="m">1</span>  --threads <span class="m">8</span>
</span></span><span class="line"><span class="cl"><span class="c1">#ÊàñËÄÖ</span>
</span></span><span class="line"><span class="cl">mmseqs easy-cluster <span class="nv">$input_fa</span> lin_res tmp <span class="se">\
</span></span></span><span class="line"><span class="cl"><span class="se"></span>    --min-seq-id 0.9 -c 0.9 --cov-mode <span class="m">1</span>  --threads <span class="m">8</span>
</span></span><span class="line"><span class="cl">
</span></span><span class="line"><span class="cl"><span class="c1">##‰ª•‰∏ãÊòØ‰∏∫‰∫ÜÁêÜËß£Ê≠•È™§ÂÅöÁöÑÔºå‰ΩøÁî®‰∏äÈù¢üëÜÁöÑeasyÊ®°ÂºèÂç≥ÂèØ</span>
</span></span><span class="line"><span class="cl"><span class="c1">#Âª∫Â∫ì</span>
</span></span><span class="line"><span class="cl"><span class="nv">input_fa</span><span class="o">=</span>../prodigal/C1.gene.fa
</span></span><span class="line"><span class="cl"><span class="nv">DB</span><span class="o">=</span>C1.geneDB
</span></span><span class="line"><span class="cl"><span class="nv">DB_clu</span><span class="o">=</span>mmseq_out
</span></span><span class="line"><span class="cl">mmseqs createdb <span class="nv">$input_fa</span> <span class="nv">$DB</span>
</span></span><span class="line"><span class="cl"><span class="c1">#ËÅöÁ±ª</span>
</span></span><span class="line"><span class="cl">mmseqs cluster <span class="nv">$DB</span> <span class="nv">$DB_clu</span> tmp <span class="se">\
</span></span></span><span class="line"><span class="cl"><span class="se"></span>    --min-seq-id 0.9 -c 0.9 --cov-mode <span class="m">1</span>  --threads <span class="m">8</span>  --rescore-mode <span class="m">3</span>
</span></span><span class="line"><span class="cl"><span class="c1">#ËæìÂá∫ËΩ¨Êç¢</span>
</span></span><span class="line"><span class="cl"><span class="c1">#Á¨¨‰∏ÄÂàóÊòØ‰ª£Ë°®Â∫èÂàóidÔºåÁ¨¨‰∫åÂàóÊòØÊàêÂëòÂ∫èÂàóid</span>
</span></span><span class="line"><span class="cl"><span class="c1">#mmseqs createtsv $DB $DB mmseq_out mmseq_out.tsv</span>
</span></span><span class="line"><span class="cl">
</span></span><span class="line"><span class="cl"><span class="c1">#mmseqs createseqfiledb $DB $DB_clu ${DB_clu}_seq</span>
</span></span><span class="line"><span class="cl"><span class="c1">#mmseqs result2flat $DB $DB ${DB_clu}_seq ${DB_clu}_seq.fasta</span>
</span></span><span class="line"><span class="cl">
</span></span><span class="line"><span class="cl"><span class="c1">#Ëé∑Âèñrepresent sequence</span>
</span></span><span class="line"><span class="cl">mmseqs createsubdb <span class="nv">$DB_clu</span> <span class="nv">$DB</span> <span class="si">${</span><span class="nv">DB_clu</span><span class="si">}</span>_rep
</span></span><span class="line"><span class="cl">mmseqs convert2fasta <span class="si">${</span><span class="nv">DB_clu</span><span class="si">}</span>_rep <span class="si">${</span><span class="nv">DB_clu</span><span class="si">}</span>_rep.fasta   
</span></span></code></pre></td></tr></table>
</div>
</div><h3 id="Âü∫Âõ†ÂÆöÈáèsalmon">Âü∫Âõ†ÂÆöÈáèÔºösalmon</h3>
<ol>
<li>Âª∫Á´ãÁ¥¢Âºï</li>
</ol>
<div class="highlight"><div class="chroma">
<table class="lntable"><tr><td class="lntd">
<pre tabindex="0" class="chroma"><code><span class="lnt"> 1
</span><span class="lnt"> 2
</span><span class="lnt"> 3
</span><span class="lnt"> 4
</span><span class="lnt"> 5
</span><span class="lnt"> 6
</span><span class="lnt"> 7
</span><span class="lnt"> 8
</span><span class="lnt"> 9
</span><span class="lnt">10
</span><span class="lnt">11
</span><span class="lnt">12
</span><span class="lnt">13
</span><span class="lnt">14
</span><span class="lnt">15
</span><span class="lnt">16
</span><span class="lnt">17
</span><span class="lnt">18
</span><span class="lnt">19
</span><span class="lnt">20
</span><span class="lnt">21
</span><span class="lnt">22
</span></code></pre></td>
<td class="lntd">
<pre tabindex="0" class="chroma"><code class="language-bash" data-lang="bash"><span class="line"><span class="cl"><span class="cp">#!/bin/bash
</span></span></span><span class="line"><span class="cl"><span class="cp"></span><span class="c1">#SBATCH --job-name=salmon-index</span>
</span></span><span class="line"><span class="cl"><span class="c1">#SBATCH --output=/share/home/jianglab/pengchen/work/asthma/%x_%a.out</span>
</span></span><span class="line"><span class="cl"><span class="c1">#SBATCH --error=/share/home/jianglab/pengchen/work/asthma/%x_%a.err</span>
</span></span><span class="line"><span class="cl"><span class="c1">#SBATCH --cpus-per-task=32</span>
</span></span><span class="line"><span class="cl">
</span></span><span class="line"><span class="cl"><span class="c1">#SBATCH --partition=short</span>
</span></span><span class="line"><span class="cl">
</span></span><span class="line"><span class="cl"><span class="nb">echo</span> start: <span class="sb">`</span>date +<span class="s1">&#39;%Y-%m-%d %T&#39;</span><span class="sb">`</span>
</span></span><span class="line"><span class="cl"><span class="nv">start</span><span class="o">=</span><span class="sb">`</span>date +%s<span class="sb">`</span>
</span></span><span class="line"><span class="cl"><span class="c1">##############</span>
</span></span><span class="line"><span class="cl">mkdir -p temp/salmon
</span></span><span class="line"><span class="cl">
</span></span><span class="line"><span class="cl"><span class="c1">## Âª∫Á¥¢Âºï, -tÂ∫èÂàó, -i Á¥¢ÂºïÔºå10s</span>
</span></span><span class="line"><span class="cl">salmon index <span class="se">\
</span></span></span><span class="line"><span class="cl"><span class="se"></span>  -t NR/nucleotide.fa <span class="se">\
</span></span></span><span class="line"><span class="cl"><span class="se"></span>  -p <span class="m">32</span> <span class="se">\
</span></span></span><span class="line"><span class="cl"><span class="se"></span>  -i temp/salmon/index 
</span></span><span class="line"><span class="cl"><span class="c1">##############</span>
</span></span><span class="line"><span class="cl"><span class="nb">echo</span> end: <span class="sb">`</span>date +<span class="s1">&#39;%Y-%m-%d %T&#39;</span><span class="sb">`</span>
</span></span><span class="line"><span class="cl"><span class="nv">end</span><span class="o">=</span><span class="sb">`</span>date +%s<span class="sb">`</span>
</span></span><span class="line"><span class="cl"><span class="nb">echo</span> TIME:<span class="sb">`</span>expr <span class="nv">$end</span> - <span class="nv">$start</span><span class="sb">`</span>s
</span></span></code></pre></td></tr></table>
</div>
</div><ol start="2">
<li>ÂØπÊØè‰∏™Ê†∑Êú¨ÂÆöÈáè</li>
</ol>
<div class="highlight"><div class="chroma">
<table class="lntable"><tr><td class="lntd">
<pre tabindex="0" class="chroma"><code><span class="lnt"> 1
</span><span class="lnt"> 2
</span><span class="lnt"> 3
</span><span class="lnt"> 4
</span><span class="lnt"> 5
</span><span class="lnt"> 6
</span><span class="lnt"> 7
</span><span class="lnt"> 8
</span><span class="lnt"> 9
</span><span class="lnt">10
</span><span class="lnt">11
</span><span class="lnt">12
</span><span class="lnt">13
</span><span class="lnt">14
</span><span class="lnt">15
</span><span class="lnt">16
</span><span class="lnt">17
</span><span class="lnt">18
</span><span class="lnt">19
</span><span class="lnt">20
</span><span class="lnt">21
</span><span class="lnt">22
</span><span class="lnt">23
</span><span class="lnt">24
</span><span class="lnt">25
</span><span class="lnt">26
</span><span class="lnt">27
</span></code></pre></td>
<td class="lntd">
<pre tabindex="0" class="chroma"><code class="language-bash" data-lang="bash"><span class="line"><span class="cl"><span class="cp">#!/bin/bash
</span></span></span><span class="line"><span class="cl"><span class="cp"></span><span class="c1">#SBATCH --job-name=salmon</span>
</span></span><span class="line"><span class="cl"><span class="c1">#SBATCH --output=/share/home/jianglab/pengchen/work/asthma/salmon/log/%x_%a.out</span>
</span></span><span class="line"><span class="cl"><span class="c1">#SBATCH --error=/share/home/jianglab/pengchen/work/asthma/salmon/log/%x_%a.err</span>
</span></span><span class="line"><span class="cl"><span class="c1">#SBATCH --array=1-33</span>
</span></span><span class="line"><span class="cl"><span class="c1">#SBATCH --partition=cpu</span>
</span></span><span class="line"><span class="cl"><span class="c1">#SBATCH --cpus-per-task=32</span>
</span></span><span class="line"><span class="cl">
</span></span><span class="line"><span class="cl"><span class="nb">echo</span> start: <span class="sb">`</span>date +<span class="s1">&#39;%Y-%m-%d %T&#39;</span><span class="sb">`</span>
</span></span><span class="line"><span class="cl"><span class="nv">start</span><span class="o">=</span><span class="sb">`</span>date +%s<span class="sb">`</span>
</span></span><span class="line"><span class="cl"><span class="nb">echo</span> <span class="s2">&#34;SLURM_ARRAY_TASK_ID: &#34;</span> <span class="nv">$SLURM_ARRAY_TASK_ID</span>
</span></span><span class="line"><span class="cl"><span class="nv">sample</span><span class="o">=</span><span class="k">$(</span>head -n <span class="nv">$SLURM_ARRAY_TASK_ID</span> ~/work/asthma/data/namelist <span class="p">|</span> tail -1<span class="k">)</span>
</span></span><span class="line"><span class="cl"><span class="c1">#sample=$(head -n 1 namelist | tail -1)</span>
</span></span><span class="line"><span class="cl"><span class="nb">echo</span> handling: <span class="nv">$sample</span>
</span></span><span class="line"><span class="cl"><span class="c1">####################</span>
</span></span><span class="line"><span class="cl"><span class="c1">## ËæìÂÖ•Êñá‰ª∂ÔºöÂéªÂÜó‰ΩôÂêéÁöÑÂü∫Âõ†ÂíåËõãÁôΩÂ∫èÂàóÔºöNR/nucleotide.fa</span>
</span></span><span class="line"><span class="cl"><span class="c1">## ËæìÂá∫Êñá‰ª∂ÔºöSalmonÂÆöÈáèÂêéÁöÑÁªìÊûúÔºösalmon/gene.count;salmon/gene.TPM</span>
</span></span><span class="line"><span class="cl"><span class="c1">## ÂÆöÈáèÔºålÊñáÂ∫ìÁ±ªÂûãËá™Âä®ÈÄâÊã©ÔºåpÁ∫øÁ®ãÔºå--metaÂÆèÂü∫Âõ†ÁªÑÊ®°Âºè</span>
</span></span><span class="line"><span class="cl">salmon quant <span class="se">\
</span></span></span><span class="line"><span class="cl"><span class="se"></span>    -i temp/salmon/index -l A -p <span class="m">32</span> --meta <span class="se">\
</span></span></span><span class="line"><span class="cl"><span class="se"></span>    -1 data/alldata/<span class="si">${</span><span class="nv">sample</span><span class="si">}</span>_f1.fastq <span class="se">\
</span></span></span><span class="line"><span class="cl"><span class="se"></span>    -2 data/alldata/<span class="si">${</span><span class="nv">sample</span><span class="si">}</span>_r2.fastq <span class="se">\
</span></span></span><span class="line"><span class="cl"><span class="se"></span>    -o temp/salmon/<span class="si">${</span><span class="nv">sample</span><span class="si">}</span>.quant
</span></span><span class="line"><span class="cl"><span class="c1">####################</span>
</span></span><span class="line"><span class="cl"><span class="nb">echo</span> end: <span class="sb">`</span>date +<span class="s1">&#39;%Y-%m-%d %T&#39;</span><span class="sb">`</span>
</span></span><span class="line"><span class="cl"><span class="nv">end</span><span class="o">=</span><span class="sb">`</span>date +%s<span class="sb">`</span>
</span></span><span class="line"><span class="cl"><span class="nb">echo</span> TIME:<span class="sb">`</span>expr <span class="nv">$end</span> - <span class="nv">$start</span><span class="sb">`</span>s
</span></span></code></pre></td></tr></table>
</div>
</div><ol start="3">
<li>ÂêàÂπ∂ÂêÑÊ†∑Êú¨ÁªìÊûú</li>
</ol>
<div class="highlight"><div class="chroma">
<table class="lntable"><tr><td class="lntd">
<pre tabindex="0" class="chroma"><code><span class="lnt"> 1
</span><span class="lnt"> 2
</span><span class="lnt"> 3
</span><span class="lnt"> 4
</span><span class="lnt"> 5
</span><span class="lnt"> 6
</span><span class="lnt"> 7
</span><span class="lnt"> 8
</span><span class="lnt"> 9
</span><span class="lnt">10
</span><span class="lnt">11
</span><span class="lnt">12
</span></code></pre></td>
<td class="lntd">
<pre tabindex="0" class="chroma"><code class="language-bash" data-lang="bash"><span class="line"><span class="cl"><span class="c1">## ÂêàÂπ∂</span>
</span></span><span class="line"><span class="cl">mkdir -p salmon
</span></span><span class="line"><span class="cl">salmon quantmerge <span class="se">\
</span></span></span><span class="line"><span class="cl"><span class="se"></span>    --quants temp/salmon/*.quant <span class="se">\
</span></span></span><span class="line"><span class="cl"><span class="se"></span>    -o salmon/gene.TPM
</span></span><span class="line"><span class="cl">salmon quantmerge <span class="se">\
</span></span></span><span class="line"><span class="cl"><span class="se"></span>    --quants temp/salmon/*.quant <span class="se">\
</span></span></span><span class="line"><span class="cl"><span class="se"></span>    --column NumReads -o salmon/gene.count
</span></span><span class="line"><span class="cl">sed -i <span class="s1">&#39;1 s/.quant//g&#39;</span> salmon/gene.*
</span></span><span class="line"><span class="cl">
</span></span><span class="line"><span class="cl"><span class="c1">## È¢ÑËßàÁªìÊûúË°®Ê†º</span>
</span></span><span class="line"><span class="cl">head -n3 salmon/gene.*
</span></span></code></pre></td></tr></table>
</div>
</div><h3 id="ÂäüËÉΩÂü∫Âõ†Ê≥®Èáä">ÂäüËÉΩÂü∫Âõ†Ê≥®Èáä</h3>
<p>‰∏ä‰∏ÄÊ≠•Â∑≤ÁªèÊúâ‰∫ÜÊâÄÊúâÁöÑÂü∫Âõ†ÂíåÊØè‰∏™Ê†∑Êú¨ÊâÄÊúâÂü∫Âõ†ÁöÑread countÂÆöÈáèÁªìÊûúÔºåÊàë‰ª¨Âè™ÈúÄË¶ÅÂØπ‰∏ä‰∏ÄÊ≠•ÁöÑÂü∫Âõ†Â∫èÂàóÔºàÊàñËõãÁôΩË¥®Â∫èÂàóÔºâËøõË°å‰∏çÂêåÊï∞ÊçÆÂ∫ìÁöÑÊ≥®ÈáäÔºàÂæàÂ§öËΩØ‰ª∂ÈÉΩÊòØÁî®diamondÊØîÂØπÔºåÂ¶ÇÊûúÊ≤°Êúâ‰∏ìÁî®ËΩØ‰ª∂ÁöÑÊï∞ÊçÆÂ∫ìÊàë‰ª¨‰πüÂèØ‰ª•Ëá™Â∑±Áî®diamondÊØîÂØπÔºâÔºåÂêàÂπ∂Ê≥®ÈáäÁªìÊûúÂæóÂà∞ÁöÑÂ∞±ÊòØÂäüËÉΩ‰∏∞Â∫¶Ë°®„ÄÇ</p>
<p>diamondÈÄâÊã©&ndash;outfmt 6ÁöÑËæìÂá∫ÁªìÊûúÂíåblastp‰∏ÄÊ†∑Ôºö</p>
<div class="table-wrapper"><table>
<thead>
<tr>
<th></th>
<th></th>
<th></th>
</tr>
</thead>
<tbody>
<tr>
<td>1.</td>
<td>qseqid</td>
<td>query sequence id</td>
</tr>
<tr>
<td>2.</td>
<td>sseqid</td>
<td>subject (e.g., reference genome) sequence id</td>
</tr>
<tr>
<td>3.</td>
<td>pident</td>
<td>percentage of identical matches</td>
</tr>
<tr>
<td>4.</td>
<td>length</td>
<td>alignment length</td>
</tr>
<tr>
<td>5.</td>
<td>mismatch</td>
<td>number of mismatches</td>
</tr>
<tr>
<td>6.</td>
<td>gapopen</td>
<td>number of gap openings</td>
</tr>
<tr>
<td>7.</td>
<td>qstart</td>
<td>start of alignment in query</td>
</tr>
<tr>
<td>8.</td>
<td>qend</td>
<td>end of alignment in query</td>
</tr>
<tr>
<td>9.</td>
<td>sstart</td>
<td>start of alignment in subject</td>
</tr>
<tr>
<td>10.</td>
<td>send</td>
<td>end of alignment in subject</td>
</tr>
<tr>
<td>11.</td>
<td>evalue</td>
<td>expect value</td>
</tr>
<tr>
<td>12.</td>
<td>bitscore</td>
<td>bit score</td>
</tr>
</tbody>
</table></div>
<h4 id="1-eggnogcogkeggcazy">1 eggNOG(COG/KEGG/CAZy)</h4>
<p><a class="link" href="http://eggnogdb.embl.de/"  target="_blank" rel="noopener"
    >EggNOGÊï∞ÊçÆÂ∫ì</a>Êî∂ÈõÜ‰∫ÜCOGÔºàClusters of Orthologous Groups of proteinsÔºåÁõ¥Á≥ªÂêåÊ∫êËõãÁôΩÁ∞áÔºâ,ÊûÑÊàêÊØè‰∏™COGÁöÑËõãÁôΩÈÉΩÊòØË¢´ÂÅáÂÆö‰∏∫Êù•Ëá™‰∫é‰∏Ä‰∏™Á•ñÂÖàËõãÁôΩÔºåÂõ†Ê≠§ÊòØorthologsÊàñËÄÖÊòØparalogs„ÄÇÈÄöËøáÊääÊâÄÊúâÂÆåÊï¥Âü∫Âõ†ÁªÑÁöÑÁºñÁ†ÅËõãÁôΩ‰∏Ä‰∏™‰∏Ä‰∏™ÁöÑ‰∫íÁõ∏ÊØîËæÉÁ°ÆÂÆöÁöÑ„ÄÇÂú®ËÄÉËôëÊù•Ëá™‰∏Ä‰∏™ÁªôÂÆöÂü∫Âõ†ÁªÑÁöÑËõãÁôΩÊó∂ÔºåËøôÁßçÊØîËæÉÂ∞ÜÁªôÂá∫ÊØè‰∏™ÂÖ∂‰ªñÂü∫Âõ†ÁªÑÁöÑ‰∏Ä‰∏™ÊúÄÁõ∏‰ººÁöÑËõãÁôΩÔºàÂõ†Ê≠§ÈúÄË¶ÅÁî®ÂÆåÊï¥ÁöÑÂü∫Âõ†ÁªÑÊù•ÂÆö‰πâCOGÔºâÔºåËøô‰∫õÂü∫Âõ†ÁöÑÊØè‰∏Ä‰∏™ÈÉΩËΩÆÁï™Âú∞Ë¢´ËÄÉËôë„ÄÇÂ¶ÇÊûúÂú®Ëøô‰∫õËõãÁôΩÔºàÊàñÂ≠êÈõÜÔºâ‰πãÈó¥‰∏Ä‰∏™Áõ∏‰∫íÁöÑÊúÄ‰Ω≥ÂåπÈÖçÂÖ≥Á≥ªË¢´ÂèëÁé∞ÔºåÈÇ£‰πàÈÇ£‰∫õÁõ∏‰∫íÁöÑÊúÄ‰Ω≥ÂåπÈÖçÂ∞ÜÂΩ¢Êàê‰∏Ä‰∏™COG„ÄÇËøôÊ†∑Ôºå‰∏Ä‰∏™COG‰∏≠ÁöÑÊàêÂëòÂ∞Ü‰∏éËøô‰∏™COG‰∏≠ÁöÑÂÖ∂‰ªñÊàêÂëòÊØîËµ∑Ë¢´ÊØîËæÉÁöÑÂü∫Âõ†ÁªÑ‰∏≠ÁöÑÂÖ∂‰ªñËõãÁôΩÊõ¥Áõ∏ÂÉè„ÄÇ</p>
<p><a class="link" href="http://eggnogdb.embl.de/"  target="_blank" rel="noopener"
    >EggNOG</a>ÈáåÈù¢ÂåÖÂê´‰∫ÜGOÔºåKEGGÔºåCAZyÁ≠â„ÄÇ</p>
<div class="highlight"><div class="chroma">
<table class="lntable"><tr><td class="lntd">
<pre tabindex="0" class="chroma"><code><span class="lnt">1
</span><span class="lnt">2
</span><span class="lnt">3
</span><span class="lnt">4
</span><span class="lnt">5
</span><span class="lnt">6
</span><span class="lnt">7
</span><span class="lnt">8
</span><span class="lnt">9
</span></code></pre></td>
<td class="lntd">
<pre tabindex="0" class="chroma"><code class="language-bash" data-lang="bash"><span class="line"><span class="cl"><span class="c1">## ‰∏ãËΩΩÂ∏∏Áî®Êï∞ÊçÆÂ∫ìÔºåÊ≥®ÊÑèËÆæÁΩÆ‰∏ãËΩΩ‰ΩçÁΩÆ</span>
</span></span><span class="line"><span class="cl">mkdir -p <span class="si">${</span><span class="nv">db</span><span class="si">}</span>/eggnog5 <span class="o">&amp;&amp;</span> <span class="nb">cd</span> <span class="si">${</span><span class="nv">db</span><span class="si">}</span>/eggnog5
</span></span><span class="line"><span class="cl"><span class="c1">## -yÈªòËÆ§ÂêåÊÑèÔºå-fÂº∫Âà∂‰∏ãËΩΩÔºåeggnog.db.gz 7.9G+4.9G</span>
</span></span><span class="line"><span class="cl">download_eggnog_data.py -y -f --data_dir ./
</span></span><span class="line"><span class="cl">
</span></span><span class="line"><span class="cl"><span class="c1">## ‰∏ãËΩΩÊñπÂºè2(ÂèØÈÄâ)ÔºöÈìæÊé•Áõ¥Êé•‰∏ãËΩΩ</span>
</span></span><span class="line"><span class="cl">wget -c http://eggnog5.embl.de/download/emapperdb-5.0.0/eggnog.db.gz <span class="c1">## 7.9G</span>
</span></span><span class="line"><span class="cl">wget -c http://eggnog5.embl.de/download/emapperdb-5.0.0/eggnog_proteins.dmnd.gz <span class="c1">## 4.9G</span>
</span></span><span class="line"><span class="cl">gunzip *.gz
</span></span></code></pre></td></tr></table>
</div>
</div><div class="highlight"><div class="chroma">
<table class="lntable"><tr><td class="lntd">
<pre tabindex="0" class="chroma"><code><span class="lnt"> 1
</span><span class="lnt"> 2
</span><span class="lnt"> 3
</span><span class="lnt"> 4
</span><span class="lnt"> 5
</span><span class="lnt"> 6
</span><span class="lnt"> 7
</span><span class="lnt"> 8
</span><span class="lnt"> 9
</span><span class="lnt">10
</span><span class="lnt">11
</span><span class="lnt">12
</span><span class="lnt">13
</span><span class="lnt">14
</span><span class="lnt">15
</span><span class="lnt">16
</span><span class="lnt">17
</span><span class="lnt">18
</span><span class="lnt">19
</span><span class="lnt">20
</span><span class="lnt">21
</span><span class="lnt">22
</span><span class="lnt">23
</span><span class="lnt">24
</span><span class="lnt">25
</span><span class="lnt">26
</span><span class="lnt">27
</span><span class="lnt">28
</span><span class="lnt">29
</span><span class="lnt">30
</span><span class="lnt">31
</span><span class="lnt">32
</span><span class="lnt">33
</span></code></pre></td>
<td class="lntd">
<pre tabindex="0" class="chroma"><code class="language-bash" data-lang="bash"><span class="line"><span class="cl"><span class="cp">#!/bin/bash
</span></span></span><span class="line"><span class="cl"><span class="cp"></span><span class="c1">#SBATCH --job-name=eggo</span>
</span></span><span class="line"><span class="cl"><span class="c1">#SBATCH --output=/share/home/jianglab/pengchen/work/asthma/%x_%j.out</span>
</span></span><span class="line"><span class="cl"><span class="c1">#SBATCH --error=/share/home/jianglab/pengchen/work/asthma/%x_%j.err</span>
</span></span><span class="line"><span class="cl"><span class="c1">#SBATCH --cpus-per-task=32</span>
</span></span><span class="line"><span class="cl"><span class="c1">#SBATCH --partition=cpu</span>
</span></span><span class="line"><span class="cl">
</span></span><span class="line"><span class="cl"><span class="nb">echo</span> start: <span class="sb">`</span>date +<span class="s1">&#39;%Y-%m-%d %T&#39;</span><span class="sb">`</span>
</span></span><span class="line"><span class="cl"><span class="nv">start</span><span class="o">=</span><span class="sb">`</span>date +%s<span class="sb">`</span>
</span></span><span class="line"><span class="cl"><span class="c1">##############</span>
</span></span><span class="line"><span class="cl"><span class="c1">#ÂàáÊç¢ÁéØÂ¢É</span>
</span></span><span class="line"><span class="cl"><span class="c1">## diamondÊØîÂØπÂü∫Âõ†Ëá≥eggNOG 5.0Êï∞ÊçÆÂ∫ì, 1~9hÔºåÈªòËÆ§diamond 1e-3</span>
</span></span><span class="line"><span class="cl">mkdir -p temp/eggnog
</span></span><span class="line"><span class="cl">emapper.py --no_annot --no_file_comments --override <span class="se">\
</span></span></span><span class="line"><span class="cl"><span class="se"></span>  --data_dir ~/db/eggnog5 <span class="se">\
</span></span></span><span class="line"><span class="cl"><span class="se"></span>  -i NR/protein.fa <span class="se">\
</span></span></span><span class="line"><span class="cl"><span class="se"></span>  --cpu <span class="m">32</span> -m diamond <span class="se">\
</span></span></span><span class="line"><span class="cl"><span class="se"></span>  -o temp/eggnog/protein
</span></span><span class="line"><span class="cl"><span class="c1">## ÊØîÂØπÁªìÊûúÂäüËÉΩÊ≥®Èáä, 1h</span>
</span></span><span class="line"><span class="cl">emapper.py <span class="se">\
</span></span></span><span class="line"><span class="cl"><span class="se"></span>  --annotate_hits_table temp/eggnog/protein.emapper.seed_orthologs <span class="se">\
</span></span></span><span class="line"><span class="cl"><span class="se"></span>  --data_dir ~/db/eggnog5 <span class="se">\
</span></span></span><span class="line"><span class="cl"><span class="se"></span>  --cpu <span class="m">32</span> --no_file_comments --override <span class="se">\
</span></span></span><span class="line"><span class="cl"><span class="se"></span>  -o temp/eggnog/output
</span></span><span class="line"><span class="cl">
</span></span><span class="line"><span class="cl"><span class="c1">## Ê∑ªË°®Â§¥, 1Âàó‰∏∫IDÔºå9ÂàóKOÔºå16ÂàóCAZyÔºå21ÂàóCOGÔºå22ÂàóÊèèËø∞</span>
</span></span><span class="line"><span class="cl">sed <span class="s1">&#39;1 i Name\tortholog\tevalue\tscore\ttaxonomic\tprotein\tGO\tEC\tKO\tPathway\tModule\tReaction\trclass\tBRITE\tTC\tCAZy\tBiGG\ttax_scope\tOG\tbestOG\tCOG\tdescription&#39;</span> <span class="se">\
</span></span></span><span class="line"><span class="cl"><span class="se"></span>  temp/eggnog/output.emapper.annotations <span class="se">\
</span></span></span><span class="line"><span class="cl"><span class="se"></span>  &gt; temp/eggnog/output
</span></span><span class="line"><span class="cl"><span class="c1">##############</span>
</span></span><span class="line"><span class="cl"><span class="nb">echo</span> end: <span class="sb">`</span>date +<span class="s1">&#39;%Y-%m-%d %T&#39;</span><span class="sb">`</span>
</span></span><span class="line"><span class="cl"><span class="nv">end</span><span class="o">=</span><span class="sb">`</span>date +%s<span class="sb">`</span>
</span></span><span class="line"><span class="cl"><span class="nb">echo</span> TIME:<span class="sb">`</span>expr <span class="nv">$end</span> - <span class="nv">$start</span><span class="sb">`</span>s
</span></span></code></pre></td></tr></table>
</div>
</div><h4 id="2-Á¢≥Ê∞¥ÂåñÂêàÁâ©dbcan2">2 Á¢≥Ê∞¥ÂåñÂêàÁâ©dbCAN2</h4>
<div class="highlight"><div class="chroma">
<table class="lntable"><tr><td class="lntd">
<pre tabindex="0" class="chroma"><code><span class="lnt"> 1
</span><span class="lnt"> 2
</span><span class="lnt"> 3
</span><span class="lnt"> 4
</span><span class="lnt"> 5
</span><span class="lnt"> 6
</span><span class="lnt"> 7
</span><span class="lnt"> 8
</span><span class="lnt"> 9
</span><span class="lnt">10
</span><span class="lnt">11
</span><span class="lnt">12
</span><span class="lnt">13
</span><span class="lnt">14
</span></code></pre></td>
<td class="lntd">
<pre tabindex="0" class="chroma"><code class="language-bash" data-lang="bash"><span class="line"><span class="cl"><span class="c1">## dbCAN2 http://bcb.unl.edu/dbCAN2</span>
</span></span><span class="line"><span class="cl"><span class="c1">## ÂàõÂª∫Êï∞ÊçÆÂ∫ìÂ≠òÊîæÁõÆÂΩïÂπ∂ËøõÂÖ•</span>
</span></span><span class="line"><span class="cl">mkdir -p <span class="si">${</span><span class="nv">db</span><span class="si">}</span>/dbCAN2 <span class="o">&amp;&amp;</span> <span class="nb">cd</span> <span class="si">${</span><span class="nv">db</span><span class="si">}</span>/dbCAN2
</span></span><span class="line"><span class="cl"><span class="c1">## ‰∏ãËΩΩÂ∫èÂàóÂíåÊèèËø∞</span>
</span></span><span class="line"><span class="cl">wget -c http://bcb.unl.edu/dbCAN2/download/CAZyDB.07312020.fa
</span></span><span class="line"><span class="cl">wget -c http://bcb.unl.edu/dbCAN2/download/Databases/CAZyDB.07302020.fam-activities.txt
</span></span><span class="line"><span class="cl"><span class="c1">## Â§áÁî®Êï∞ÊçÆÂ∫ì‰∏ãËΩΩÂú∞ÂùÄÂπ∂Ëß£Âéã </span>
</span></span><span class="line"><span class="cl"><span class="c1">#wget -c http://210.75.224.110/db/dbcan2/CAZyDB.07312020.fa.gz</span>
</span></span><span class="line"><span class="cl"><span class="c1">#gunzip CAZyDB.07312020.fa.gz</span>
</span></span><span class="line"><span class="cl"><span class="c1">## diamondÂª∫Á¥¢ÂºïÔºå800MÔºå1m</span>
</span></span><span class="line"><span class="cl">diamond --version <span class="c1">## 0.8.22/2.0.5</span>
</span></span><span class="line"><span class="cl"><span class="nb">time</span> diamond makedb <span class="se">\
</span></span></span><span class="line"><span class="cl"><span class="se"></span>  --in CAZyDB.07312020.fa <span class="se">\
</span></span></span><span class="line"><span class="cl"><span class="se"></span>  --db CAZyDB.07312020
</span></span></code></pre></td></tr></table>
</div>
</div><div class="highlight"><div class="chroma">
<table class="lntable"><tr><td class="lntd">
<pre tabindex="0" class="chroma"><code><span class="lnt">1
</span><span class="lnt">2
</span><span class="lnt">3
</span><span class="lnt">4
</span><span class="lnt">5
</span><span class="lnt">6
</span><span class="lnt">7
</span><span class="lnt">8
</span></code></pre></td>
<td class="lntd">
<pre tabindex="0" class="chroma"><code class="language-bash" data-lang="bash"><span class="line"><span class="cl"><span class="c1">## ÊØîÂØπCAZyÊï∞ÊçÆÂ∫ì, Áî®Êó∂2~18m; Âä†--sensitiveÊõ¥ÂÖ®‰ΩÜÊÖ¢Ëá≥1h</span>
</span></span><span class="line"><span class="cl">mkdir -p temp/dbcan2
</span></span><span class="line"><span class="cl">diamond blastp   <span class="se">\
</span></span></span><span class="line"><span class="cl"><span class="se"></span>	--db ~/db/dbcan2/CAZyDB.07312020  <span class="se">\
</span></span></span><span class="line"><span class="cl"><span class="se"></span>	--query NR/protein.fa   <span class="se">\
</span></span></span><span class="line"><span class="cl"><span class="se"></span>	--threads <span class="m">64</span> -e 1e-5 --outfmt <span class="m">6</span> <span class="se">\
</span></span></span><span class="line"><span class="cl"><span class="se"></span>	--max-target-seqs <span class="m">1</span> --quiet <span class="se">\
</span></span></span><span class="line"><span class="cl"><span class="se"></span>	--out temp/dbcan2/gene_diamond.f6
</span></span></code></pre></td></tr></table>
</div>
</div><h4 id="3-argscard">3 ARGsÔºàCARDÔºâ</h4>
<p>## Github: <a class="link" href="https://github.com/arpcard/rgi"  target="_blank" rel="noopener"
    >https://github.com/arpcard/rgi</a></p>
<div class="highlight"><div class="chroma">
<table class="lntable"><tr><td class="lntd">
<pre tabindex="0" class="chroma"><code><span class="lnt"> 1
</span><span class="lnt"> 2
</span><span class="lnt"> 3
</span><span class="lnt"> 4
</span><span class="lnt"> 5
</span><span class="lnt"> 6
</span><span class="lnt"> 7
</span><span class="lnt"> 8
</span><span class="lnt"> 9
</span><span class="lnt">10
</span><span class="lnt">11
</span><span class="lnt">12
</span><span class="lnt">13
</span><span class="lnt">14
</span><span class="lnt">15
</span><span class="lnt">16
</span><span class="lnt">17
</span><span class="lnt">18
</span><span class="lnt">19
</span><span class="lnt">20
</span><span class="lnt">21
</span></code></pre></td>
<td class="lntd">
<pre tabindex="0" class="chroma"><code class="language-bash" data-lang="bash"><span class="line"><span class="cl"><span class="cp">#!/bin/bash
</span></span></span><span class="line"><span class="cl"><span class="cp"></span><span class="c1">#SBATCH --job-name=rgi</span>
</span></span><span class="line"><span class="cl"><span class="c1">#SBATCH --output=/share/home/jianglab/pengchen/work/asthma/%x_%a.out</span>
</span></span><span class="line"><span class="cl"><span class="c1">#SBATCH --error=/share/home/jianglab/pengchen/work/asthma/%x_%a.err</span>
</span></span><span class="line"><span class="cl"><span class="c1">#SBATCH --cpus-per-task=32</span>
</span></span><span class="line"><span class="cl"><span class="c1">#SBATCH --partition=cpu</span>
</span></span><span class="line"><span class="cl">
</span></span><span class="line"><span class="cl"><span class="nb">echo</span> start: <span class="sb">`</span>date +<span class="s1">&#39;%Y-%m-%d %T&#39;</span><span class="sb">`</span>
</span></span><span class="line"><span class="cl"><span class="nv">start</span><span class="o">=</span><span class="sb">`</span>date +%s<span class="sb">`</span>
</span></span><span class="line"><span class="cl"><span class="c1">##############</span>
</span></span><span class="line"><span class="cl">mkdir -p card
</span></span><span class="line"><span class="cl">
</span></span><span class="line"><span class="cl">rgi main --input_sequence ~/work/asthma/temp/protein.fa <span class="se">\
</span></span></span><span class="line"><span class="cl"><span class="se"></span>--output_file card/protein <span class="se">\
</span></span></span><span class="line"><span class="cl"><span class="se"></span>--input_type protein --num_threads <span class="m">32</span> <span class="se">\
</span></span></span><span class="line"><span class="cl"><span class="se"></span>--clean --alignment_tool DIAMOND
</span></span><span class="line"><span class="cl">
</span></span><span class="line"><span class="cl"><span class="c1">##############</span>
</span></span><span class="line"><span class="cl"><span class="nb">echo</span> end: <span class="sb">`</span>date +<span class="s1">&#39;%Y-%m-%d %T&#39;</span><span class="sb">`</span>
</span></span><span class="line"><span class="cl"><span class="nv">end</span><span class="o">=</span><span class="sb">`</span>date +%s<span class="sb">`</span>
</span></span><span class="line"><span class="cl"><span class="nb">echo</span> TIME:<span class="sb">`</span>expr <span class="nv">$end</span> - <span class="nv">$start</span><span class="sb">`</span>s
</span></span></code></pre></td></tr></table>
</div>
</div><h4 id="4-ÊØíÂäõÂõ†Â≠êvfdb">4 ÊØíÂäõÂõ†Â≠êVFDB</h4>
<p>ÂÆòÁΩëÂú∞ÂùÄÔºö<a class="link" href="http://www.mgc.ac.cn/VFs/"  target="_blank" rel="noopener"
    >http://www.mgc.ac.cn/VFs/</a> Âú®ÂÆòÁΩë‰∏ãËΩΩÊï∞ÊçÆÂ∫ìÊó∂ÔºåÂ∏¶ÊúâsetA ÁöÑÂ∫ì‰∏∫VFDBÊï∞ÊçÆÂ∫ìÊ†∏ÂøÉÂ∫ì(set A)ÔºåËÄåsetB‰∏∫ÂÖ®Â∫ì(setB), ÂÖ∂‰∏≠setA‰ªÖÂåÖÂê´ÁªèÂÆûÈ™åÈ™åËØÅËøáÁöÑÊØíÂäõÂü∫Âõ†ÔºåËÄåsetBÂàôÂú®setAÁöÑÂü∫Á°Ä‰∏äÂ¢ûÂä†‰∫ÜÈ¢ÑÊµãÁöÑÊØíÂäõÂü∫Âõ†ÔºåÈÄâÊã©Â•ΩÊï∞ÊçÆÂ∫ìÂêéÔºåÁõ¥Êé•Áî®blast/diamondÂç≥ÂèØÂÆåÊàêÊ≥®Èáä„ÄÇ</p>
<div class="highlight"><div class="chroma">
<table class="lntable"><tr><td class="lntd">
<pre tabindex="0" class="chroma"><code><span class="lnt">1
</span><span class="lnt">2
</span><span class="lnt">3
</span><span class="lnt">4
</span><span class="lnt">5
</span><span class="lnt">6
</span><span class="lnt">7
</span></code></pre></td>
<td class="lntd">
<pre tabindex="0" class="chroma"><code class="language-bash" data-lang="bash"><span class="line"><span class="cl">mkdir -p temp/vfdb
</span></span><span class="line"><span class="cl">diamond blastp   <span class="se">\
</span></span></span><span class="line"><span class="cl"><span class="se"></span>	--db ~/db/VFDB/VFDB_setB_pro  <span class="se">\
</span></span></span><span class="line"><span class="cl"><span class="se"></span>	--query NR/protein.fa   <span class="se">\
</span></span></span><span class="line"><span class="cl"><span class="se"></span>	--threads <span class="m">32</span> -e 1e-5 --outfmt <span class="m">6</span> <span class="se">\
</span></span></span><span class="line"><span class="cl"><span class="se"></span>	--max-target-seqs <span class="m">1</span> --quiet <span class="se">\
</span></span></span><span class="line"><span class="cl"><span class="se"></span>	--out temp/vfdb/gene_diamond.f6
</span></span></code></pre></td></tr></table>
</div>
</div><h4 id="5-ÂÖ∂‰ªñÂêÑÁßçÊï∞ÊçÆÂ∫ì">5 ÂÖ∂‰ªñÂêÑÁßçÊï∞ÊçÆÂ∫ì</h4>
<h3 id="ÂäüËÉΩÊ≥®ÈáäÂêàÂπ∂">ÂäüËÉΩÊ≥®ÈáäÂêàÂπ∂</h3>
<p>ÂÜô‰∏Ä‰∏™pythonËÑöÊú¨ÔºåÂ∞ÜË°®1ÔºàÂü∫Âõ†-ÂäüËÉΩÁöÑÂØπÂ∫îË°®Ôºâ‰∏éË°®2ÔºàÂü∫Âõ†‰∏∞Â∫¶Ë°®ÔºâÂêàÂπ∂ÔºåÂç≥‰∏çÂêåÂü∫Âõ†ÂèØËÉΩÊ≥®ÈáäÂà∞Áõ∏ÂêåÂäüËÉΩÔºåÊääÂÆÉ‰ª¨ÁöÑ‰∏∞Â∫¶Âä†Âú®‰∏ÄËµ∑ÂæóÂà∞Êñ∞Ë°®3ÔºàÂäüËÉΩ‰∏∞Â∫¶Ë°®Ôºâ</p>
<h2 id="binning">binning</h2>
<p>ÂÆèÂü∫Âõ†ÁªÑbinningÊòØÊåáÂ∞Ü‰∏çÂêåÁöÑÂ∫èÂàóÈõÜÂêàÔºàÂ¶ÇmetagenomeÂ∫èÂàóÈõÜÂêàÔºâÊ†πÊçÆÂÆÉ‰ª¨ÁöÑÁâ©ÁßçÂΩíÁ±ªÂà∞‰∏çÂêåÁöÑbins‰∏≠Ôºå‰ª•‰æøËøõ‰∏ÄÊ≠•Á†îÁ©∂ÂÆÉ‰ª¨ÁöÑÁªÑÊàêÂíåÂäüËÉΩ„ÄÇËøô‰∏™ËøáÁ®ãÂèØ‰ª•Â∞ÜÁ±ª‰ººÁöÑÂ∫èÂàóÁªÑÂêàÂú®‰∏ÄËµ∑ÔºåÂΩ¢Êàê‰ª£Ë°®‰∏çÂêåÁâ©ÁßçÊàñÂü∫Âõ†ÁªÑÁöÑbinsÔºå‰ª•‰æøËøõË°åÂêéÁª≠ÂàÜÊûêÔºåÂ¶ÇÁâ©ÁßçÊ≥®Èáä„ÄÅÂü∫Âõ†ÁªÑÁªÑË£ÖÁ≠â„ÄÇ</p>
<p>‰ª•‰∏ãÊòØÂ∏∏Áî®ÁöÑÂÆèÂü∫Âõ†ÁªÑbinningÊñπÊ≥ïÔºö</p>
<ol>
<li>
<p>Âü∫‰∫éËÅöÁ±ªÁöÑÊñπÊ≥ïÔºöËØ•ÊñπÊ≥ï‰ΩøÁî®Â∫èÂàóËÅöÁ±ªÂ∞ÜÁõ∏‰ººÂ∫èÂàóÂàÜÂà∞Âêå‰∏Ä‰∏™bin‰∏≠„ÄÇ‰∏ÄËà¨Êù•ËØ¥ÔºåËÅöÁ±ªÁÆóÊ≥ïÂèØÂàÜ‰∏∫‰∏§Á±ªÔºöÊó†ÁõëÁù£ËÅöÁ±ªÔºàÂ¶Çk-means„ÄÅDBSCANÁ≠âÔºâÂíåÊúâÁõëÁù£ËÅöÁ±ªÔºàÂ¶ÇCAMI„ÄÅMyCCÁ≠âÔºâ„ÄÇ</p>
</li>
<li>
<p>Âü∫‰∫éÁªÑË£ÖÁöÑÊñπÊ≥ïÔºöËØ•ÊñπÊ≥ï‰ΩøÁî®de novoÁªÑË£ÖÊù•Â∞ÜÁõ∏‰ººÂ∫èÂàóÁªÑË£ÖÊàêËøûÁª≠ÁöÑÂ∫èÂàóÔºåÂÜçÊ†πÊçÆËøô‰∫õÂ∫èÂàóÁöÑÂü∫Âõ†ÁªÑ‰ø°ÊÅØÊù•Â∞ÜÂÖ∂ÂàÜÁ±ªÂà∞‰∏çÂêåÁöÑbins‰∏≠„ÄÇËøôÁßçÊñπÊ≥ïÁöÑ‰ºòÁÇπÊòØÂèØ‰ª•Êõ¥Â•ΩÂú∞Â§ÑÁêÜÈáçÂ§çÂ∫èÂàóÔºåÁº∫ÁÇπÊòØÈúÄË¶ÅÂ§ßÈáèÁöÑËÆ°ÁÆóËµÑÊ∫êÂíåÊó∂Èó¥„ÄÇ</p>
</li>
<li>
<p>Âü∫‰∫éÂàÜÁ±ªÂô®ÁöÑÊñπÊ≥ïÔºöËØ•ÊñπÊ≥ï‰ΩøÁî®Êú∫Âô®Â≠¶‰π†ÂàÜÁ±ªÂô®Êù•Â∞ÜÂ∫èÂàóÂàÜÈÖçÂà∞‰∏çÂêåÁöÑbins‰∏≠„ÄÇËøôÁßçÊñπÊ≥ïÁöÑ‰ºòÁÇπÊòØÂèØ‰ª•Ëá™Âä®Â≠¶‰π†ÁâπÂæÅÂπ∂Âú®Â§ÑÁêÜÂ§ßËßÑÊ®°Êï∞ÊçÆÊó∂ÊïàÁéáÈ´òÔºåÁº∫ÁÇπÊòØÈúÄË¶ÅÂÖàÂª∫Á´ã‰∏Ä‰∏™ÂàÜÁ±ªÂô®Âπ∂ËøõË°åËÆ≠ÁªÉ„ÄÇ</p>
</li>
</ol>
<p>Âú®ËøõË°åÂÆèÂü∫Âõ†ÁªÑbinningÊó∂ÔºåÈÄöÂ∏∏ÈúÄË¶Å‰ΩøÁî®Â§ö‰∏™ÊñπÊ≥ïËøõË°åÊØîËæÉÔºå‰ª•ÈÄâÊã©ÊúÄÈÄÇÂêàÊï∞ÊçÆÈõÜÁöÑÊñπÊ≥ï„ÄÇÂèØ‰ª•‰ΩøÁî®‰∏Ä‰∫õÊµÅË°åÁöÑÂ∑•ÂÖ∑Êù•ËøõË°åbinningÔºåÂ¶ÇMetaBAT„ÄÅMaxBin„ÄÅCONCOCTÂíåMEGANÁ≠â„ÄÇËøô‰∫õÂ∑•ÂÖ∑ÈÄöÂ∏∏ÂåÖÂê´ÂêÑÁßçbinningÊñπÊ≥ïÔºåÂèØ‰ª•Ê†πÊçÆÊï∞ÊçÆÈõÜÂíåÂàÜÊûêÁõÆÁöÑÈÄâÊã©ÈÄÇÂêàÁöÑÊñπÊ≥ï„ÄÇ</p>
<p>ÁØáÂπÖÈôêÂà∂ÔºåÂÖ∑‰ΩìÁöÑÊñπÊ≥ïÊîæÂú®Âè¶‰∏ÄÁØáÈáåÈù¢ËÆ≤Ëß£Âêß„ÄÇ</p>
<h2 id="reference">Reference</h2>
<div id="refs" class="references csl-bib-body">
<div id="ref-liuPracticalGuideAmplicon2021" class="csl-entry">
<p><span class="csl-left-margin">1. </span><span class="csl-right-inline">Y.-X. Liu, Y. Qin, T. Chen, M. Lu, X. Qian, X. Guo, Y. Bai, <a class="link" href="https://doi.org/10.1007/s13238-020-00724-8"  target="_blank" rel="noopener"
    >A practical guide to amplicon and metagenomic analysis of microbiome data</a>. <em>Protein &amp; Cell</em>. <strong>12</strong>, 315‚Äì330 (2021).</span></p>
</div>
</div>

</section>


    <footer class="article-footer">
    
    <section class="article-tags">
        
            <a href="/tags/shell/">shell</a>
        
            <a href="/tags/%E7%BC%96%E7%A8%8B/">ÁºñÁ®ã</a>
        
    </section>


    
    <section class="article-copyright">
        <svg xmlns="http://www.w3.org/2000/svg" class="icon icon-tabler icon-tabler-copyright" width="24" height="24" viewBox="0 0 24 24" stroke-width="2" stroke="currentColor" fill="none" stroke-linecap="round" stroke-linejoin="round">
  <path stroke="none" d="M0 0h24v24H0z"/>
  <circle cx="12" cy="12" r="9" />
  <path d="M14.5 9a3.5 4 0 1 0 0 6" />
</svg>



        <span>Licensed under CC BY-NC-SA 4.0</span>
    </section>
    </footer>


    
        <link 
                rel="stylesheet" 
                href="https://cdn.jsdelivr.net/npm/katex@0.15.6/dist/katex.min.css"integrity="sha256-J&#43;iAE0sgH8QSz9hpcDxXIftnj65JEZgNhGcgReTTK9s="crossorigin="anonymous"
            ><script 
                src="https://cdn.jsdelivr.net/npm/katex@0.15.6/dist/katex.min.js"integrity="sha256-InsNdER1b2xUewP&#43;pKCUJpkhiqwHgqiPXDlIk7GzBu4="crossorigin="anonymous"
                defer
                >
            </script><script 
                src="https://cdn.jsdelivr.net/npm/katex@0.15.6/dist/contrib/auto-render.min.js"integrity="sha256-y39Mpg7V3D4lhBX4x6O0bUqTV4pSrfgwEfGKfxkOdgI="crossorigin="anonymous"
                defer
                >
            </script><script>
    window.addEventListener("DOMContentLoaded", () => {
        renderMathInElement(document.querySelector(`.article-content`), {
            delimiters: [
                { left: "$$", right: "$$", display: true },
                { left: "$", right: "$", display: false },
                { left: "\\(", right: "\\)", display: false },
                { left: "\\[", right: "\\]", display: true }
            ],
            ignoredClasses: ["gist"]
        });})
</script>
    
</article>

    

    

<aside class="related-content--wrapper">
    <h2 class="section-title">Related content</h2>
    <div class="related-content">
        <div class="flex article-list--tile">
            
                
<article class="has-image">
    <a href="/p/reporter-score/">
        
        
            <div class="article-image">
                <img src="/p/reporter-score/images/misuse.d0024bb6a2d1c513c17b8a0329a1bfbc_huc67289fcd675657fb391b94803a3cc8a_1235442_250x150_fill_box_smart1_3.png" 
                        width="250" 
                        height="150" 
                        loading="lazy"
                        alt="Featured image of post Reporter Score ÂæÆÁîüÁâ©ÂäüËÉΩÂØåÈõÜÂàÜÊûê"
                        data-key="reporter-score" 
                        data-hash="md5-0AJLtqLRxRPBe4oDKaG/vA==">
                
            </div>
        

        <div class="article-details">
            <h2 class="article-title">Reporter Score ÂæÆÁîüÁâ©ÂäüËÉΩÂØåÈõÜÂàÜÊûê</h2>
        </div>
    </a>
</article>

            
                
<article class="">
    <a href="/p/batch-effect/">
        
        

        <div class="article-details">
            <h2 class="article-title">ÁªÑÂ≠¶Êï∞ÊçÆÂéªÈô§ÊâπÊ¨°ÊïàÂ∫îÔºàbatch effectÔºâ</h2>
        </div>
    </a>
</article>

            
                
<article class="">
    <a href="/p/mmseqs2/">
        
        

        <div class="article-details">
            <h2 class="article-title">Mmseqs2ÁöÑÂü∫Á°Ä‰ΩøÁî®</h2>
        </div>
    </a>
</article>

            
        </div>
    </div>
</aside>

     
    
        
    <script src="https://utteranc.es/client.js" 
        repo="Asa12138/pc_blog"
        issue-term="pathname"
        
        label="none"
        
        crossorigin="anonymous"
        async
        >
</script>

<style>
    .utterances {
        max-width: unset;
    }
</style>

<script>
    let utterancesLoaded = false;

    function setUtterancesTheme(theme) {
        let utterances = document.querySelector('.utterances iframe');
        if (utterances) {
            utterances.contentWindow.postMessage(
                {
                    type: 'set-theme',
                    theme: `github-${theme}`
                },
                'https://utteranc.es'
            );
        }
    }

    addEventListener('message', event => {
        if (event.origin !== 'https://utteranc.es') return;

        
        utterancesLoaded = true;
        setUtterancesTheme(document.documentElement.dataset.scheme)
    });

    window.addEventListener('onColorSchemeChange', (e) => {
        if (!utterancesLoaded) return;
        setUtterancesTheme(e.detail)
    })
</script>


    

    <footer class="site-footer">
    <section class="copyright">
        &copy; 
        
            2022 - 
        
        2023 Asa&#39;s blog
    </section>
    
    <section class="powerby">
        
            Email: pengchen2001@zju.edu.cn <br/>
        Built with <a href="https://gohugo.io/" target="_blank" rel="noopener">Hugo</a> <br />
        Theme <b><a href="https://github.com/CaiJimmy/hugo-theme-stack" target="_blank" rel="noopener" data-version="3.16.0">Stack</a></b> designed by <a href="https://jimmycai.com" target="_blank" rel="noopener">Jimmy</a>
    </section>
</footer>


    
<div class="pswp" tabindex="-1" role="dialog" aria-hidden="true">

    
    <div class="pswp__bg"></div>

    
    <div class="pswp__scroll-wrap">

        
        <div class="pswp__container">
            <div class="pswp__item"></div>
            <div class="pswp__item"></div>
            <div class="pswp__item"></div>
        </div>

        
        <div class="pswp__ui pswp__ui--hidden">

            <div class="pswp__top-bar">

                

                <div class="pswp__counter"></div>

                <button class="pswp__button pswp__button--close" title="Close (Esc)"></button>

                <button class="pswp__button pswp__button--share" title="Share"></button>

                <button class="pswp__button pswp__button--fs" title="Toggle fullscreen"></button>

                <button class="pswp__button pswp__button--zoom" title="Zoom in/out"></button>

                
                
                <div class="pswp__preloader">
                    <div class="pswp__preloader__icn">
                        <div class="pswp__preloader__cut">
                            <div class="pswp__preloader__donut"></div>
                        </div>
                    </div>
                </div>
            </div>

            <div class="pswp__share-modal pswp__share-modal--hidden pswp__single-tap">
                <div class="pswp__share-tooltip"></div>
            </div>

            <button class="pswp__button pswp__button--arrow--left" title="Previous (arrow left)">
            </button>

            <button class="pswp__button pswp__button--arrow--right" title="Next (arrow right)">
            </button>

            <div class="pswp__caption">
                <div class="pswp__caption__center"></div>
            </div>

        </div>

    </div>

</div><script 
                src="https://cdn.jsdelivr.net/npm/photoswipe@4.1.3/dist/photoswipe.min.js"integrity="sha256-ePwmChbbvXbsO02lbM3HoHbSHTHFAeChekF1xKJdleo="crossorigin="anonymous"
                defer
                >
            </script><script 
                src="https://cdn.jsdelivr.net/npm/photoswipe@4.1.3/dist/photoswipe-ui-default.min.js"integrity="sha256-UKkzOn/w1mBxRmLLGrSeyB4e1xbrp4xylgAWb3M42pU="crossorigin="anonymous"
                defer
                >
            </script><link 
                rel="stylesheet" 
                href="https://cdn.jsdelivr.net/npm/photoswipe@4.1.3/dist/default-skin/default-skin.min.css"crossorigin="anonymous"
            ><link 
                rel="stylesheet" 
                href="https://cdn.jsdelivr.net/npm/photoswipe@4.1.3/dist/photoswipe.min.css"crossorigin="anonymous"
            >

            </main>
        </div>
        <script 
                src="https://cdn.jsdelivr.net/npm/node-vibrant@3.1.6/dist/vibrant.min.js"integrity="sha256-awcR2jno4kI5X0zL8ex0vi2z&#43;KMkF24hUW8WePSA9HM="crossorigin="anonymous"
                
                >
            </script><script type="text/javascript" src="/ts/main.js" defer></script>
<script>
    (function () {
        const customFont = document.createElement('link');
        customFont.href = "https://fonts.googleapis.com/css2?family=Lato:wght@300;400;700&display=swap";

        customFont.type = "text/css";
        customFont.rel = "stylesheet";

        document.head.appendChild(customFont);
    }());
</script>

    </body>
</html>
