<!DOCTYPE html>
<html lang="en-us" dir="ltr">
    <head><meta charset='utf-8'>
<meta name='viewport' content='width=device-width, initial-scale=1'><meta name='description' content='å®åŸºå› ç»„ï¼ˆMetagenomeï¼‰æ˜¯æŒ‡å¯¹ä¸€ä¸ªç”Ÿæ€ç³»ç»Ÿä¸­çš„æ‰€æœ‰å¾®ç”Ÿç‰©è¿›è¡ŒDNAåˆ†æçš„è¿‡ç¨‹ï¼Œå¯ä»¥å¸®åŠ©ç ”ç©¶äººå‘˜äº†è§£å¾®ç”Ÿç‰©çš„å¤šæ ·æ€§ã€åŠŸèƒ½å’Œäº’ä½œå…³ç³»ã€‚è¿™é‡Œä»‹ç»å¸¸ç”¨åˆ†ææµç¨‹ã€‚'>
<title>Metagenomic workflow</title>

<link rel='canonical' href='/p/metagenomic-workflow/'>

<link rel="stylesheet" href="/scss/style.min.8191399262444ab68b72a18c97392f5349be20a1615d77445be51e974c144cff.css"><meta property='og:title' content='Metagenomic workflow'>
<meta property='og:description' content='å®åŸºå› ç»„ï¼ˆMetagenomeï¼‰æ˜¯æŒ‡å¯¹ä¸€ä¸ªç”Ÿæ€ç³»ç»Ÿä¸­çš„æ‰€æœ‰å¾®ç”Ÿç‰©è¿›è¡ŒDNAåˆ†æçš„è¿‡ç¨‹ï¼Œå¯ä»¥å¸®åŠ©ç ”ç©¶äººå‘˜äº†è§£å¾®ç”Ÿç‰©çš„å¤šæ ·æ€§ã€åŠŸèƒ½å’Œäº’ä½œå…³ç³»ã€‚è¿™é‡Œä»‹ç»å¸¸ç”¨åˆ†ææµç¨‹ã€‚'>
<meta property='og:url' content='/p/metagenomic-workflow/'>
<meta property='og:site_name' content='Asa&#39;s blog'>
<meta property='og:type' content='article'><meta property='article:section' content='Post' /><meta property='article:tag' content='shell' /><meta property='article:tag' content='ç¼–ç¨‹' /><meta property='article:published_time' content='2023-03-26T00:00:00&#43;00:00'/><meta property='article:modified_time' content='2023-03-26T00:00:00&#43;00:00'/><meta property='og:image' content='/' />
<meta name="twitter:title" content="Metagenomic workflow">
<meta name="twitter:description" content="å®åŸºå› ç»„ï¼ˆMetagenomeï¼‰æ˜¯æŒ‡å¯¹ä¸€ä¸ªç”Ÿæ€ç³»ç»Ÿä¸­çš„æ‰€æœ‰å¾®ç”Ÿç‰©è¿›è¡ŒDNAåˆ†æçš„è¿‡ç¨‹ï¼Œå¯ä»¥å¸®åŠ©ç ”ç©¶äººå‘˜äº†è§£å¾®ç”Ÿç‰©çš„å¤šæ ·æ€§ã€åŠŸèƒ½å’Œäº’ä½œå…³ç³»ã€‚è¿™é‡Œä»‹ç»å¸¸ç”¨åˆ†ææµç¨‹ã€‚"><meta name="twitter:card" content="summary_large_image">
    <meta name="twitter:image" content='/' /><link rel="apple-touch-icon" sizes="180x180" href="/apple-touch-icon.png">
<link rel="icon" type="image/png" sizes="32x32" href="/favicon-32x32.png">
<link rel="icon" type="image/png" sizes="16x16" href="/favicon-16x16.png">
<link rel="manifest" href="/site.webmanifest">
<link rel="shortcut icon" href="/favicon-32x32.png" />
    </head>
    <body class="
    article-page
    ">
    <script>
        (function() {
            const colorSchemeKey = 'StackColorScheme';
            if(!localStorage.getItem(colorSchemeKey)){
                localStorage.setItem(colorSchemeKey, "auto");
            }
        })();
    </script><script>
    (function() {
        const colorSchemeKey = 'StackColorScheme';
        const colorSchemeItem = localStorage.getItem(colorSchemeKey);
        const supportDarkMode = window.matchMedia('(prefers-color-scheme: dark)').matches === true;

        if (colorSchemeItem == 'dark' || colorSchemeItem === 'auto' && supportDarkMode) {
            

            document.documentElement.dataset.scheme = 'dark';
        } else {
            document.documentElement.dataset.scheme = 'light';
        }
    })();
</script>
<div class="container main-container flex on-phone--column extended"><aside class="sidebar left-sidebar sticky ">
    <button class="hamburger hamburger--spin" type="button" id="toggle-menu" aria-label="Toggle Menu">
        <span class="hamburger-box">
            <span class="hamburger-inner"></span>
        </span>
    </button>

    <header>
        
            
            <figure class="site-avatar">
                <a href="/">
                
                    
                    
                    
                        
                        <img src="/img/guodong1_hu19ef19e107d2cae4138b147ef6582252_202975_300x0_resize_q75_box.jpg" width="300"
                            height="302" class="site-logo" loading="lazy" alt="Avatar">
                    
                
                </a>
                
                    <span class="emoji">ğŸ›ğŸ¶</span>
                
            </figure>
            
        
        
        <div class="site-meta">
            <h1 class="site-name"><a href="/">Asa&#39;s blog</a></h1>
            <h2 class="site-description">work hard, live happily.</h2>
        </div>
    </header><ol class="social-menu">
            
                <li>
                    <a 
                        href='https://github.com/Asa12138'
                        target="_blank"
                        title="GitHub"
                        rel="me"
                    >
                        
                        
                            <svg xmlns="http://www.w3.org/2000/svg" class="icon icon-tabler icon-tabler-brand-github" width="24" height="24" viewBox="0 0 24 24" stroke-width="2" stroke="currentColor" fill="none" stroke-linecap="round" stroke-linejoin="round">
  <path stroke="none" d="M0 0h24v24H0z" fill="none"/>
  <path d="M9 19c-4.3 1.4 -4.3 -2.5 -6 -3m12 5v-3.5c0 -1 .1 -1.4 -.5 -2c2.8 -.3 5.5 -1.4 5.5 -6a4.6 4.6 0 0 0 -1.3 -3.2a4.2 4.2 0 0 0 -.1 -3.2s-1.1 -.3 -3.5 1.3a12.3 12.3 0 0 0 -6.2 0c-2.4 -1.6 -3.5 -1.3 -3.5 -1.3a4.2 4.2 0 0 0 -.1 3.2a4.6 4.6 0 0 0 -1.3 3.2c0 4.6 2.7 5.7 5.5 6c-.6 .6 -.6 1.2 -.5 2v3.5" />
</svg>



                        
                    </a>
                </li>
            
                <li>
                    <a 
                        href='https://twitter.com'
                        target="_blank"
                        title="Twitter"
                        rel="me"
                    >
                        
                        
                            <svg xmlns="http://www.w3.org/2000/svg" class="icon icon-tabler icon-tabler-brand-twitter" width="24" height="24" viewBox="0 0 24 24" stroke-width="2" stroke="currentColor" fill="none" stroke-linecap="round" stroke-linejoin="round">
  <path stroke="none" d="M0 0h24v24H0z" fill="none"/>
  <path d="M22 4.01c-1 .49 -1.98 .689 -3 .99c-1.121 -1.265 -2.783 -1.335 -4.38 -.737s-2.643 2.06 -2.62 3.737v1c-3.245 .083 -6.135 -1.395 -8 -4c0 0 -4.182 7.433 4 11c-1.872 1.247 -3.739 2.088 -6 2c3.308 1.803 6.913 2.423 10.034 1.517c3.58 -1.04 6.522 -3.723 7.651 -7.742a13.84 13.84 0 0 0 .497 -3.753c-.002 -.249 1.51 -2.772 1.818 -4.013z" />
</svg>



                        
                    </a>
                </li>
            
        </ol><ol class="menu" id="main-menu">
        
        
        
        <li >
            <a href='/' >
                
                
                
                    <svg xmlns="http://www.w3.org/2000/svg" class="icon icon-tabler icon-tabler-home" width="24" height="24" viewBox="0 0 24 24" stroke-width="2" stroke="currentColor" fill="none" stroke-linecap="round" stroke-linejoin="round">
  <path stroke="none" d="M0 0h24v24H0z"/>
  <polyline points="5 12 3 12 12 3 21 12 19 12" />
  <path d="M5 12v7a2 2 0 0 0 2 2h10a2 2 0 0 0 2 -2v-7" />
  <path d="M9 21v-6a2 2 0 0 1 2 -2h2a2 2 0 0 1 2 2v6" />
</svg>



                
                <span>Home</span>
            </a>
        </li>
        
        
        <li >
            <a href='/about/' >
                
                
                
                    <svg xmlns="http://www.w3.org/2000/svg" class="icon icon-tabler icon-tabler-user" width="24" height="24" viewBox="0 0 24 24" stroke-width="2" stroke="currentColor" fill="none" stroke-linecap="round" stroke-linejoin="round">
  <path stroke="none" d="M0 0h24v24H0z"/>
  <circle cx="12" cy="7" r="4" />
  <path d="M6 21v-2a4 4 0 0 1 4 -4h4a4 4 0 0 1 4 4v2" />
</svg>



                
                <span>About</span>
            </a>
        </li>
        
        
        <li >
            <a href='/archives/' >
                
                
                
                    <svg xmlns="http://www.w3.org/2000/svg" class="icon icon-tabler icon-tabler-archive" width="24" height="24" viewBox="0 0 24 24" stroke-width="2" stroke="currentColor" fill="none" stroke-linecap="round" stroke-linejoin="round">
  <path stroke="none" d="M0 0h24v24H0z"/>
  <rect x="3" y="4" width="18" height="4" rx="2" />
  <path d="M5 8v10a2 2 0 0 0 2 2h10a2 2 0 0 0 2 -2v-10" />
  <line x1="10" y1="12" x2="14" y2="12" />
</svg>



                
                <span>Archives</span>
            </a>
        </li>
        
        
        <li >
            <a href='/search/' >
                
                
                
                    <svg xmlns="http://www.w3.org/2000/svg" class="icon icon-tabler icon-tabler-search" width="24" height="24" viewBox="0 0 24 24" stroke-width="2" stroke="currentColor" fill="none" stroke-linecap="round" stroke-linejoin="round">
  <path stroke="none" d="M0 0h24v24H0z"/>
  <circle cx="10" cy="10" r="7" />
  <line x1="21" y1="21" x2="15" y2="15" />
</svg>



                
                <span>Search</span>
            </a>
        </li>
        
        
        <li >
            <a href='/links/' >
                
                
                
                    <svg xmlns="http://www.w3.org/2000/svg" class="icon icon-tabler icon-tabler-link" width="24" height="24" viewBox="0 0 24 24" stroke-width="2" stroke="currentColor" fill="none" stroke-linecap="round" stroke-linejoin="round">
  <path stroke="none" d="M0 0h24v24H0z"/>
  <path d="M10 14a3.5 3.5 0 0 0 5 0l4 -4a3.5 3.5 0 0 0 -5 -5l-.5 .5" />
  <path d="M14 10a3.5 3.5 0 0 0 -5 0l-4 4a3.5 3.5 0 0 0 5 5l.5 -.5" />
</svg>



                
                <span>Links</span>
            </a>
        </li>
        

        <div class="menu-bottom-section">
                <li id="i18n-switch">  
                    <svg xmlns="http://www.w3.org/2000/svg" class="icon icon-tabler icon-tabler-language" width="24" height="24" viewBox="0 0 24 24" stroke-width="2" stroke="currentColor" fill="none" stroke-linecap="round" stroke-linejoin="round">
  <path stroke="none" d="M0 0h24v24H0z" fill="none"/>
  <path d="M4 5h7" />
  <path d="M9 3v2c0 4.418 -2.239 8 -5 8" />
  <path d="M5 9c-.003 2.144 2.952 3.908 6.7 4" />
  <path d="M12 20l4 -9l4 9" />
  <path d="M19.1 18h-6.2" />
</svg>



                    <select name="language" onchange="window.location.href = this.selectedOptions[0].value">
                        
                            <option value="/" selected>English</option>
                        
                            <option value="/zh-cn/" >ä¸­æ–‡</option>
                        
                    </select>
                </li>
            
            
            
                <li id="dark-mode-toggle">
                    <svg xmlns="http://www.w3.org/2000/svg" class="icon icon-tabler icon-tabler-toggle-left" width="24" height="24" viewBox="0 0 24 24" stroke-width="2" stroke="currentColor" fill="none" stroke-linecap="round" stroke-linejoin="round">
  <path stroke="none" d="M0 0h24v24H0z"/>
  <circle cx="8" cy="12" r="2" />
  <rect x="2" y="6" width="20" height="12" rx="6" />
</svg>



                    <svg xmlns="http://www.w3.org/2000/svg" class="icon icon-tabler icon-tabler-toggle-right" width="24" height="24" viewBox="0 0 24 24" stroke-width="2" stroke="currentColor" fill="none" stroke-linecap="round" stroke-linejoin="round">
  <path stroke="none" d="M0 0h24v24H0z"/>
  <circle cx="16" cy="12" r="2" />
  <rect x="2" y="6" width="20" height="12" rx="6" />
</svg>



                    <span>Dark Mode</span>
                </li>
            
        </div>
    </ol>
</aside>

    <aside class="sidebar right-sidebar sticky">
        
            
                
    <section class="widget archives">
        <div class="widget-icon">
            <svg xmlns="http://www.w3.org/2000/svg" class="icon icon-tabler icon-tabler-hash" width="24" height="24" viewBox="0 0 24 24" stroke-width="2" stroke="currentColor" fill="none" stroke-linecap="round" stroke-linejoin="round">
  <path stroke="none" d="M0 0h24v24H0z"/>
  <line x1="5" y1="9" x2="19" y2="9" />
  <line x1="5" y1="15" x2="19" y2="15" />
  <line x1="11" y1="4" x2="7" y2="20" />
  <line x1="17" y1="4" x2="13" y2="20" />
</svg>



        </div>
        <h2 class="widget-title section-title">Table of contents</h2>
        
        <div class="widget--toc">
            <nav id="TableOfContents">
  <ol>
    <li><a href="#introduction">Introduction</a></li>
    <li><a href="#preprocess">preprocess</a>
      <ol>
        <li><a href="#è´¨æ§fastp">è´¨æ§ï¼šfastp</a></li>
        <li><a href="#å»å®¿ä¸»bowtie2">å»å®¿ä¸»ï¼šbowtie2</a></li>
        <li><a href="#åŸºæœ¬ä¿¡æ¯ç»Ÿè®¡">åŸºæœ¬ä¿¡æ¯ç»Ÿè®¡</a></li>
      </ol>
    </li>
    <li><a href="#reads-based">reads-based</a>
      <ol>
        <li><a href="#ç‰©ç§æ³¨é‡Škraken2">ç‰©ç§æ³¨é‡Šï¼škraken2</a>
          <ol>
            <li><a href="#è¾“å‡ºæ–‡ä»¶æ ¼å¼">è¾“å‡ºæ–‡ä»¶æ ¼å¼</a></li>
          </ol>
        </li>
        <li><a href="#humann">HUMAnN</a></li>
      </ol>
    </li>
    <li><a href="#contigs-based">contigs-based</a>
      <ol>
        <li><a href="#ç»„è£…megahit">ç»„è£…ï¼šmegahit</a></li>
        <li><a href="#ç»„è£…è¯„ä¼°quast">ç»„è£…è¯„ä¼°ï¼šQUAST</a></li>
        <li><a href="#åŸºå› é¢„æµ‹prodigal">åŸºå› é¢„æµ‹ï¼šProdigal</a></li>
        <li><a href="#å»å†—ä½™">å»å†—ä½™</a>
          <ol>
            <li><a href="#cd-hit">Cd-hit</a></li>
            <li><a href="#mmseq2">mmseq2</a></li>
          </ol>
        </li>
        <li><a href="#åŸºå› å®šé‡salmon">åŸºå› å®šé‡ï¼šsalmon</a></li>
        <li><a href="#åŠŸèƒ½åŸºå› æ³¨é‡Š">åŠŸèƒ½åŸºå› æ³¨é‡Š</a>
          <ol>
            <li><a href="#1-eggnogcogkeggcazy">1 eggNOG(COG/KEGG/CAZy)</a></li>
            <li><a href="#2-ç¢³æ°´åŒ–åˆç‰©dbcan2">2 ç¢³æ°´åŒ–åˆç‰©dbCAN2</a></li>
            <li><a href="#3-argscard">3 ARGsï¼ˆCARDï¼‰</a></li>
            <li><a href="#4-æ¯’åŠ›å› å­vfdb">4 æ¯’åŠ›å› å­VFDB</a></li>
            <li><a href="#5-å…¶ä»–å„ç§æ•°æ®åº“">5 å…¶ä»–å„ç§æ•°æ®åº“</a></li>
          </ol>
        </li>
        <li><a href="#åŠŸèƒ½æ³¨é‡Šåˆå¹¶">åŠŸèƒ½æ³¨é‡Šåˆå¹¶</a></li>
      </ol>
    </li>
    <li><a href="#binning">binning</a></li>
    <li><a href="#reference">Reference</a></li>
  </ol>
</nav>
        </div>
    </section>

            
        
    </aside>


            <main class="main full-width">
    <article class="main-article">
    <header class="article-header">

    <div class="article-details">
    
    <header class="article-category">
        
            <a href="/categories/metagenomic/" style="background-color: #8dd3c7; color: #fff;">
                metagenomic
            </a>
        
    </header>
    

    <div class="article-title-wrapper">
        <h2 class="article-title">
            <a href="/p/metagenomic-workflow/">Metagenomic workflow</a>
        </h2>
    
        
        <h3 class="article-subtitle">
            å®åŸºå› ç»„ï¼ˆMetagenomeï¼‰æ˜¯æŒ‡å¯¹ä¸€ä¸ªç”Ÿæ€ç³»ç»Ÿä¸­çš„æ‰€æœ‰å¾®ç”Ÿç‰©è¿›è¡ŒDNAåˆ†æçš„è¿‡ç¨‹ï¼Œå¯ä»¥å¸®åŠ©ç ”ç©¶äººå‘˜äº†è§£å¾®ç”Ÿç‰©çš„å¤šæ ·æ€§ã€åŠŸèƒ½å’Œäº’ä½œå…³ç³»ã€‚è¿™é‡Œä»‹ç»å¸¸ç”¨åˆ†ææµç¨‹ã€‚
        </h3>
        
    </div>

    
    
    
    
    <footer class="article-time">
        
            <div>
                <svg xmlns="http://www.w3.org/2000/svg" class="icon icon-tabler icon-tabler-calendar-time" width="56" height="56" viewBox="0 0 24 24" stroke-width="2" stroke="currentColor" fill="none" stroke-linecap="round" stroke-linejoin="round">
  <path stroke="none" d="M0 0h24v24H0z"/>
  <path d="M11.795 21h-6.795a2 2 0 0 1 -2 -2v-12a2 2 0 0 1 2 -2h12a2 2 0 0 1 2 2v4" />
  <circle cx="18" cy="18" r="4" />
  <path d="M15 3v4" />
  <path d="M7 3v4" />
  <path d="M3 11h16" />
  <path d="M18 16.496v1.504l1 1" />
</svg>
                <time class="article-time--published">Mar 26, 2023</time>
            </div>
        

        
            <div>
                <svg xmlns="http://www.w3.org/2000/svg" class="icon icon-tabler icon-tabler-clock" width="24" height="24" viewBox="0 0 24 24" stroke-width="2" stroke="currentColor" fill="none" stroke-linecap="round" stroke-linejoin="round">
  <path stroke="none" d="M0 0h24v24H0z"/>
  <circle cx="12" cy="12" r="9" />
  <polyline points="12 7 12 12 15 15" />
</svg>



                <time class="article-time--reading">
                    10 minute read
                </time>
            </div>
        
    </footer>
    

    
</div>

</header>

    <section class="article-content">
    
    
    <h2 id="introduction">Introduction</h2>
<p>å®åŸºå› ç»„ï¼ˆMetagenomeï¼‰æ˜¯æŒ‡å¯¹ä¸€ä¸ªç”Ÿæ€ç³»ç»Ÿä¸­çš„æ‰€æœ‰å¾®ç”Ÿç‰©è¿›è¡ŒDNAåˆ†æçš„è¿‡ç¨‹ï¼Œå¯ä»¥å¸®åŠ©ç ”ç©¶äººå‘˜äº†è§£å¾®ç”Ÿç‰©çš„å¤šæ ·æ€§ã€åŠŸèƒ½å’Œäº’ä½œå…³ç³»ã€‚</p>
<p>å®åŸºå› ç»„çš„åº”ç”¨éå¸¸å¹¿æ³›ï¼ŒåŒ…æ‹¬ï¼š</p>
<ul>
<li>
<p>ç”Ÿç‰©å¤šæ ·æ€§ç ”ç©¶ï¼šé€šè¿‡å¯¹å®åŸºå› ç»„è¿›è¡Œåˆ†æï¼Œå¯ä»¥äº†è§£ä¸åŒç”Ÿæ€ç³»ç»Ÿä¸­å¾®ç”Ÿç‰©çš„å¤šæ ·æ€§å’Œåˆ†å¸ƒæƒ…å†µã€‚</p>
</li>
<li>
<p>ç”Ÿæ€å­¦ç ”ç©¶ï¼šå®åŸºå› ç»„å¯ä»¥å¸®åŠ©ç ”ç©¶äººå‘˜äº†è§£å¾®ç”Ÿç‰©åœ¨ç”Ÿæ€ç³»ç»Ÿä¸­çš„åŠŸèƒ½ã€äº’ä½œå…³ç³»å’Œç”Ÿæ€ä½ç­‰ã€‚</p>
</li>
<li>
<p>ç”Ÿç‰©æŠ€æœ¯ï¼šå®åŸºå› ç»„å¯ä»¥ç”¨äºç­›é€‰å…·æœ‰ç‰¹å®šåŠŸèƒ½çš„å¾®ç”Ÿç‰©ï¼Œä¾‹å¦‚ï¼Œå¯»æ‰¾èƒ½å¤Ÿé™è§£æœ‰å®³ç‰©è´¨çš„å¾®ç”Ÿç‰©ã€‚</p>
</li>
</ul>
<p>å®åŸºå› ç»„çš„åˆ†æä¸€èˆ¬åŒ…æ‹¬ä»¥ä¸‹æ­¥éª¤ï¼š</p>
<ol>
<li>
<p>DNAæå–ä¸å»ºåº“ã€‚</p>
</li>
<li>
<p>é«˜é€šé‡æµ‹åºï¼šä½¿ç”¨é«˜é€šé‡æµ‹åºæŠ€æœ¯å¯¹æ‰©å¢åçš„DNAè¿›è¡Œæµ‹åºï¼Œå¾—åˆ°åŸå§‹åºåˆ—æ•°æ®ã€‚</p>
</li>
<li>
<p>æ•°æ®æ¸…æ´—å’Œç»„è£…ï¼šå¯¹åŸå§‹æ•°æ®è¿›è¡Œè´¨é‡æ§åˆ¶ã€å»é™¤ä½è´¨é‡åºåˆ—å’Œå†—ä½™åºåˆ—ï¼Œå°†åºåˆ—æ‹¼æ¥æˆè¾ƒé•¿çš„è¿ç»­åºåˆ—ï¼ˆcontigsï¼‰ã€‚</p>
</li>
<li>
<p>åŸºå› æ³¨é‡Šï¼šå°†contigsä¸­çš„åŸºå› è¿›è¡Œæ³¨é‡Šï¼Œå¾—åˆ°åŸºå› åŠŸèƒ½ä¿¡æ¯ã€‚</p>
</li>
<li>
<p>æ•°æ®åˆ†æï¼šäº†è§£å¾®ç”Ÿç‰©å¤šæ ·æ€§ã€ç¾¤è½ç»“æ„ã€åŠŸèƒ½ç‰¹å¾ç­‰ä¿¡æ¯ï¼ˆæ›´å¤šæ˜¯æŒ‡è·å–äº†ç‰©ç§ä¸°åº¦è¡¨æˆ–åŠŸèƒ½ä¸°åº¦è¡¨ä¹‹åçš„è¿›ä¸€æ­¥åˆ†æï¼‰ã€‚</p>
</li>
<li>
<p>MAGs binningï¼Œ è¿›åŒ–åŠ¨æ€ç­‰è¿›ä¸€æ­¥åˆ†æ</p>
</li>
</ol>
<p>è¿™æ˜¯æˆ‘å¸¸ç”¨çš„ä¸€å¥—åŸºæœ¬æµç¨‹(Figure <a href="#fig:1-work">1</a>)ï¼Œå½“ç„¶åœ¨é¢å¯¹ä¸åŒé¡¹ç›®æ—¶åº”è¯¥æœ‰ä¸åŒçš„ä¾§é‡ç‚¹å’Œé€‚ç”¨çš„åˆ†ææ–¹æ³•ï¼Œå¯ä»¥åœ¨æ­¤åŸºç¡€ä¸Šæ·»åŠ æˆ–ä¿®æ”¹ã€‚</p>
<p>æœ€æ—©è¿™æ–¹é¢çš„åˆ†ææˆ‘éƒ½æ˜¯å‚è€ƒåˆ˜æ°¸é‘«è€å¸ˆçš„<a class="link" href="https://github.com/YongxinLiu/EasyMetagenome"  target="_blank" rel="noopener"
    ><strong>EasyMetagenome</strong></a>,ç°åœ¨è¿™å¥—æµç¨‹ä¹Ÿå‘æ–‡ç« äº† (<a class="link" href="#ref-liuPracticalGuideAmplicon2021" ><em>1</em></a>)ï¼Œå€¼å¾—å‚è€ƒï¼Œå¯¹ä¸Šæ‰‹16Sæµ‹åºæ•°æ®æˆ–å®åŸºå› ç»„æ•°æ®éƒ½å¾ˆæœ‰å¸®åŠ©ã€‚</p>
<div class="figure">
<img src="images/workflow.png" alt="Basic workflow" width="80%" />
<p class="caption">
Figure 1: Basic workflow
</p>
</div>
<h2 id="preprocess">preprocess</h2>
<p>ç»å¤§å¤šæ•°è¿™é‡Œä»‹ç»çš„è½¯ä»¶éƒ½æ˜¯ä»…æ”¯æŒlinuxå¹³å°çš„ï¼Œæˆ‘ä»¬åšæµ‹åºæ–‡ä»¶çš„ä¸Šæ¸¸åˆ†æä¹Ÿè‚¯å®šæ˜¯åœ¨æœåŠ¡å™¨ä¸Šåšï¼Œä¸ªäººPCä¸€èˆ¬å¾ˆéš¾æ»¡è¶³éœ€æ±‚ï¼Œæ‰€ä»¥åœ¨åšè¿™äº›åˆ†æå‰å¿…é¡»å…ˆå­¦ä¹ linuxåŸºç¡€çŸ¥è¯†å¦‚æ–‡ä»¶ç³»ç»Ÿï¼Œshellè„šæœ¬ç¼–å†™ï¼Œè½¯ä»¶å®‰è£…ç­‰ã€‚</p>
<p>å®‰è£…è½¯ä»¶å»ºè®®ä½¿ç”¨condaæˆ–mambaï¼ˆæ–°å»ºç¯å¢ƒå’Œç®¡ç†ï¼‰ï¼Œæœ‰å¾ˆå¤šå‚è€ƒæ–¹æ³•ã€‚</p>
<p>æˆ‘ä»¬æœåŠ¡å™¨ä½¿ç”¨çš„æ˜¯slurmä½œä¸šç®¡ç†ç³»ç»Ÿï¼Œå°½é‡å…ˆå­¦ä¹ ä¸€ä¸‹<a class="link" href="https://docs.hpc.sjtu.edu.cn/job/slurm.html"  target="_blank" rel="noopener"
    >slurmçš„ä½¿ç”¨</a>å†å°è¯•æäº¤ä½œä¸šã€‚</p>
<p>ä¸€èˆ¬æŠŠæ‰€æœ‰æ ·æœ¬çš„æµ‹åºåŒç«¯æ–‡ä»¶æ”¾åœ¨ä¸€ä¸ªæ–‡ä»¶å¤¹ä¸‹</p>
<h3 id="è´¨æ§fastp">è´¨æ§ï¼šfastp</h3>
<div class="highlight"><div class="chroma">
<table class="lntable"><tr><td class="lntd">
<pre tabindex="0" class="chroma"><code><span class="lnt"> 1
</span><span class="lnt"> 2
</span><span class="lnt"> 3
</span><span class="lnt"> 4
</span><span class="lnt"> 5
</span><span class="lnt"> 6
</span><span class="lnt"> 7
</span><span class="lnt"> 8
</span><span class="lnt"> 9
</span><span class="lnt">10
</span><span class="lnt">11
</span><span class="lnt">12
</span><span class="lnt">13
</span><span class="lnt">14
</span><span class="lnt">15
</span><span class="lnt">16
</span><span class="lnt">17
</span><span class="lnt">18
</span><span class="lnt">19
</span><span class="lnt">20
</span><span class="lnt">21
</span><span class="lnt">22
</span><span class="lnt">23
</span><span class="lnt">24
</span></code></pre></td>
<td class="lntd">
<pre tabindex="0" class="chroma"><code class="language-bash" data-lang="bash"><span class="line"><span class="cl"><span class="cp">#!/bin/bash
</span></span></span><span class="line"><span class="cl"><span class="cp"></span><span class="c1">#SBATCH --job-name=fastp</span>
</span></span><span class="line"><span class="cl"><span class="c1">#SBATCH --output=/share/home/jianglab/pengchen/work/asthma/fastp/log/%x_%a.out</span>
</span></span><span class="line"><span class="cl"><span class="c1">#SBATCH --error=/share/home/jianglab/pengchen/work/asthma/fastp/log/%x_%a.err</span>
</span></span><span class="line"><span class="cl"><span class="c1">#SBATCH --array=1-33</span>
</span></span><span class="line"><span class="cl"><span class="c1">#SBATCH --partition=short</span>
</span></span><span class="line"><span class="cl"><span class="c1">#SBATCH --cpus-per-task=8</span>
</span></span><span class="line"><span class="cl">
</span></span><span class="line"><span class="cl">
</span></span><span class="line"><span class="cl"><span class="nb">echo</span> start: <span class="sb">`</span>date +<span class="s1">&#39;%Y-%m-%d %T&#39;</span><span class="sb">`</span>
</span></span><span class="line"><span class="cl"><span class="nv">start</span><span class="o">=</span><span class="sb">`</span>date +%s<span class="sb">`</span>
</span></span><span class="line"><span class="cl"><span class="nb">echo</span> <span class="s2">&#34;SLURM_ARRAY_TASK_ID: &#34;</span> <span class="nv">$SLURM_ARRAY_TASK_ID</span>
</span></span><span class="line"><span class="cl"><span class="nv">sample</span><span class="o">=</span><span class="k">$(</span>head -n <span class="nv">$SLURM_ARRAY_TASK_ID</span> ~/work/asthma/data/namelist <span class="p">|</span> tail -1<span class="k">)</span>
</span></span><span class="line"><span class="cl"><span class="c1">#sample=$(head -n 1 namelist | tail -1)</span>
</span></span><span class="line"><span class="cl"><span class="nb">echo</span> handling: <span class="nv">$sample</span>
</span></span><span class="line"><span class="cl"><span class="c1">####################</span>
</span></span><span class="line"><span class="cl">
</span></span><span class="line"><span class="cl">fastp -w <span class="m">8</span> -i ~/work/asthma/data/<span class="nv">$sample</span>/<span class="nv">$sample</span><span class="s1">&#39;_f1.fastq&#39;</span> -o <span class="si">${</span><span class="nv">sample</span><span class="si">}</span>_1 <span class="se">\
</span></span></span><span class="line"><span class="cl"><span class="se"></span>-I ~/work/asthma/data/<span class="nv">$sample</span>/<span class="nv">$sample</span><span class="s1">&#39;_r2.fastq&#39;</span> -O <span class="si">${</span><span class="nv">sample</span><span class="si">}</span>_2 -j ~/work/asthma/fastp/<span class="si">${</span><span class="nv">i</span><span class="si">}</span>.json
</span></span><span class="line"><span class="cl">
</span></span><span class="line"><span class="cl"><span class="c1">####################</span>
</span></span><span class="line"><span class="cl"><span class="nb">echo</span> end: <span class="sb">`</span>date +<span class="s1">&#39;%Y-%m-%d %T&#39;</span><span class="sb">`</span>
</span></span><span class="line"><span class="cl"><span class="nv">end</span><span class="o">=</span><span class="sb">`</span>date +%s<span class="sb">`</span>
</span></span><span class="line"><span class="cl"><span class="nb">echo</span> TIME:<span class="sb">`</span>expr <span class="nv">$end</span> - <span class="nv">$start</span><span class="sb">`</span>s
</span></span></code></pre></td></tr></table>
</div>
</div><p>åé¢æ¥ä¸€ä¸ªpythonè„šæœ¬å°±å¯ä»¥ç»Ÿè®¡å¸¸ç”¨æŒ‡æ ‡äº†ã€‚</p>
<p>æŠŠæ‰€æœ‰çš„.jsonæ–‡ä»¶ç§»åˆ°ä¸€ä¸ªæ–‡ä»¶å¤¹é‡Œï¼Œreport/ä¸‹ï¼Œå°±å¯ä»¥ç»Ÿè®¡äº†ã€‚</p>
<h3 id="å»å®¿ä¸»bowtie2">å»å®¿ä¸»ï¼šbowtie2</h3>
<p>å…¶å®å°±æ˜¯å°†åºåˆ—æ¯”å¯¹åˆ°äººç±»åŸºå› ç»„ä¸Šï¼Œæ²¡æœ‰æ¯”å¯¹åˆ°çš„åºåˆ—æ•´åˆæˆæ–°æ–‡ä»¶å°±æ˜¯å»å®¿ä¸»åçš„äº†ã€‚</p>
<div class="highlight"><div class="chroma">
<table class="lntable"><tr><td class="lntd">
<pre tabindex="0" class="chroma"><code><span class="lnt"> 1
</span><span class="lnt"> 2
</span><span class="lnt"> 3
</span><span class="lnt"> 4
</span><span class="lnt"> 5
</span><span class="lnt"> 6
</span><span class="lnt"> 7
</span><span class="lnt"> 8
</span><span class="lnt"> 9
</span><span class="lnt">10
</span><span class="lnt">11
</span><span class="lnt">12
</span><span class="lnt">13
</span><span class="lnt">14
</span><span class="lnt">15
</span><span class="lnt">16
</span><span class="lnt">17
</span><span class="lnt">18
</span><span class="lnt">19
</span></code></pre></td>
<td class="lntd">
<pre tabindex="0" class="chroma"><code class="language-bash" data-lang="bash"><span class="line"><span class="cl"><span class="cp">#!/bin/bash
</span></span></span><span class="line"><span class="cl"><span class="cp"></span><span class="c1">#SBATCH --job-name=rm_human</span>
</span></span><span class="line"><span class="cl"><span class="c1">#SBATCH --output=/share/home/jianglab/pengchen/work/meta/%x_%a.out</span>
</span></span><span class="line"><span class="cl"><span class="c1">#SBATCH --error=/share/home/jianglab/pengchen/work/meta/%x_%a.err</span>
</span></span><span class="line"><span class="cl"><span class="c1">#SBATCH --cpus-per-task=32</span>
</span></span><span class="line"><span class="cl"><span class="c1">#SBATCH --partition=short</span>
</span></span><span class="line"><span class="cl">
</span></span><span class="line"><span class="cl"><span class="nb">echo</span> start: <span class="sb">`</span>date +<span class="s1">&#39;%Y-%m-%d %T&#39;</span><span class="sb">`</span>
</span></span><span class="line"><span class="cl"><span class="nv">start</span><span class="o">=</span><span class="sb">`</span>date +%s<span class="sb">`</span>
</span></span><span class="line"><span class="cl"><span class="c1">#############</span>
</span></span><span class="line"><span class="cl"><span class="k">for</span> i in C1 C2
</span></span><span class="line"><span class="cl"><span class="k">do</span>
</span></span><span class="line"><span class="cl">bowtie2 -p <span class="m">32</span> -x ~/db/humangenome/hg38 -1 seq/<span class="si">${</span><span class="nv">i</span><span class="si">}</span>_1.fq.gz <span class="se">\
</span></span></span><span class="line"><span class="cl"><span class="se"></span> -2 seq/<span class="si">${</span><span class="nv">i</span><span class="si">}</span>_2.fq.gz -S <span class="si">${</span><span class="nv">i</span><span class="si">}</span>.sam --un-conc <span class="si">${</span><span class="nv">i</span><span class="si">}</span>.fq --very-sensitive
</span></span><span class="line"><span class="cl"><span class="k">done</span>
</span></span><span class="line"><span class="cl"><span class="c1">##############</span>
</span></span><span class="line"><span class="cl"><span class="nb">echo</span> end: <span class="sb">`</span>date +<span class="s1">&#39;%Y-%m-%d %T&#39;</span><span class="sb">`</span>
</span></span><span class="line"><span class="cl"><span class="nv">end</span><span class="o">=</span><span class="sb">`</span>date +%s<span class="sb">`</span>
</span></span><span class="line"><span class="cl"><span class="nb">echo</span> TIME:<span class="sb">`</span>expr <span class="nv">$end</span> - <span class="nv">$start</span><span class="sb">`</span>s
</span></span></code></pre></td></tr></table>
</div>
</div><h3 id="åŸºæœ¬ä¿¡æ¯ç»Ÿè®¡">åŸºæœ¬ä¿¡æ¯ç»Ÿè®¡</h3>
<p>å¯ä»¥ç”¨FastqCountï¼š</p>
<div class="highlight"><div class="chroma">
<table class="lntable"><tr><td class="lntd">
<pre tabindex="0" class="chroma"><code><span class="lnt">1
</span><span class="lnt">2
</span><span class="lnt">3
</span><span class="lnt">4
</span></code></pre></td>
<td class="lntd">
<pre tabindex="0" class="chroma"><code class="language-bash" data-lang="bash"><span class="line"><span class="cl">~/biosoft/FastqCount-master/FastqCount_v0.5 xx.fastq.gz
</span></span><span class="line"><span class="cl">
</span></span><span class="line"><span class="cl">Total Reads     Total Bases     N Bases Q20     Q30     GC
</span></span><span class="line"><span class="cl"><span class="m">11568822</span> <span class="o">(</span>11.57 M<span class="o">)</span>      <span class="m">1702829127</span> <span class="o">(</span>1.70 G<span class="o">)</span>     0.00%   98.00%  94.00%  54.00%
</span></span></code></pre></td></tr></table>
</div>
</div><h2 id="reads-based">reads-based</h2>
<h3 id="ç‰©ç§æ³¨é‡Škraken2">ç‰©ç§æ³¨é‡Šï¼škraken2</h3>
<p>Kraken 2æ˜¯ä¸€ä¸ªç”¨äºå¯¹é«˜é€šé‡æµ‹åºæ•°æ®è¿›è¡Œåˆ†ç±»å’Œæ ‡è¯†ç‰©ç§çš„è½¯ä»¶ã€‚å®ƒä½¿ç”¨å‚è€ƒæ•°æ®åº“ä¸­çš„åŸºå› ç»„åºåˆ—æ¥è¿›è¡Œåˆ†ç±»ï¼Œå¹¶ä½¿ç”¨k-meræ–¹æ³•æ¥å®ç°å¿«é€Ÿå’Œå‡†ç¡®çš„åˆ†ç±»ã€‚</p>
<p>ä½¿ç”¨Kraken 2è¿›è¡ŒåŸºæœ¬åˆ†ç±»çš„ç®€å•æ­¥éª¤ï¼š</p>
<ol>
<li>
<p>å‡†å¤‡å‚è€ƒæ•°æ®åº“ï¼šKraken 2éœ€è¦ä¸€ä¸ªå‚è€ƒæ•°æ®åº“ï¼Œä»¥ä¾¿å¯¹æµ‹åºæ•°æ®è¿›è¡Œåˆ†ç±»ã€‚å¯ä»¥ä»NCBIã€Ensemblæˆ–å…¶ä»–æ•°æ®åº“ä¸‹è½½ç›¸åº”çš„åŸºå› ç»„åºåˆ—ï¼Œå¹¶ä½¿ç”¨Kraken 2å†…ç½®çš„å·¥å…·æ¥æ„å»ºæ•°æ®åº“ã€‚</p>
</li>
<li>
<p>å®‰è£…Kraken 2ï¼šå¯ä»¥ä»Kraken 2å®˜æ–¹ç½‘ç«™ä¸‹è½½å¹¶å®‰è£…Kraken 2è½¯ä»¶ã€‚</p>
</li>
<li>
<p>è¿è¡ŒKraken 2ï¼šä½¿ç”¨Kraken 2å¯¹æµ‹åºæ•°æ®è¿›è¡Œåˆ†ç±»éœ€è¦ä½¿ç”¨ä»¥ä¸‹å‘½ä»¤ï¼š</p>
</li>
</ol>
<p><code>kraken2 \--db \&lt;path_to_database\&gt; \&lt;input_file\&gt; \--output \&lt;output_file\&gt;</code></p>
<p>è¿™é‡Œï¼Œ**<code>&lt;path_to_database&gt;</code><strong>æ˜¯å‚è€ƒæ•°æ®åº“çš„è·¯å¾„ï¼Œ</strong><code>&lt;input_file&gt;</code><strong>æ˜¯éœ€è¦è¿›è¡Œåˆ†ç±»çš„è¾“å…¥æ–‡ä»¶ï¼Œ</strong><code>&lt;output_file&gt;</code>**æ˜¯è¾“å‡ºæ–‡ä»¶çš„åç§°ã€‚Kraken 2å°†è¾“å‡ºä¸€ä¸ªåˆ†ç±»æŠ¥å‘Šæ–‡ä»¶å’Œä¸€ä¸ªåºåˆ—æ–‡ä»¶ã€‚</p>
<p><code>kraken2-build --standard --threads 24 --db ./</code></p>
<p>&ndash;standardæ ‡å‡†æ¨¡å¼ä¸‹åªä¸‹è½½5ç§æ•°æ®åº“ï¼šå¤èŒarchaeaã€ç»†èŒbacteriaã€äººç±»humanã€è½½ä½“UniVec_Coreã€ç—…æ¯’viralã€‚ä¹Ÿå¯é€‰ç›´æ¥ä¸‹è½½ä½œè€…æ„å»ºçš„ç´¢å¼•ï¼Œè¿˜åŒ…æ‹¬brackençš„ç´¢å¼•ã€‚</p>
<p>è¿™ä¸ªkrakenæ•°æ®åº“æ˜¯å¯ä»¥è‡ªå·±æ„å»ºçš„ï¼Œæ‰€ä»¥é€‚ç”¨äºå„ç§é¡¹ç›®çš„ç‰©ç§æ³¨é‡Šï¼Œæˆ‘åšçš„æ¯”è¾ƒå¤šçš„æ˜¯ç¯å¢ƒæ ·æœ¬çš„å®åŸºå› ç»„ï¼Œå°±å¯èƒ½éœ€è¦æ›´å…¨é¢çš„ç‰©ç§æ•°æ®åº“ï¼ˆç”šè‡³é™¤äº†å„ç§å¾®ç”Ÿç‰©ï¼Œè¿˜è¦åŠ¨æ¤ç‰©æ•°æ®ç­‰ï¼‰ï¼Œå®éªŒå®¤çš„WXå¸ˆå§æ”¶é›†æ„å»ºäº†ä¸€ä¸ªè¶…å¤§çš„ç‰©ç§åº“ã€‚</p>
<p>éœ€è¦æ³¨æ„çš„æ˜¯krakenè¿è¡Œè‡³å°‘è¦æä¾›æ•°æ®åº“å¤§å°çš„å†…å­˜å¤§å°ï¼ˆè¿è¡Œå†…å­˜ï¼‰ï¼Œå› ä¸ºå®ƒä¼šæŠŠæ•´ä¸ªæ•°æ®åº“è½½å…¥å†…å­˜åè¿›è¡Œåºåˆ—çš„æ³¨é‡Šï¼Œæ‰€ä»¥å¦‚æœå‘ç°æ— æ³•è½½å…¥æ•°æ®åº“çš„æŠ¥é”™ï¼Œå¯ä»¥å°è¯•è°ƒå¤§å†…å­˜èµ„æºã€‚</p>
<p>krakenè½¯ä»¶è¿è¡Œæ—¶è½½å…¥æ•°æ®åº“æ˜¯ä¸€ä¸ªååˆ†è€—æ—¶çš„æ­¥éª¤ï¼Œè€Œæ¯æ¡åºåˆ—çš„é‰´å®šæ—¶é—´å·®ä¸å¤šï¼Œæ‰€ä»¥æˆ‘ä»¬å¯ä»¥å°†å¾ˆå¤šæ ·æœ¬çš„fastqæ–‡ä»¶åˆå¹¶æˆä¸€ä¸ªå¤§æ–‡ä»¶åè¾“å…¥krakenæ³¨é‡Šï¼Œä¹‹åå†æŒ‰ç…§åºåˆ—çš„æ•°é‡æ‹†åˆ†ç»“æœæ–‡ä»¶ï¼Œè¿™æ ·å¤šä¸ªæ ·æœ¬ä¹Ÿåªéœ€è¦è½½å…¥ä¸€æ¬¡æ•°æ®åº“ï¼ŒèŠ‚çœæ—¶é—´ã€‚</p>
<div class="highlight"><div class="chroma">
<table class="lntable"><tr><td class="lntd">
<pre tabindex="0" class="chroma"><code><span class="lnt"> 1
</span><span class="lnt"> 2
</span><span class="lnt"> 3
</span><span class="lnt"> 4
</span><span class="lnt"> 5
</span><span class="lnt"> 6
</span><span class="lnt"> 7
</span><span class="lnt"> 8
</span><span class="lnt"> 9
</span><span class="lnt">10
</span><span class="lnt">11
</span><span class="lnt">12
</span><span class="lnt">13
</span><span class="lnt">14
</span><span class="lnt">15
</span><span class="lnt">16
</span><span class="lnt">17
</span><span class="lnt">18
</span></code></pre></td>
<td class="lntd">
<pre tabindex="0" class="chroma"><code class="language-bash" data-lang="bash"><span class="line"><span class="cl"><span class="cp">#!/bin/bash
</span></span></span><span class="line"><span class="cl"><span class="cp"></span><span class="c1">#SBATCH --job-name=kraken2M</span>
</span></span><span class="line"><span class="cl"><span class="c1">#SBATCH --output=/share/home/jianglab/pengchen/work/asthma/kraken/%x_%a.out</span>
</span></span><span class="line"><span class="cl"><span class="c1">#SBATCH --error=/share/home/jianglab/pengchen/work/asthma/kraken/%x_%a.err</span>
</span></span><span class="line"><span class="cl"><span class="c1">#SBATCH --time=14-00:00:00</span>
</span></span><span class="line"><span class="cl"><span class="c1">#SBATCH --partition=mem</span>
</span></span><span class="line"><span class="cl"><span class="c1">#SBATCH --cpus-per-task=32</span>
</span></span><span class="line"><span class="cl"><span class="c1">#SBATCH --mem-per-cpu=100G</span>
</span></span><span class="line"><span class="cl">
</span></span><span class="line"><span class="cl"><span class="nv">fqp</span><span class="o">=</span>~/work/asthma/data/CRR205159/
</span></span><span class="line"><span class="cl">python /share/home/jianglab/shared/krakenDB/K2ols/kraken2M.py -t <span class="m">32</span> <span class="se">\
</span></span></span><span class="line"><span class="cl"><span class="se"></span>    -i <span class="si">${</span><span class="nv">fqp</span><span class="si">}</span> <span class="se">\
</span></span></span><span class="line"><span class="cl"><span class="se"></span>    -c 0.05 <span class="se">\
</span></span></span><span class="line"><span class="cl"><span class="se"></span>    -s _f1.fastq,_r2.fastq <span class="se">\
</span></span></span><span class="line"><span class="cl"><span class="se"></span>    -o ~/work/asthma/kraken/ <span class="se">\
</span></span></span><span class="line"><span class="cl"><span class="se"></span>    -d /share/home/jianglab/shared/krakenDB/mydb2 <span class="se">\
</span></span></span><span class="line"><span class="cl"><span class="se"></span>    -k ~/miniconda3/envs/waste/bin/kraken2 <span class="se">\
</span></span></span><span class="line"><span class="cl"><span class="se"></span>    -kt /share/home/jianglab/shared/krakenDB/K2ols/KrakenTools 
</span></span></code></pre></td></tr></table>
</div>
</div><h4 id="è¾“å‡ºæ–‡ä»¶æ ¼å¼">è¾“å‡ºæ–‡ä»¶æ ¼å¼</h4>
<p>Krakenæ ‡å‡†è¾“å‡ºæ ¼å¼</p>
<p>äº”åˆ—è¡¨ output</p>
<ul>
<li>
<p>C/Uä»£è¡¨åˆ†ç±»classifiedæˆ–éåˆ†ç±»unclassifed</p>
</li>
<li>
<p>åºåˆ—ID</p>
</li>
<li>
<p>ç‰©ç§æ³¨é‡Š</p>
</li>
<li>
<p>æ¯”åºåˆ—æ³¨é‡Šçš„åŒºåŸŸï¼Œå¦‚98|94ä»£è¡¨å·¦ç«¯98bpï¼Œå³ç«¯94bpæ¯”å¯¹è‡³æ•°æ®åº“</p>
</li>
<li>
<p>LCAæ¯”å¯¹ç»“æœï¼Œå¦‚â€562:13 561:4â€ä»£è¡¨13 k-meræ¯”å¯¹è‡³ç‰©ç§#562ï¼Œ4 k-meræ¯”å¯¹è‡³#561ç‰©ç§</p>
</li>
</ul>
<p>æŠ¥å‘Šè¾“å‡ºæ ¼å¼ report</p>
<p>åŒ…æ‹¬6åˆ—ï¼Œæ–¹ä¾¿æ•´ç†ä¸‹æ¸¸åˆ†æã€‚</p>
<ol>
<li>
<p>ç™¾åˆ†æ¯”</p>
</li>
<li>
<p>count</p>
</li>
<li>
<p>countæœ€ä¼˜</p>
</li>
<li>
<p>(U)nclassified, (R)oot, (D)omain, (K)ingdom, (P)hylum, (C)lass, (O)rder, (F)amily, (G)enus, or (S)pecies. â€œG2â€ä»£è¡¨ä½äºå±ä¸€ç§é—´</p>
</li>
<li>
<p>NCBIç‰©ç§ID</p>
</li>
<li>
<p>ç§‘å­¦ç‰©ç§å</p>
</li>
</ol>
<p>å¸¸ç”¨çš„ç‰©ç§ä¸°åº¦è¡¨æ ¼å¼é™¤äº†kraken reportï¼Œè¿˜æœ‰mpaï¼Œspfï¼Œkronaç­‰æ ¼å¼ï¼Œå…³äºkrakenç»“æœçš„æ•´ç†ä»¥åŠæ ¼å¼è½¬æ¢æ–¹å¼ï¼Œæœ‰ä¸€äº›ç°æˆçš„è„šæœ¬æˆ–è€…è‡ªå·±å†™ã€‚</p>
<p><a class="link" href="https://ccb.jhu.edu/software/krakentools/"  target="_blank" rel="noopener"
    >KrakenTools (jhu.edu)</a> å°±æ˜¯ä¸€å¥—å¾ˆå¥½ç”¨çš„krakenå·¥å…·åŒ…ï¼Œå…¶ä¸­å¸¸ç”¨çš„æœ‰ï¼š</p>
<ol>
<li>extract_kraken_reads.py</li>
</ol>
<p>æ­¤ç¨‹åºæå–è¯»å–åœ¨ä»»ä½•ç”¨æˆ·æŒ‡å®šçš„åˆ†ç±»idå¤„åˆ†ç±»çš„å†…å®¹ã€‚ç”¨æˆ·å¿…é¡»æŒ‡å®šKrakenè¾“å‡ºæ–‡ä»¶ã€åºåˆ—æ–‡ä»¶å’Œè‡³å°‘ä¸€ä¸ªåˆ†ç±»æ³•IDã€‚ä¸‹é¢æŒ‡å®šäº†å…¶ä»–é€‰é¡¹ã€‚æˆªè‡³2021å¹´4æœˆ19æ—¥ï¼Œæ­¤è„šæœ¬ä¸KrakenUniq/Kraken2UniqæŠ¥å‘Šå…¼å®¹ã€‚</p>
<ol start="2">
<li>combine_kreports.py</li>
</ol>
<p>This script combines multiple Kraken reports into a combined report file.</p>
<p><code>python combine_kreports.py</code></p>
<ul>
<li>
<p>-r 1.KREPORT 2.KREPORT&hellip;&hellip;&hellip;&hellip;&hellip;&hellip;&hellip;&hellip;Kraken-style reports to combine</p>
</li>
<li>
<p>-o COMBINED.KREPORT&hellip;&hellip;&hellip;&hellip;&hellip;&hellip;&hellip;&hellip;&hellip;Output file</p>
</li>
</ul>
<ol start="3">
<li>kreport2krona.py</li>
</ol>
<p>This program takes a Kraken report file and prints out a krona-compatible TEXT file</p>
<p>æ¢æˆkronaæ–‡ä»¶å¥½ç”»å›¾ã€‚å˜¿å˜¿</p>
<p>kronaè£…äº†ä¸€ä¸ªexcelçš„æ’ä»¶å¯ä»¥å¾ˆå®¹æ˜“ç”»å›¾<br>
<code>python kreport2krona.py</code></p>
<ul>
<li>
<p>-r/&ndash;report MYFILE.KREPORT&hellip;.â€¦.Kraken report file</p>
</li>
<li>
<p>-o/&ndash;output MYFILE.KRONA&hellip;&hellip;.â€¦Output Krona text file</p>
</li>
</ul>
<p>then, <code>ktImportText MYSAMPLE.krona -o MYSAMPLE.krona.html</code></p>
<p>å¥½çœ‹çš„ç½‘é¡µå°±å‡ºæ¥äº†ã€‚</p>
<ol start="4">
<li>kreport2mpa.py</li>
</ol>
<p>This program takes a Kraken report file and prints out a mpa (MetaPhlAn) -style TEXT file</p>
<p><code>python kreport2mpa.py</code></p>
<ul>
<li>
<p>-r/&ndash;report MYFILE.KREPORT&hellip;.â€¦.Kraken report file</p>
</li>
<li>
<p>-o/&ndash;output MYFILE.MPA.TXT&hellip;.â€¦.Output MPA-STYLE text file</p>
</li>
</ul>
<ol start="5">
<li>combine_mpa.py</li>
</ol>
<p><code>python combine_mpa.py</code></p>
<ul>
<li>
<p>-i/&ndash;input MYFILE1.MPA MYFILE2.MPA&hellip;.â€¦Multiple MPA-STYLE text files (separated by spaces)</p>
</li>
<li>
<p>-o/&ndash;output MYFILE.COMBINED.MPA&hellip;&hellip;.â€¦Output MPA-STYLE text file</p>
</li>
</ul>
<p>This program combines multiple outputs from <a class="link" href="#kreport2mpapy" >kreport2mpa.py</a>. Files to be combined must have been generated using the same kreport2mpa.py options.</p>
<p><code>python combine_mpa.py -i -o --intermediate-ranks</code></p>
<h3 id="humann">HUMAnN</h3>
<p>HUMAnN2ï¼ˆThe HMP Unified Metabolic Analysis Network 2ï¼‰æ˜¯ä¸€æ¬¾ç”¨äºåˆ†æäººç±»å¾®ç”Ÿç‰©ç»„çš„åŠŸèƒ½å’Œä»£è°¢èƒ½åŠ›çš„å·¥å…·ã€‚å®ƒé€šè¿‡å°†å®åŸºå› ç»„åºåˆ—ä¸å‚è€ƒåŸºå› ç»„æ•°æ®åº“æ¯”å¯¹ï¼Œåˆ©ç”¨MetaCycä»£è°¢é€šè·¯æ•°æ®åº“å’ŒUniRefè›‹ç™½è´¨åºåˆ—æ•°æ®åº“ï¼Œåˆ†æå¾®ç”Ÿç‰©ç»„åœ¨åŠŸèƒ½å’Œä»£è°¢é€šè·¯æ°´å¹³ä¸Šçš„ç»„æˆå’Œæ´»æ€§ã€‚HUMAnN2è¿˜æä¾›äº†å¤šæ ·æ€§åˆ†æã€å…³è”åˆ†æå’Œå¯è§†åŒ–å·¥å…·ï¼Œå¯ç”¨äºæ·±å…¥ç ”ç©¶äººç±»å¾®ç”Ÿç‰©ç»„å¯¹å®¿ä¸»å¥åº·çš„å½±å“å’Œæ²»ç–—ç­–ç•¥çš„åˆ¶å®šç­‰æ–¹é¢ã€‚</p>
<p>HUMAnN2æ˜¯ç”±ç¾å›½å›½å®¶äººç±»å¾®ç”Ÿç‰©ç»„è®¡åˆ’ï¼ˆHMPï¼‰å¼€å‘çš„ï¼Œç›®å‰æœ€æ–°ç‰ˆæœ¬ä¸º<a class="link" href="https://github.com/biobakery/humann"  target="_blank" rel="noopener"
    >HUMAnN3</a>ï¼Œäº2020å¹´å‘å¸ƒã€‚ä¸HUMAnN2ç›¸æ¯”ï¼ŒHUMAnN3æ”¹è¿›äº†åŸºå› å®¶æ—æ³¨é‡Šçš„æ–¹æ³•ï¼Œæé«˜äº†æ³¨é‡Šç²¾åº¦å’Œé€Ÿåº¦ï¼Œå¹¶æä¾›äº†æ–°çš„åŠŸèƒ½å’Œå·¥å…·ï¼Œå¦‚åŠŸèƒ½éŸ§åº¦åˆ†æã€ä»£è°¢æŒ‡çº¹è¯†åˆ«å’Œå¤šæ ·æ€§åˆ†æç­‰ã€‚</p>
<p>ä½†æ˜¯HUMAnN2çš„æ•°æ®åº“åŸºæœ¬éƒ½æ˜¯ä¸äººç›¸å…³çš„å¾®ç”Ÿç‰©ï¼Œæ¯”è¾ƒé€‚åˆåšå„ç§äººä½“å¾®ç”Ÿç‰©ç»„ï¼ˆè‚ é“ï¼Œè‚ºéƒ¨ï¼Œå£è…”ï¼Œçš®è‚¤ç­‰ç­‰ï¼‰ï¼Œå¯¹äºç¯å¢ƒæ ·æœ¬å¯èƒ½unclassifiedæ¯”è¾ƒå¤šã€‚</p>
<p><strong>HUMAnN2è¦æ±‚åŒç«¯åºåˆ—åˆå¹¶çš„æ–‡ä»¶ä½œä¸ºè¾“å…¥</strong>ï¼Œforå¾ªç¯æ ¹æ®å®éªŒè®¾è®¡æ ·æœ¬åæ‰¹é‡åŒç«¯åºåˆ—åˆå¹¶ã€‚</p>
<ul>
<li>
<p><strong>ç‰©ç§ç»„æˆè°ƒç”¨MetaPhlAn2, bowtie2æ¯”å¯¹è‡³æ ¸é…¸åºåˆ—</strong>ï¼Œè§£å†³æœ‰å“ªäº›å¾®ç”Ÿç‰©å­˜åœ¨çš„é—®é¢˜ï¼›</p>
</li>
<li>
<p><strong>åŠŸèƒ½ç»„æˆä¸ºhumann2è°ƒç”¨diamondæ¯”å¯¹è‡³è›‹ç™½åº“11Gb</strong>ï¼Œè§£å†³è¿™äº›å¾®ç”Ÿç‰©å‚ä¸å“ªäº›åŠŸèƒ½é€šè·¯çš„é—®é¢˜ï¼›</p>
</li>
</ul>
<div class="highlight"><div class="chroma">
<table class="lntable"><tr><td class="lntd">
<pre tabindex="0" class="chroma"><code><span class="lnt"> 1
</span><span class="lnt"> 2
</span><span class="lnt"> 3
</span><span class="lnt"> 4
</span><span class="lnt"> 5
</span><span class="lnt"> 6
</span><span class="lnt"> 7
</span><span class="lnt"> 8
</span><span class="lnt"> 9
</span><span class="lnt">10
</span><span class="lnt">11
</span><span class="lnt">12
</span><span class="lnt">13
</span><span class="lnt">14
</span><span class="lnt">15
</span><span class="lnt">16
</span><span class="lnt">17
</span><span class="lnt">18
</span><span class="lnt">19
</span><span class="lnt">20
</span><span class="lnt">21
</span><span class="lnt">22
</span><span class="lnt">23
</span><span class="lnt">24
</span><span class="lnt">25
</span><span class="lnt">26
</span><span class="lnt">27
</span><span class="lnt">28
</span><span class="lnt">29
</span><span class="lnt">30
</span><span class="lnt">31
</span><span class="lnt">32
</span><span class="lnt">33
</span><span class="lnt">34
</span><span class="lnt">35
</span><span class="lnt">36
</span><span class="lnt">37
</span><span class="lnt">38
</span><span class="lnt">39
</span><span class="lnt">40
</span><span class="lnt">41
</span></code></pre></td>
<td class="lntd">
<pre tabindex="0" class="chroma"><code class="language-bash" data-lang="bash"><span class="line"><span class="cl"><span class="nb">cd</span> alldata
</span></span><span class="line"><span class="cl"><span class="k">for</span> i in <span class="sb">`</span>cat ~/work/asthma/data/namelist<span class="sb">`</span>
</span></span><span class="line"><span class="cl"><span class="k">do</span>
</span></span><span class="line"><span class="cl"><span class="nb">echo</span> <span class="nv">$i</span>
</span></span><span class="line"><span class="cl">cat <span class="si">${</span><span class="nv">i</span><span class="si">}</span>_f1.fastq <span class="si">${</span><span class="nv">i</span><span class="si">}</span>_r2.fastq &gt;<span class="si">${</span><span class="nv">i</span><span class="si">}</span>_paired.fastq
</span></span><span class="line"><span class="cl"><span class="k">done</span>
</span></span><span class="line"><span class="cl">
</span></span><span class="line"><span class="cl"><span class="c1">#!/bin/bash</span>
</span></span><span class="line"><span class="cl"><span class="c1">#SBATCH --job-name=humann2</span>
</span></span><span class="line"><span class="cl"><span class="c1">#SBATCH --output=/share/home/jianglab/pengchen/work/asthma/humann/%x_%a.out</span>
</span></span><span class="line"><span class="cl"><span class="c1">#SBATCH --error=/share/home/jianglab/pengchen/work/asthma/humann/%x_%a.err</span>
</span></span><span class="line"><span class="cl"><span class="c1">#SBATCH --array=1-32</span>
</span></span><span class="line"><span class="cl"><span class="c1">#SBATCH --cpus-per-task=24</span>
</span></span><span class="line"><span class="cl"><span class="c1">#SBATCH --partition=cpu</span>
</span></span><span class="line"><span class="cl">
</span></span><span class="line"><span class="cl"><span class="nb">echo</span> start: <span class="sb">`</span>date +<span class="s1">&#39;%Y-%m-%d %T&#39;</span><span class="sb">`</span>
</span></span><span class="line"><span class="cl"><span class="nv">start</span><span class="o">=</span><span class="sb">`</span>date +%s<span class="sb">`</span>
</span></span><span class="line"><span class="cl"><span class="c1">##############</span>
</span></span><span class="line"><span class="cl"><span class="nv">myarray</span><span class="o">=(</span><span class="sb">`</span>cat ~/work/asthma/data/namelist<span class="sb">`</span><span class="o">)</span>
</span></span><span class="line"><span class="cl"><span class="nb">echo</span> <span class="nv">$SLURM_ARRAY_TASK_ID</span>
</span></span><span class="line"><span class="cl"><span class="c1">#this is your single file name</span>
</span></span><span class="line"><span class="cl"><span class="nv">sample</span><span class="o">=</span><span class="si">${</span><span class="nv">myarray</span><span class="p">[</span><span class="si">${</span><span class="nv">SLURM_ARRAY_TASK_ID</span><span class="si">}</span><span class="p">]</span><span class="si">}</span>
</span></span><span class="line"><span class="cl"><span class="nb">echo</span> handling: <span class="nv">$sample</span>
</span></span><span class="line"><span class="cl">humann2 --input data/alldata/<span class="si">${</span><span class="nv">sample</span><span class="si">}</span>_paired.fastq  <span class="se">\
</span></span></span><span class="line"><span class="cl"><span class="se"></span>  --output temp/humann2/ --threads <span class="m">24</span>
</span></span><span class="line"><span class="cl">  
</span></span><span class="line"><span class="cl"><span class="c1">## é“¾æ¥é‡è¦æ–‡ä»¶è‡³humann2ç›®å½•</span>
</span></span><span class="line"><span class="cl">ln temp/humann2/<span class="si">${</span><span class="nv">sample</span><span class="si">}</span>_paired_humann2_temp/<span class="si">${</span><span class="nv">sample</span><span class="si">}</span>_paired_metaphlan_bugs_list.tsv temp/humann2/
</span></span><span class="line"><span class="cl"><span class="c1">## åˆ é™¤ä¸´æ—¶æ–‡ä»¶</span>
</span></span><span class="line"><span class="cl">rm -rf temp/humann2/<span class="si">${</span><span class="nv">sample</span><span class="si">}</span>_paired_humann2_temp
</span></span><span class="line"><span class="cl"><span class="c1">##############</span>
</span></span><span class="line"><span class="cl"><span class="nb">echo</span> end: <span class="sb">`</span>date +<span class="s1">&#39;%Y-%m-%d %T&#39;</span><span class="sb">`</span>
</span></span><span class="line"><span class="cl"><span class="nv">end</span><span class="o">=</span><span class="sb">`</span>date +%s<span class="sb">`</span>
</span></span><span class="line"><span class="cl"><span class="nb">echo</span> TIME:<span class="sb">`</span>expr <span class="nv">$end</span> - <span class="nv">$start</span><span class="sb">`</span>s
</span></span><span class="line"><span class="cl">
</span></span><span class="line"><span class="cl">
</span></span><span class="line"><span class="cl"><span class="c1">## åˆå¹¶ã€ä¿®æ­£æ ·æœ¬åã€é¢„è§ˆ</span>
</span></span><span class="line"><span class="cl">merge_metaphlan_tables2.py <span class="se">\
</span></span></span><span class="line"><span class="cl"><span class="se"></span>  temp/humann2/*_metaphlan_bugs_list.tsv <span class="p">|</span> <span class="se">\
</span></span></span><span class="line"><span class="cl"><span class="se"></span>  sed <span class="s1">&#39;s/_metaphlan_bugs_list//g&#39;</span> <span class="se">\
</span></span></span><span class="line"><span class="cl"><span class="se"></span>  &gt; metaphlan2/taxonomy.tsv
</span></span></code></pre></td></tr></table>
</div>
</div><h2 id="contigs-based">contigs-based</h2>
<h3 id="ç»„è£…megahit">ç»„è£…ï¼šmegahit</h3>
<p>MegaHitæ˜¯ä¸€ä¸ªç”¨äºå¯¹é«˜é€šé‡æµ‹åºæ•°æ®è¿›è¡Œde novoç»„è£…çš„è½¯ä»¶ã€‚å®ƒä½¿ç”¨äº†ä¸€ç§åŸºäºçŸ­è¯»æ¯”å¯¹å’Œå›¾å½¢æ„å»ºçš„ç®—æ³•æ¥ç»„è£…åŸºå› ç»„ï¼Œå¯ä»¥é«˜æ•ˆåœ°å¤„ç†å¤§è§„æ¨¡çš„æ•°æ®é›†ã€‚ä»¥ä¸‹æ˜¯MegaHitçš„ä¸€äº›ä¼˜ç‚¹å’Œé€‚ç”¨æƒ…å†µï¼š</p>
<ol>
<li>
<p>é€Ÿåº¦å¿«ï¼šMegaHitçš„ç®—æ³•éå¸¸é«˜æ•ˆï¼Œå¯ä»¥å¤„ç†å¤§è§„æ¨¡çš„æ•°æ®é›†ï¼Œé€šå¸¸æ¯”å…¶ä»–de novoç»„è£…å·¥å…·æ›´å¿«ã€‚</p>
</li>
<li>
<p>é«˜è´¨é‡çš„ç»„è£…ï¼šMegaHitåœ¨ç»„è£…ç»“æœçš„è¿é€šæ€§å’Œå‡†ç¡®æ€§æ–¹é¢è¡¨ç°ä¼˜å¼‚ï¼Œå°¤å…¶åœ¨å¤„ç†é«˜GCå«é‡åŸºå› ç»„æ—¶æ•ˆæœæ˜¾è‘—ã€‚</p>
</li>
<li>
<p>é€‚ç”¨äºä¸åŒç±»å‹çš„æµ‹åºæ•°æ®ï¼šMegaHitæ”¯æŒå¤šç§ä¸åŒç±»å‹çš„æµ‹åºæ•°æ®ï¼ŒåŒ…æ‹¬ Illumina HiSeq/MiSeqã€IonTorrentå’ŒPacBioç­‰å¹³å°ã€‚</p>
</li>
<li>
<p>æ˜“äºä½¿ç”¨ï¼šMegaHitå…·æœ‰ç®€å•çš„å‘½ä»¤è¡Œè¯­æ³•ï¼Œæ–¹ä¾¿ç”¨æˆ·è¿›è¡Œç»„è£…æ“ä½œï¼Œä¸”å…·æœ‰ä¸­æ–­ç‚¹ï¼Œé¿å…å¤±è´¥åå…¨éƒ¨é‡è·‘ã€‚</p>
</li>
</ol>
<div class="highlight"><div class="chroma">
<table class="lntable"><tr><td class="lntd">
<pre tabindex="0" class="chroma"><code><span class="lnt"> 1
</span><span class="lnt"> 2
</span><span class="lnt"> 3
</span><span class="lnt"> 4
</span><span class="lnt"> 5
</span><span class="lnt"> 6
</span><span class="lnt"> 7
</span><span class="lnt"> 8
</span><span class="lnt"> 9
</span><span class="lnt">10
</span><span class="lnt">11
</span><span class="lnt">12
</span><span class="lnt">13
</span><span class="lnt">14
</span><span class="lnt">15
</span><span class="lnt">16
</span><span class="lnt">17
</span><span class="lnt">18
</span><span class="lnt">19
</span><span class="lnt">20
</span><span class="lnt">21
</span></code></pre></td>
<td class="lntd">
<pre tabindex="0" class="chroma"><code class="language-bash" data-lang="bash"><span class="line"><span class="cl"><span class="cp">#!/bin/bash
</span></span></span><span class="line"><span class="cl"><span class="cp"></span><span class="c1">#SBATCH --job-name=asthma_megahit</span>
</span></span><span class="line"><span class="cl"><span class="c1">#SBATCH --output=/share/home/jianglab/pengchen/work/asthma/megahit/log/%x_%a.out</span>
</span></span><span class="line"><span class="cl"><span class="c1">#SBATCH --error=/share/home/jianglab/pengchen/work/asthma/megahit/log/%x_%a.err</span>
</span></span><span class="line"><span class="cl"><span class="c1">#SBATCH --array=1-33</span>
</span></span><span class="line"><span class="cl"><span class="c1">#SBATCH --partition=cpu</span>
</span></span><span class="line"><span class="cl"><span class="c1">#SBATCH --cpus-per-task=32</span>
</span></span><span class="line"><span class="cl">
</span></span><span class="line"><span class="cl"><span class="nb">echo</span> start: <span class="sb">`</span>date +<span class="s1">&#39;%Y-%m-%d %T&#39;</span><span class="sb">`</span>
</span></span><span class="line"><span class="cl"><span class="nv">start</span><span class="o">=</span><span class="sb">`</span>date +%s<span class="sb">`</span>
</span></span><span class="line"><span class="cl"><span class="nb">echo</span> <span class="s2">&#34;SLURM_ARRAY_TASK_ID: &#34;</span> <span class="nv">$SLURM_ARRAY_TASK_ID</span>
</span></span><span class="line"><span class="cl"><span class="nv">sample</span><span class="o">=</span><span class="k">$(</span>head -n <span class="nv">$SLURM_ARRAY_TASK_ID</span> ~/work/asthma/data/namelist <span class="p">|</span> tail -1<span class="k">)</span>
</span></span><span class="line"><span class="cl"><span class="c1">#sample=$(head -n 1 namelist | tail -1)</span>
</span></span><span class="line"><span class="cl"><span class="nb">echo</span> handling: <span class="nv">$sample</span>	
</span></span><span class="line"><span class="cl"><span class="c1">####################</span>
</span></span><span class="line"><span class="cl">megahit -t <span class="m">32</span> -1 ~/work/asthma/data/<span class="nv">$sample</span>/<span class="nv">$sample</span><span class="s1">&#39;_f1.fastq&#39;</span> <span class="se">\
</span></span></span><span class="line"><span class="cl"><span class="se"></span>-2 ~/work/asthma/data/<span class="nv">$sample</span>/<span class="nv">$sample</span><span class="s1">&#39;_r2.fastq&#39;</span> -o ~/work/asthma/megahit/<span class="nv">$sample</span> --out-prefix <span class="nv">$sample</span>
</span></span><span class="line"><span class="cl"><span class="c1">####################</span>
</span></span><span class="line"><span class="cl"><span class="nb">echo</span> end: <span class="sb">`</span>date +<span class="s1">&#39;%Y-%m-%d %T&#39;</span><span class="sb">`</span>
</span></span><span class="line"><span class="cl"><span class="nv">end</span><span class="o">=</span><span class="sb">`</span>date +%s<span class="sb">`</span>
</span></span><span class="line"><span class="cl"><span class="nb">echo</span> TIME:<span class="sb">`</span>expr <span class="nv">$end</span> - <span class="nv">$start</span><span class="sb">`</span>s
</span></span></code></pre></td></tr></table>
</div>
</div><h3 id="ç»„è£…è¯„ä¼°quast">ç»„è£…è¯„ä¼°ï¼šQUAST</h3>
<p><strong>QUAST</strong>ä»£è¡¨è´¨é‡è¯„ä¼°å·¥å…·ã€‚ QUASTå¯ä»¥ä½¿ç”¨å‚è€ƒåŸºå› ç»„ä»¥åŠä¸ä½¿ç”¨å‚è€ƒåŸºå› ç»„æ¥è¯„ä¼°è£…é…ã€‚ QUASTç”Ÿæˆè¯¦ç»†çš„æŠ¥å‘Šï¼Œè¡¨æ ¼å’Œå›¾è§£ï¼Œä»¥æ˜¾ç¤ºè£…é…çš„ä¸åŒæ–¹é¢ã€‚</p>
<h3 id="åŸºå› é¢„æµ‹prodigal">åŸºå› é¢„æµ‹ï¼šProdigal</h3>
<p>è¾“å…¥æ–‡ä»¶ï¼šæ‹¼è£…å¥½çš„åºåˆ—æ–‡ä»¶ megahit/final.contigs.fa</p>
<p>è¾“å‡ºæ–‡ä»¶ï¼šprodigalé¢„æµ‹çš„åŸºå› åºåˆ— prodigal/gene.fa</p>
<p>prodigalä¸æ”¯æŒå¤šçº¿ç¨‹è¿è¡Œï¼Œæ‰€ä»¥æˆ‘ä»¬å¯ä»¥è‡ªè¡Œåˆ†å‰²åºåˆ—æ–‡ä»¶è°ƒç”¨å¤šä¸ªprodigalç¨‹åºåˆ†åˆ«è·‘å®ç°ä¼ªå¤šçº¿ç¨‹ã€‚</p>
<div class="highlight"><div class="chroma">
<table class="lntable"><tr><td class="lntd">
<pre tabindex="0" class="chroma"><code><span class="lnt"> 1
</span><span class="lnt"> 2
</span><span class="lnt"> 3
</span><span class="lnt"> 4
</span><span class="lnt"> 5
</span><span class="lnt"> 6
</span><span class="lnt"> 7
</span><span class="lnt"> 8
</span><span class="lnt"> 9
</span><span class="lnt">10
</span><span class="lnt">11
</span><span class="lnt">12
</span><span class="lnt">13
</span><span class="lnt">14
</span><span class="lnt">15
</span><span class="lnt">16
</span><span class="lnt">17
</span><span class="lnt">18
</span><span class="lnt">19
</span><span class="lnt">20
</span><span class="lnt">21
</span><span class="lnt">22
</span><span class="lnt">23
</span><span class="lnt">24
</span><span class="lnt">25
</span><span class="lnt">26
</span><span class="lnt">27
</span><span class="lnt">28
</span><span class="lnt">29
</span><span class="lnt">30
</span><span class="lnt">31
</span><span class="lnt">32
</span><span class="lnt">33
</span><span class="lnt">34
</span><span class="lnt">35
</span><span class="lnt">36
</span><span class="lnt">37
</span></code></pre></td>
<td class="lntd">
<pre tabindex="0" class="chroma"><code class="language-bash" data-lang="bash"><span class="line"><span class="cl"><span class="cp">#!/bin/bash
</span></span></span><span class="line"><span class="cl"><span class="cp"></span><span class="c1">#SBATCH --job-name=prodigal</span>
</span></span><span class="line"><span class="cl"><span class="c1">#SBATCH --output=/share/home/jianglab/pengchen/work/asthma/prodigal/log/%x_%a.out</span>
</span></span><span class="line"><span class="cl"><span class="c1">#SBATCH --error=/share/home/jianglab/pengchen/work/asthma/prodigal/log/%x_%a.err</span>
</span></span><span class="line"><span class="cl"><span class="c1">#SBATCH --array=1-33</span>
</span></span><span class="line"><span class="cl"><span class="c1">#SBATCH --partition=cpu</span>
</span></span><span class="line"><span class="cl"><span class="c1">#SBATCH --cpus-per-task=1</span>
</span></span><span class="line"><span class="cl">
</span></span><span class="line"><span class="cl"><span class="nb">echo</span> start: <span class="sb">`</span>date +<span class="s1">&#39;%Y-%m-%d %T&#39;</span><span class="sb">`</span>
</span></span><span class="line"><span class="cl"><span class="nv">start</span><span class="o">=</span><span class="sb">`</span>date +%s<span class="sb">`</span>
</span></span><span class="line"><span class="cl"><span class="nb">echo</span> <span class="s2">&#34;SLURM_ARRAY_TASK_ID: &#34;</span> <span class="nv">$SLURM_ARRAY_TASK_ID</span>
</span></span><span class="line"><span class="cl"><span class="nv">sample</span><span class="o">=</span><span class="k">$(</span>head -n <span class="nv">$SLURM_ARRAY_TASK_ID</span> ~/work/asthma/data/namelist <span class="p">|</span> tail -1<span class="k">)</span>
</span></span><span class="line"><span class="cl"><span class="c1">#sample=$(head -n 1 namelist | tail -1)</span>
</span></span><span class="line"><span class="cl"><span class="nb">echo</span> handling: <span class="nv">$sample</span>
</span></span><span class="line"><span class="cl"><span class="c1">####################</span>
</span></span><span class="line"><span class="cl">prodigal -i ~/work/asthma/megahit/contigs/<span class="nv">$sample</span>.fa <span class="se">\
</span></span></span><span class="line"><span class="cl"><span class="se"></span>    -d ~/work/asthma/prodigal/<span class="nv">$sample</span>.gene.fa <span class="se">\
</span></span></span><span class="line"><span class="cl"><span class="se"></span>    -o ~/work/asthma/prodigal/<span class="nv">$sample</span>.gene.gff <span class="se">\
</span></span></span><span class="line"><span class="cl"><span class="se"></span>    -p meta -f gff 
</span></span><span class="line"><span class="cl">    
</span></span><span class="line"><span class="cl">grep <span class="s1">&#39;partial=00&#39;</span> ~/work/asthma/prodigal/<span class="nv">$sample</span>.gene.fa <span class="p">|</span> cut -f1 -d <span class="s1">&#39; &#39;</span><span class="p">|</span> sed <span class="s1">&#39;s/&gt;//&#39;</span> &gt; ~/work/asthma/prodigal/<span class="nv">$sample</span>.fullid
</span></span><span class="line"><span class="cl">seqkit grep -f ~/work/asthma/prodigal/<span class="nv">$sample</span>.fullid ~/work/asthma/prodigal/<span class="nv">$sample</span>.gene.fa &gt; ~/work/asthma/prodigal/fullgene/<span class="nv">$sample</span>.gene.fa
</span></span><span class="line"><span class="cl">
</span></span><span class="line"><span class="cl"><span class="c1">####################</span>
</span></span><span class="line"><span class="cl"><span class="nb">echo</span> end: <span class="sb">`</span>date +<span class="s1">&#39;%Y-%m-%d %T&#39;</span><span class="sb">`</span>
</span></span><span class="line"><span class="cl"><span class="nv">end</span><span class="o">=</span><span class="sb">`</span>date +%s<span class="sb">`</span>
</span></span><span class="line"><span class="cl"><span class="nb">echo</span> TIME:<span class="sb">`</span>expr <span class="nv">$end</span> - <span class="nv">$start</span><span class="sb">`</span><span class="nv">s</span>
</span></span><span class="line"><span class="cl">
</span></span><span class="line"><span class="cl"><span class="o">============================================================================</span>
</span></span><span class="line"><span class="cl"><span class="c1">## ç»Ÿè®¡åŸºå› æ•°é‡</span>
</span></span><span class="line"><span class="cl">grep -c <span class="s1">&#39;&gt;&#39;</span> temp/prodigal/gene.fa 
</span></span><span class="line"><span class="cl"><span class="c1">## ç»Ÿè®¡å®Œæ•´åŸºå› æ•°é‡ï¼Œæ•°æ®é‡å¤§å¯åªç”¨å®Œæ•´åŸºå› éƒ¨åˆ†</span>
</span></span><span class="line"><span class="cl">grep -c <span class="s1">&#39;partial=00&#39;</span> temp/prodigal/gene.fa 
</span></span><span class="line"><span class="cl"><span class="c1">## æå–å®Œæ•´åŸºå› (å®Œæ•´ç‰‡æ®µè·å¾—çš„åŸºå› å…¨ä¸ºå®Œæ•´ï¼Œå¦‚æˆç¯çš„ç»†èŒåŸºå› ç»„)</span>
</span></span><span class="line"><span class="cl">grep <span class="s1">&#39;partial=00&#39;</span> temp/prodigal/gene.fa <span class="p">|</span> cut -f1 -d <span class="s1">&#39; &#39;</span><span class="p">|</span> sed <span class="s1">&#39;s/&gt;//&#39;</span> &gt; temp/prodigal/full_length.id
</span></span><span class="line"><span class="cl">seqkit grep -f temp/prodigal/full_length.id temp/prodigal/gene.fa &gt; temp/prodigal/full_length.fa
</span></span><span class="line"><span class="cl">seqkit stat temp/prodigal/full_length.fa
</span></span></code></pre></td></tr></table>
</div>
</div><h3 id="å»å†—ä½™">å»å†—ä½™</h3>
<p>ä¸Šé¢äº§ç”Ÿäº†nä¸ªæ ·æœ¬çš„åŸºå› é¢„æµ‹ç»“æœæ–‡ä»¶ï¼Œgene.faæ–‡ä»¶è¦æƒ³åŠæ³•æ•´åˆä¸ºä¸€ä¸ªæ–‡ä»¶å†å»å»å†—ä½™ã€‚</p>
<div class="highlight"><div class="chroma">
<table class="lntable"><tr><td class="lntd">
<pre tabindex="0" class="chroma"><code><span class="lnt"> 1
</span><span class="lnt"> 2
</span><span class="lnt"> 3
</span><span class="lnt"> 4
</span><span class="lnt"> 5
</span><span class="lnt"> 6
</span><span class="lnt"> 7
</span><span class="lnt"> 8
</span><span class="lnt"> 9
</span><span class="lnt">10
</span></code></pre></td>
<td class="lntd">
<pre tabindex="0" class="chroma"><code class="language-bash" data-lang="bash"><span class="line"><span class="cl"><span class="cp">#!/bin/bash
</span></span></span><span class="line"><span class="cl"><span class="cp"></span><span class="c1">#ä¿®æ”¹æ¯æ¡åºåˆ—çš„åç§°ï¼ŒåŠ ä¸Šæ ·æœ¬å</span>
</span></span><span class="line"><span class="cl"><span class="k">for</span> i in <span class="sb">`</span>cat ~/work/asthma/data/namelist<span class="sb">`</span>
</span></span><span class="line"><span class="cl"><span class="k">do</span>
</span></span><span class="line"><span class="cl"><span class="nb">echo</span> <span class="nv">$i</span>
</span></span><span class="line"><span class="cl">sed -i <span class="s2">&#34;/&gt;/s/&gt;/&gt;</span><span class="si">${</span><span class="nv">i</span><span class="si">}</span><span class="s2">_/&#34;</span> <span class="nv">$i</span>.gene.fa 
</span></span><span class="line"><span class="cl"><span class="k">done</span>
</span></span><span class="line"><span class="cl"><span class="nb">echo</span> <span class="s1">&#39;start merge&#39;</span>
</span></span><span class="line"><span class="cl">cat *.gene.fa&gt;all.fullgene.fa
</span></span><span class="line"><span class="cl"><span class="nb">echo</span> <span class="s1">&#39;done&#39;</span>
</span></span></code></pre></td></tr></table>
</div>
</div><h4 id="cd-hit">Cd-hit</h4>
<div class="highlight"><div class="chroma">
<table class="lntable"><tr><td class="lntd">
<pre tabindex="0" class="chroma"><code><span class="lnt"> 1
</span><span class="lnt"> 2
</span><span class="lnt"> 3
</span><span class="lnt"> 4
</span><span class="lnt"> 5
</span><span class="lnt"> 6
</span><span class="lnt"> 7
</span><span class="lnt"> 8
</span><span class="lnt"> 9
</span><span class="lnt">10
</span><span class="lnt">11
</span><span class="lnt">12
</span><span class="lnt">13
</span><span class="lnt">14
</span><span class="lnt">15
</span><span class="lnt">16
</span><span class="lnt">17
</span><span class="lnt">18
</span><span class="lnt">19
</span><span class="lnt">20
</span><span class="lnt">21
</span><span class="lnt">22
</span><span class="lnt">23
</span><span class="lnt">24
</span><span class="lnt">25
</span><span class="lnt">26
</span><span class="lnt">27
</span><span class="lnt">28
</span><span class="lnt">29
</span><span class="lnt">30
</span><span class="lnt">31
</span><span class="lnt">32
</span><span class="lnt">33
</span><span class="lnt">34
</span></code></pre></td>
<td class="lntd">
<pre tabindex="0" class="chroma"><code class="language-bash" data-lang="bash"><span class="line"><span class="cl"><span class="cp">#!/bin/bash
</span></span></span><span class="line"><span class="cl"><span class="cp"></span><span class="c1">#SBATCH --job-name=cdhit</span>
</span></span><span class="line"><span class="cl"><span class="c1">#SBATCH --output=/share/home/jianglab/pengchen/work/asthma/%x_%a.out</span>
</span></span><span class="line"><span class="cl"><span class="c1">#SBATCH --error=/share/home/jianglab/pengchen/work/asthma/%x_%a.err</span>
</span></span><span class="line"><span class="cl"><span class="c1">#SBATCH --cpus-per-task=32</span>
</span></span><span class="line"><span class="cl"><span class="c1">#SBATCH --partition=short</span>
</span></span><span class="line"><span class="cl">
</span></span><span class="line"><span class="cl"><span class="nb">echo</span> start: <span class="sb">`</span>date +<span class="s1">&#39;%Y-%m-%d %T&#39;</span><span class="sb">`</span>
</span></span><span class="line"><span class="cl"><span class="nv">start</span><span class="o">=</span><span class="sb">`</span>date +%s<span class="sb">`</span>
</span></span><span class="line"><span class="cl"><span class="c1">##############</span>
</span></span><span class="line"><span class="cl"><span class="c1">## è¾“å…¥æ–‡ä»¶ï¼šprodigalé¢„æµ‹çš„åŸºå› åºåˆ— all.fullgene.fa</span>
</span></span><span class="line"><span class="cl"><span class="c1">## è¾“å‡ºæ–‡ä»¶ï¼šå»å†—ä½™åçš„åŸºå› å’Œè›‹ç™½åºåˆ—ï¼šNR/nucleotide.fa;NR/protein.fa</span>
</span></span><span class="line"><span class="cl">mkdir NR
</span></span><span class="line"><span class="cl"><span class="c1">## aSè¦†ç›–åº¦ï¼Œcç›¸ä¼¼åº¦ï¼ŒGå±€éƒ¨æ¯”å¯¹ï¼Œgæœ€ä¼˜è§£ï¼ŒTå¤šçº¿ç¨‹ï¼ŒMå†…å­˜0ä¸é™åˆ¶</span>
</span></span><span class="line"><span class="cl"><span class="c1">## 2ä¸‡åŸºå› 2mï¼Œ2åƒä¸‡éœ€è¦2000hï¼Œå¤šçº¿ç¨‹å¯åŠ é€Ÿ	</span>
</span></span><span class="line"><span class="cl">cd-hit-est -i prodigal/fullgene/all.fullgene.fa <span class="se">\
</span></span></span><span class="line"><span class="cl"><span class="se"></span>    -o NR/nucleotide.fa <span class="se">\
</span></span></span><span class="line"><span class="cl"><span class="se"></span>    -aS 0.9 -c 0.9 -G <span class="m">0</span> -g <span class="m">0</span> -T <span class="m">0</span> -M <span class="m">0</span>
</span></span><span class="line"><span class="cl"><span class="c1">## ç»Ÿè®¡éå†—ä½™åŸºå› æ•°é‡ï¼Œå•æ¬¡æ‹¼æ¥ç»“æœæ•°é‡ä¸‹é™ä¸å¤§ï¼Œå¤šæ‰¹æ‹¼æ¥å†—ä½™åº¦é«˜</span>
</span></span><span class="line"><span class="cl">
</span></span><span class="line"><span class="cl"><span class="nb">echo</span> <span class="s1">&#39;after remove, the number of genes: &#39;</span>
</span></span><span class="line"><span class="cl">grep -c <span class="s1">&#39;&gt;&#39;</span> NR/nucleotide.fa
</span></span><span class="line"><span class="cl"><span class="c1">## ç¿»è¯‘æ ¸é…¸ä¸ºå¯¹åº”è›‹ç™½åºåˆ—ï¼Œemboss</span>
</span></span><span class="line"><span class="cl"><span class="c1">## emboss transeqå·¥å…·ï¼Œ93.9 MB</span>
</span></span><span class="line"><span class="cl">conda install emboss -y
</span></span><span class="line"><span class="cl">
</span></span><span class="line"><span class="cl">transeq -sequence NR/nucleotide.fa <span class="se">\
</span></span></span><span class="line"><span class="cl"><span class="se"></span>  -outseq NR/protein.fa -trim Y 
</span></span><span class="line"><span class="cl"><span class="c1">## åºåˆ—åè‡ªåŠ¨æ·»åŠ äº†_1ï¼Œä¸ºä¸æ ¸é…¸å¯¹åº”è¦å»é™¤</span>
</span></span><span class="line"><span class="cl">sed -i <span class="s1">&#39;s/_1 / /&#39;</span> NR/protein.fa
</span></span><span class="line"><span class="cl"><span class="c1">##############</span>
</span></span><span class="line"><span class="cl"><span class="nb">echo</span> end: <span class="sb">`</span>date +<span class="s1">&#39;%Y-%m-%d %T&#39;</span><span class="sb">`</span>
</span></span><span class="line"><span class="cl"><span class="nv">end</span><span class="o">=</span><span class="sb">`</span>date +%s<span class="sb">`</span>
</span></span><span class="line"><span class="cl"><span class="nb">echo</span> TIME:<span class="sb">`</span>expr <span class="nv">$end</span> - <span class="nv">$start</span><span class="sb">`</span>s
</span></span></code></pre></td></tr></table>
</div>
</div><h4 id="mmseq2">mmseq2</h4>
<p>è¿™ä¸ªè¦æ¯”cd-hitå¿«éå¸¸å¤š</p>
<div class="highlight"><div class="chroma">
<table class="lntable"><tr><td class="lntd">
<pre tabindex="0" class="chroma"><code><span class="lnt"> 1
</span><span class="lnt"> 2
</span><span class="lnt"> 3
</span><span class="lnt"> 4
</span><span class="lnt"> 5
</span><span class="lnt"> 6
</span><span class="lnt"> 7
</span><span class="lnt"> 8
</span><span class="lnt"> 9
</span><span class="lnt">10
</span><span class="lnt">11
</span><span class="lnt">12
</span><span class="lnt">13
</span><span class="lnt">14
</span><span class="lnt">15
</span><span class="lnt">16
</span><span class="lnt">17
</span><span class="lnt">18
</span><span class="lnt">19
</span><span class="lnt">20
</span><span class="lnt">21
</span><span class="lnt">22
</span><span class="lnt">23
</span><span class="lnt">24
</span><span class="lnt">25
</span><span class="lnt">26
</span><span class="lnt">27
</span><span class="lnt">28
</span><span class="lnt">29
</span><span class="lnt">30
</span></code></pre></td>
<td class="lntd">
<pre tabindex="0" class="chroma"><code class="language-bash" data-lang="bash"><span class="line"><span class="cl"><span class="c1">#mmseqs2</span>
</span></span><span class="line"><span class="cl"><span class="c1">##min-seq-id:identityï¼Œcè¦†ç›–åº¦ï¼Œrescore-mode3: global alignment</span>
</span></span><span class="line"><span class="cl">
</span></span><span class="line"><span class="cl"><span class="c1">#èšç±»</span>
</span></span><span class="line"><span class="cl"><span class="nv">input_fa</span><span class="o">=</span>tmp_com.gene
</span></span><span class="line"><span class="cl">mmseqs easy-linclust <span class="nv">$input_fa</span> lin_res tmp <span class="se">\
</span></span></span><span class="line"><span class="cl"><span class="se"></span>    --min-seq-id 0.9 -c 0.9 --cov-mode <span class="m">1</span>  --threads <span class="m">8</span>
</span></span><span class="line"><span class="cl"><span class="c1">#æˆ–è€…</span>
</span></span><span class="line"><span class="cl">mmseqs easy-cluster <span class="nv">$input_fa</span> lin_res tmp <span class="se">\
</span></span></span><span class="line"><span class="cl"><span class="se"></span>    --min-seq-id 0.9 -c 0.9 --cov-mode <span class="m">1</span>  --threads <span class="m">8</span>
</span></span><span class="line"><span class="cl">
</span></span><span class="line"><span class="cl"><span class="c1">##ä»¥ä¸‹æ˜¯ä¸ºäº†ç†è§£æ­¥éª¤åšçš„ï¼Œä½¿ç”¨ä¸Šé¢ğŸ‘†çš„easyæ¨¡å¼å³å¯</span>
</span></span><span class="line"><span class="cl"><span class="c1">#å»ºåº“</span>
</span></span><span class="line"><span class="cl"><span class="nv">input_fa</span><span class="o">=</span>../prodigal/C1.gene.fa
</span></span><span class="line"><span class="cl"><span class="nv">DB</span><span class="o">=</span>C1.geneDB
</span></span><span class="line"><span class="cl"><span class="nv">DB_clu</span><span class="o">=</span>mmseq_out
</span></span><span class="line"><span class="cl">mmseqs createdb <span class="nv">$input_fa</span> <span class="nv">$DB</span>
</span></span><span class="line"><span class="cl"><span class="c1">#èšç±»</span>
</span></span><span class="line"><span class="cl">mmseqs cluster <span class="nv">$DB</span> <span class="nv">$DB_clu</span> tmp <span class="se">\
</span></span></span><span class="line"><span class="cl"><span class="se"></span>    --min-seq-id 0.9 -c 0.9 --cov-mode <span class="m">1</span>  --threads <span class="m">8</span>  --rescore-mode <span class="m">3</span>
</span></span><span class="line"><span class="cl"><span class="c1">#è¾“å‡ºè½¬æ¢</span>
</span></span><span class="line"><span class="cl"><span class="c1">#ç¬¬ä¸€åˆ—æ˜¯ä»£è¡¨åºåˆ—idï¼Œç¬¬äºŒåˆ—æ˜¯æˆå‘˜åºåˆ—id</span>
</span></span><span class="line"><span class="cl"><span class="c1">#mmseqs createtsv $DB $DB mmseq_out mmseq_out.tsv</span>
</span></span><span class="line"><span class="cl">
</span></span><span class="line"><span class="cl"><span class="c1">#mmseqs createseqfiledb $DB $DB_clu ${DB_clu}_seq</span>
</span></span><span class="line"><span class="cl"><span class="c1">#mmseqs result2flat $DB $DB ${DB_clu}_seq ${DB_clu}_seq.fasta</span>
</span></span><span class="line"><span class="cl">
</span></span><span class="line"><span class="cl"><span class="c1">#è·å–represent sequence</span>
</span></span><span class="line"><span class="cl">mmseqs createsubdb <span class="nv">$DB_clu</span> <span class="nv">$DB</span> <span class="si">${</span><span class="nv">DB_clu</span><span class="si">}</span>_rep
</span></span><span class="line"><span class="cl">mmseqs convert2fasta <span class="si">${</span><span class="nv">DB_clu</span><span class="si">}</span>_rep <span class="si">${</span><span class="nv">DB_clu</span><span class="si">}</span>_rep.fasta   
</span></span></code></pre></td></tr></table>
</div>
</div><h3 id="åŸºå› å®šé‡salmon">åŸºå› å®šé‡ï¼šsalmon</h3>
<ol>
<li>å»ºç«‹ç´¢å¼•</li>
</ol>
<div class="highlight"><div class="chroma">
<table class="lntable"><tr><td class="lntd">
<pre tabindex="0" class="chroma"><code><span class="lnt"> 1
</span><span class="lnt"> 2
</span><span class="lnt"> 3
</span><span class="lnt"> 4
</span><span class="lnt"> 5
</span><span class="lnt"> 6
</span><span class="lnt"> 7
</span><span class="lnt"> 8
</span><span class="lnt"> 9
</span><span class="lnt">10
</span><span class="lnt">11
</span><span class="lnt">12
</span><span class="lnt">13
</span><span class="lnt">14
</span><span class="lnt">15
</span><span class="lnt">16
</span><span class="lnt">17
</span><span class="lnt">18
</span><span class="lnt">19
</span><span class="lnt">20
</span><span class="lnt">21
</span><span class="lnt">22
</span></code></pre></td>
<td class="lntd">
<pre tabindex="0" class="chroma"><code class="language-bash" data-lang="bash"><span class="line"><span class="cl"><span class="cp">#!/bin/bash
</span></span></span><span class="line"><span class="cl"><span class="cp"></span><span class="c1">#SBATCH --job-name=salmon-index</span>
</span></span><span class="line"><span class="cl"><span class="c1">#SBATCH --output=/share/home/jianglab/pengchen/work/asthma/%x_%a.out</span>
</span></span><span class="line"><span class="cl"><span class="c1">#SBATCH --error=/share/home/jianglab/pengchen/work/asthma/%x_%a.err</span>
</span></span><span class="line"><span class="cl"><span class="c1">#SBATCH --cpus-per-task=32</span>
</span></span><span class="line"><span class="cl">
</span></span><span class="line"><span class="cl"><span class="c1">#SBATCH --partition=short</span>
</span></span><span class="line"><span class="cl">
</span></span><span class="line"><span class="cl"><span class="nb">echo</span> start: <span class="sb">`</span>date +<span class="s1">&#39;%Y-%m-%d %T&#39;</span><span class="sb">`</span>
</span></span><span class="line"><span class="cl"><span class="nv">start</span><span class="o">=</span><span class="sb">`</span>date +%s<span class="sb">`</span>
</span></span><span class="line"><span class="cl"><span class="c1">##############</span>
</span></span><span class="line"><span class="cl">mkdir -p temp/salmon
</span></span><span class="line"><span class="cl">
</span></span><span class="line"><span class="cl"><span class="c1">## å»ºç´¢å¼•, -tåºåˆ—, -i ç´¢å¼•ï¼Œ10s</span>
</span></span><span class="line"><span class="cl">salmon index <span class="se">\
</span></span></span><span class="line"><span class="cl"><span class="se"></span>  -t NR/nucleotide.fa <span class="se">\
</span></span></span><span class="line"><span class="cl"><span class="se"></span>  -p <span class="m">32</span> <span class="se">\
</span></span></span><span class="line"><span class="cl"><span class="se"></span>  -i temp/salmon/index 
</span></span><span class="line"><span class="cl"><span class="c1">##############</span>
</span></span><span class="line"><span class="cl"><span class="nb">echo</span> end: <span class="sb">`</span>date +<span class="s1">&#39;%Y-%m-%d %T&#39;</span><span class="sb">`</span>
</span></span><span class="line"><span class="cl"><span class="nv">end</span><span class="o">=</span><span class="sb">`</span>date +%s<span class="sb">`</span>
</span></span><span class="line"><span class="cl"><span class="nb">echo</span> TIME:<span class="sb">`</span>expr <span class="nv">$end</span> - <span class="nv">$start</span><span class="sb">`</span>s
</span></span></code></pre></td></tr></table>
</div>
</div><ol start="2">
<li>å¯¹æ¯ä¸ªæ ·æœ¬å®šé‡</li>
</ol>
<div class="highlight"><div class="chroma">
<table class="lntable"><tr><td class="lntd">
<pre tabindex="0" class="chroma"><code><span class="lnt"> 1
</span><span class="lnt"> 2
</span><span class="lnt"> 3
</span><span class="lnt"> 4
</span><span class="lnt"> 5
</span><span class="lnt"> 6
</span><span class="lnt"> 7
</span><span class="lnt"> 8
</span><span class="lnt"> 9
</span><span class="lnt">10
</span><span class="lnt">11
</span><span class="lnt">12
</span><span class="lnt">13
</span><span class="lnt">14
</span><span class="lnt">15
</span><span class="lnt">16
</span><span class="lnt">17
</span><span class="lnt">18
</span><span class="lnt">19
</span><span class="lnt">20
</span><span class="lnt">21
</span><span class="lnt">22
</span><span class="lnt">23
</span><span class="lnt">24
</span><span class="lnt">25
</span><span class="lnt">26
</span><span class="lnt">27
</span></code></pre></td>
<td class="lntd">
<pre tabindex="0" class="chroma"><code class="language-bash" data-lang="bash"><span class="line"><span class="cl"><span class="cp">#!/bin/bash
</span></span></span><span class="line"><span class="cl"><span class="cp"></span><span class="c1">#SBATCH --job-name=salmon</span>
</span></span><span class="line"><span class="cl"><span class="c1">#SBATCH --output=/share/home/jianglab/pengchen/work/asthma/salmon/log/%x_%a.out</span>
</span></span><span class="line"><span class="cl"><span class="c1">#SBATCH --error=/share/home/jianglab/pengchen/work/asthma/salmon/log/%x_%a.err</span>
</span></span><span class="line"><span class="cl"><span class="c1">#SBATCH --array=1-33</span>
</span></span><span class="line"><span class="cl"><span class="c1">#SBATCH --partition=cpu</span>
</span></span><span class="line"><span class="cl"><span class="c1">#SBATCH --cpus-per-task=32</span>
</span></span><span class="line"><span class="cl">
</span></span><span class="line"><span class="cl"><span class="nb">echo</span> start: <span class="sb">`</span>date +<span class="s1">&#39;%Y-%m-%d %T&#39;</span><span class="sb">`</span>
</span></span><span class="line"><span class="cl"><span class="nv">start</span><span class="o">=</span><span class="sb">`</span>date +%s<span class="sb">`</span>
</span></span><span class="line"><span class="cl"><span class="nb">echo</span> <span class="s2">&#34;SLURM_ARRAY_TASK_ID: &#34;</span> <span class="nv">$SLURM_ARRAY_TASK_ID</span>
</span></span><span class="line"><span class="cl"><span class="nv">sample</span><span class="o">=</span><span class="k">$(</span>head -n <span class="nv">$SLURM_ARRAY_TASK_ID</span> ~/work/asthma/data/namelist <span class="p">|</span> tail -1<span class="k">)</span>
</span></span><span class="line"><span class="cl"><span class="c1">#sample=$(head -n 1 namelist | tail -1)</span>
</span></span><span class="line"><span class="cl"><span class="nb">echo</span> handling: <span class="nv">$sample</span>
</span></span><span class="line"><span class="cl"><span class="c1">####################</span>
</span></span><span class="line"><span class="cl"><span class="c1">## è¾“å…¥æ–‡ä»¶ï¼šå»å†—ä½™åçš„åŸºå› å’Œè›‹ç™½åºåˆ—ï¼šNR/nucleotide.fa</span>
</span></span><span class="line"><span class="cl"><span class="c1">## è¾“å‡ºæ–‡ä»¶ï¼šSalmonå®šé‡åçš„ç»“æœï¼šsalmon/gene.count;salmon/gene.TPM</span>
</span></span><span class="line"><span class="cl"><span class="c1">## å®šé‡ï¼Œlæ–‡åº“ç±»å‹è‡ªåŠ¨é€‰æ‹©ï¼Œpçº¿ç¨‹ï¼Œ--metaå®åŸºå› ç»„æ¨¡å¼</span>
</span></span><span class="line"><span class="cl">salmon quant <span class="se">\
</span></span></span><span class="line"><span class="cl"><span class="se"></span>    -i temp/salmon/index -l A -p <span class="m">32</span> --meta <span class="se">\
</span></span></span><span class="line"><span class="cl"><span class="se"></span>    -1 data/alldata/<span class="si">${</span><span class="nv">sample</span><span class="si">}</span>_f1.fastq <span class="se">\
</span></span></span><span class="line"><span class="cl"><span class="se"></span>    -2 data/alldata/<span class="si">${</span><span class="nv">sample</span><span class="si">}</span>_r2.fastq <span class="se">\
</span></span></span><span class="line"><span class="cl"><span class="se"></span>    -o temp/salmon/<span class="si">${</span><span class="nv">sample</span><span class="si">}</span>.quant
</span></span><span class="line"><span class="cl"><span class="c1">####################</span>
</span></span><span class="line"><span class="cl"><span class="nb">echo</span> end: <span class="sb">`</span>date +<span class="s1">&#39;%Y-%m-%d %T&#39;</span><span class="sb">`</span>
</span></span><span class="line"><span class="cl"><span class="nv">end</span><span class="o">=</span><span class="sb">`</span>date +%s<span class="sb">`</span>
</span></span><span class="line"><span class="cl"><span class="nb">echo</span> TIME:<span class="sb">`</span>expr <span class="nv">$end</span> - <span class="nv">$start</span><span class="sb">`</span>s
</span></span></code></pre></td></tr></table>
</div>
</div><ol start="3">
<li>åˆå¹¶å„æ ·æœ¬ç»“æœ</li>
</ol>
<div class="highlight"><div class="chroma">
<table class="lntable"><tr><td class="lntd">
<pre tabindex="0" class="chroma"><code><span class="lnt"> 1
</span><span class="lnt"> 2
</span><span class="lnt"> 3
</span><span class="lnt"> 4
</span><span class="lnt"> 5
</span><span class="lnt"> 6
</span><span class="lnt"> 7
</span><span class="lnt"> 8
</span><span class="lnt"> 9
</span><span class="lnt">10
</span><span class="lnt">11
</span><span class="lnt">12
</span></code></pre></td>
<td class="lntd">
<pre tabindex="0" class="chroma"><code class="language-bash" data-lang="bash"><span class="line"><span class="cl"><span class="c1">## åˆå¹¶</span>
</span></span><span class="line"><span class="cl">mkdir -p salmon
</span></span><span class="line"><span class="cl">salmon quantmerge <span class="se">\
</span></span></span><span class="line"><span class="cl"><span class="se"></span>    --quants temp/salmon/*.quant <span class="se">\
</span></span></span><span class="line"><span class="cl"><span class="se"></span>    -o salmon/gene.TPM
</span></span><span class="line"><span class="cl">salmon quantmerge <span class="se">\
</span></span></span><span class="line"><span class="cl"><span class="se"></span>    --quants temp/salmon/*.quant <span class="se">\
</span></span></span><span class="line"><span class="cl"><span class="se"></span>    --column NumReads -o salmon/gene.count
</span></span><span class="line"><span class="cl">sed -i <span class="s1">&#39;1 s/.quant//g&#39;</span> salmon/gene.*
</span></span><span class="line"><span class="cl">
</span></span><span class="line"><span class="cl"><span class="c1">## é¢„è§ˆç»“æœè¡¨æ ¼</span>
</span></span><span class="line"><span class="cl">head -n3 salmon/gene.*
</span></span></code></pre></td></tr></table>
</div>
</div><h3 id="åŠŸèƒ½åŸºå› æ³¨é‡Š">åŠŸèƒ½åŸºå› æ³¨é‡Š</h3>
<p>ä¸Šä¸€æ­¥å·²ç»æœ‰äº†æ‰€æœ‰çš„åŸºå› å’Œæ¯ä¸ªæ ·æœ¬æ‰€æœ‰åŸºå› çš„read countå®šé‡ç»“æœï¼Œæˆ‘ä»¬åªéœ€è¦å¯¹ä¸Šä¸€æ­¥çš„åŸºå› åºåˆ—ï¼ˆæˆ–è›‹ç™½è´¨åºåˆ—ï¼‰è¿›è¡Œä¸åŒæ•°æ®åº“çš„æ³¨é‡Šï¼ˆå¾ˆå¤šè½¯ä»¶éƒ½æ˜¯ç”¨diamondæ¯”å¯¹ï¼Œå¦‚æœæ²¡æœ‰ä¸“ç”¨è½¯ä»¶çš„æ•°æ®åº“æˆ‘ä»¬ä¹Ÿå¯ä»¥è‡ªå·±ç”¨diamondæ¯”å¯¹ï¼‰ï¼Œåˆå¹¶æ³¨é‡Šç»“æœå¾—åˆ°çš„å°±æ˜¯åŠŸèƒ½ä¸°åº¦è¡¨ã€‚</p>
<p>diamondé€‰æ‹©&ndash;outfmt 6çš„è¾“å‡ºç»“æœå’Œblastpä¸€æ ·ï¼š</p>
<div class="table-wrapper"><table>
<thead>
<tr>
<th></th>
<th></th>
<th></th>
</tr>
</thead>
<tbody>
<tr>
<td>1.</td>
<td>qseqid</td>
<td>query sequence id</td>
</tr>
<tr>
<td>2.</td>
<td>sseqid</td>
<td>subject (e.g., reference genome) sequence id</td>
</tr>
<tr>
<td>3.</td>
<td>pident</td>
<td>percentage of identical matches</td>
</tr>
<tr>
<td>4.</td>
<td>length</td>
<td>alignment length</td>
</tr>
<tr>
<td>5.</td>
<td>mismatch</td>
<td>number of mismatches</td>
</tr>
<tr>
<td>6.</td>
<td>gapopen</td>
<td>number of gap openings</td>
</tr>
<tr>
<td>7.</td>
<td>qstart</td>
<td>start of alignment in query</td>
</tr>
<tr>
<td>8.</td>
<td>qend</td>
<td>end of alignment in query</td>
</tr>
<tr>
<td>9.</td>
<td>sstart</td>
<td>start of alignment in subject</td>
</tr>
<tr>
<td>10.</td>
<td>send</td>
<td>end of alignment in subject</td>
</tr>
<tr>
<td>11.</td>
<td>evalue</td>
<td>expect value</td>
</tr>
<tr>
<td>12.</td>
<td>bitscore</td>
<td>bit score</td>
</tr>
</tbody>
</table></div>
<h4 id="1-eggnogcogkeggcazy">1 eggNOG(COG/KEGG/CAZy)</h4>
<p><a class="link" href="http://eggnogdb.embl.de/"  target="_blank" rel="noopener"
    >EggNOGæ•°æ®åº“</a>æ”¶é›†äº†COGï¼ˆClusters of Orthologous Groups of proteinsï¼Œç›´ç³»åŒæºè›‹ç™½ç°‡ï¼‰,æ„æˆæ¯ä¸ªCOGçš„è›‹ç™½éƒ½æ˜¯è¢«å‡å®šä¸ºæ¥è‡ªäºä¸€ä¸ªç¥–å…ˆè›‹ç™½ï¼Œå› æ­¤æ˜¯orthologsæˆ–è€…æ˜¯paralogsã€‚é€šè¿‡æŠŠæ‰€æœ‰å®Œæ•´åŸºå› ç»„çš„ç¼–ç è›‹ç™½ä¸€ä¸ªä¸€ä¸ªçš„äº’ç›¸æ¯”è¾ƒç¡®å®šçš„ã€‚åœ¨è€ƒè™‘æ¥è‡ªä¸€ä¸ªç»™å®šåŸºå› ç»„çš„è›‹ç™½æ—¶ï¼Œè¿™ç§æ¯”è¾ƒå°†ç»™å‡ºæ¯ä¸ªå…¶ä»–åŸºå› ç»„çš„ä¸€ä¸ªæœ€ç›¸ä¼¼çš„è›‹ç™½ï¼ˆå› æ­¤éœ€è¦ç”¨å®Œæ•´çš„åŸºå› ç»„æ¥å®šä¹‰COGï¼‰ï¼Œè¿™äº›åŸºå› çš„æ¯ä¸€ä¸ªéƒ½è½®ç•ªåœ°è¢«è€ƒè™‘ã€‚å¦‚æœåœ¨è¿™äº›è›‹ç™½ï¼ˆæˆ–å­é›†ï¼‰ä¹‹é—´ä¸€ä¸ªç›¸äº’çš„æœ€ä½³åŒ¹é…å…³ç³»è¢«å‘ç°ï¼Œé‚£ä¹ˆé‚£äº›ç›¸äº’çš„æœ€ä½³åŒ¹é…å°†å½¢æˆä¸€ä¸ªCOGã€‚è¿™æ ·ï¼Œä¸€ä¸ªCOGä¸­çš„æˆå‘˜å°†ä¸è¿™ä¸ªCOGä¸­çš„å…¶ä»–æˆå‘˜æ¯”èµ·è¢«æ¯”è¾ƒçš„åŸºå› ç»„ä¸­çš„å…¶ä»–è›‹ç™½æ›´ç›¸åƒã€‚</p>
<p><a class="link" href="http://eggnogdb.embl.de/"  target="_blank" rel="noopener"
    >EggNOG</a>é‡Œé¢åŒ…å«äº†GOï¼ŒKEGGï¼ŒCAZyç­‰ã€‚</p>
<div class="highlight"><div class="chroma">
<table class="lntable"><tr><td class="lntd">
<pre tabindex="0" class="chroma"><code><span class="lnt">1
</span><span class="lnt">2
</span><span class="lnt">3
</span><span class="lnt">4
</span><span class="lnt">5
</span><span class="lnt">6
</span><span class="lnt">7
</span><span class="lnt">8
</span><span class="lnt">9
</span></code></pre></td>
<td class="lntd">
<pre tabindex="0" class="chroma"><code class="language-bash" data-lang="bash"><span class="line"><span class="cl"><span class="c1">## ä¸‹è½½å¸¸ç”¨æ•°æ®åº“ï¼Œæ³¨æ„è®¾ç½®ä¸‹è½½ä½ç½®</span>
</span></span><span class="line"><span class="cl">mkdir -p <span class="si">${</span><span class="nv">db</span><span class="si">}</span>/eggnog5 <span class="o">&amp;&amp;</span> <span class="nb">cd</span> <span class="si">${</span><span class="nv">db</span><span class="si">}</span>/eggnog5
</span></span><span class="line"><span class="cl"><span class="c1">## -yé»˜è®¤åŒæ„ï¼Œ-få¼ºåˆ¶ä¸‹è½½ï¼Œeggnog.db.gz 7.9G+4.9G</span>
</span></span><span class="line"><span class="cl">download_eggnog_data.py -y -f --data_dir ./
</span></span><span class="line"><span class="cl">
</span></span><span class="line"><span class="cl"><span class="c1">## ä¸‹è½½æ–¹å¼2(å¯é€‰)ï¼šé“¾æ¥ç›´æ¥ä¸‹è½½</span>
</span></span><span class="line"><span class="cl">wget -c http://eggnog5.embl.de/download/emapperdb-5.0.0/eggnog.db.gz <span class="c1">## 7.9G</span>
</span></span><span class="line"><span class="cl">wget -c http://eggnog5.embl.de/download/emapperdb-5.0.0/eggnog_proteins.dmnd.gz <span class="c1">## 4.9G</span>
</span></span><span class="line"><span class="cl">gunzip *.gz
</span></span></code></pre></td></tr></table>
</div>
</div><div class="highlight"><div class="chroma">
<table class="lntable"><tr><td class="lntd">
<pre tabindex="0" class="chroma"><code><span class="lnt"> 1
</span><span class="lnt"> 2
</span><span class="lnt"> 3
</span><span class="lnt"> 4
</span><span class="lnt"> 5
</span><span class="lnt"> 6
</span><span class="lnt"> 7
</span><span class="lnt"> 8
</span><span class="lnt"> 9
</span><span class="lnt">10
</span><span class="lnt">11
</span><span class="lnt">12
</span><span class="lnt">13
</span><span class="lnt">14
</span><span class="lnt">15
</span><span class="lnt">16
</span><span class="lnt">17
</span><span class="lnt">18
</span><span class="lnt">19
</span><span class="lnt">20
</span><span class="lnt">21
</span><span class="lnt">22
</span><span class="lnt">23
</span><span class="lnt">24
</span><span class="lnt">25
</span><span class="lnt">26
</span><span class="lnt">27
</span><span class="lnt">28
</span><span class="lnt">29
</span><span class="lnt">30
</span><span class="lnt">31
</span><span class="lnt">32
</span><span class="lnt">33
</span></code></pre></td>
<td class="lntd">
<pre tabindex="0" class="chroma"><code class="language-bash" data-lang="bash"><span class="line"><span class="cl"><span class="cp">#!/bin/bash
</span></span></span><span class="line"><span class="cl"><span class="cp"></span><span class="c1">#SBATCH --job-name=eggo</span>
</span></span><span class="line"><span class="cl"><span class="c1">#SBATCH --output=/share/home/jianglab/pengchen/work/asthma/%x_%j.out</span>
</span></span><span class="line"><span class="cl"><span class="c1">#SBATCH --error=/share/home/jianglab/pengchen/work/asthma/%x_%j.err</span>
</span></span><span class="line"><span class="cl"><span class="c1">#SBATCH --cpus-per-task=32</span>
</span></span><span class="line"><span class="cl"><span class="c1">#SBATCH --partition=cpu</span>
</span></span><span class="line"><span class="cl">
</span></span><span class="line"><span class="cl"><span class="nb">echo</span> start: <span class="sb">`</span>date +<span class="s1">&#39;%Y-%m-%d %T&#39;</span><span class="sb">`</span>
</span></span><span class="line"><span class="cl"><span class="nv">start</span><span class="o">=</span><span class="sb">`</span>date +%s<span class="sb">`</span>
</span></span><span class="line"><span class="cl"><span class="c1">##############</span>
</span></span><span class="line"><span class="cl"><span class="c1">#åˆ‡æ¢ç¯å¢ƒ</span>
</span></span><span class="line"><span class="cl"><span class="c1">## diamondæ¯”å¯¹åŸºå› è‡³eggNOG 5.0æ•°æ®åº“, 1~9hï¼Œé»˜è®¤diamond 1e-3</span>
</span></span><span class="line"><span class="cl">mkdir -p temp/eggnog
</span></span><span class="line"><span class="cl">emapper.py --no_annot --no_file_comments --override <span class="se">\
</span></span></span><span class="line"><span class="cl"><span class="se"></span>  --data_dir ~/db/eggnog5 <span class="se">\
</span></span></span><span class="line"><span class="cl"><span class="se"></span>  -i NR/protein.fa <span class="se">\
</span></span></span><span class="line"><span class="cl"><span class="se"></span>  --cpu <span class="m">32</span> -m diamond <span class="se">\
</span></span></span><span class="line"><span class="cl"><span class="se"></span>  -o temp/eggnog/protein
</span></span><span class="line"><span class="cl"><span class="c1">## æ¯”å¯¹ç»“æœåŠŸèƒ½æ³¨é‡Š, 1h</span>
</span></span><span class="line"><span class="cl">emapper.py <span class="se">\
</span></span></span><span class="line"><span class="cl"><span class="se"></span>  --annotate_hits_table temp/eggnog/protein.emapper.seed_orthologs <span class="se">\
</span></span></span><span class="line"><span class="cl"><span class="se"></span>  --data_dir ~/db/eggnog5 <span class="se">\
</span></span></span><span class="line"><span class="cl"><span class="se"></span>  --cpu <span class="m">32</span> --no_file_comments --override <span class="se">\
</span></span></span><span class="line"><span class="cl"><span class="se"></span>  -o temp/eggnog/output
</span></span><span class="line"><span class="cl">
</span></span><span class="line"><span class="cl"><span class="c1">## æ·»è¡¨å¤´, 1åˆ—ä¸ºIDï¼Œ9åˆ—KOï¼Œ16åˆ—CAZyï¼Œ21åˆ—COGï¼Œ22åˆ—æè¿°</span>
</span></span><span class="line"><span class="cl">sed <span class="s1">&#39;1 i Name\tortholog\tevalue\tscore\ttaxonomic\tprotein\tGO\tEC\tKO\tPathway\tModule\tReaction\trclass\tBRITE\tTC\tCAZy\tBiGG\ttax_scope\tOG\tbestOG\tCOG\tdescription&#39;</span> <span class="se">\
</span></span></span><span class="line"><span class="cl"><span class="se"></span>  temp/eggnog/output.emapper.annotations <span class="se">\
</span></span></span><span class="line"><span class="cl"><span class="se"></span>  &gt; temp/eggnog/output
</span></span><span class="line"><span class="cl"><span class="c1">##############</span>
</span></span><span class="line"><span class="cl"><span class="nb">echo</span> end: <span class="sb">`</span>date +<span class="s1">&#39;%Y-%m-%d %T&#39;</span><span class="sb">`</span>
</span></span><span class="line"><span class="cl"><span class="nv">end</span><span class="o">=</span><span class="sb">`</span>date +%s<span class="sb">`</span>
</span></span><span class="line"><span class="cl"><span class="nb">echo</span> TIME:<span class="sb">`</span>expr <span class="nv">$end</span> - <span class="nv">$start</span><span class="sb">`</span>s
</span></span></code></pre></td></tr></table>
</div>
</div><h4 id="2-ç¢³æ°´åŒ–åˆç‰©dbcan2">2 ç¢³æ°´åŒ–åˆç‰©dbCAN2</h4>
<div class="highlight"><div class="chroma">
<table class="lntable"><tr><td class="lntd">
<pre tabindex="0" class="chroma"><code><span class="lnt"> 1
</span><span class="lnt"> 2
</span><span class="lnt"> 3
</span><span class="lnt"> 4
</span><span class="lnt"> 5
</span><span class="lnt"> 6
</span><span class="lnt"> 7
</span><span class="lnt"> 8
</span><span class="lnt"> 9
</span><span class="lnt">10
</span><span class="lnt">11
</span><span class="lnt">12
</span><span class="lnt">13
</span><span class="lnt">14
</span></code></pre></td>
<td class="lntd">
<pre tabindex="0" class="chroma"><code class="language-bash" data-lang="bash"><span class="line"><span class="cl"><span class="c1">## dbCAN2 http://bcb.unl.edu/dbCAN2</span>
</span></span><span class="line"><span class="cl"><span class="c1">## åˆ›å»ºæ•°æ®åº“å­˜æ”¾ç›®å½•å¹¶è¿›å…¥</span>
</span></span><span class="line"><span class="cl">mkdir -p <span class="si">${</span><span class="nv">db</span><span class="si">}</span>/dbCAN2 <span class="o">&amp;&amp;</span> <span class="nb">cd</span> <span class="si">${</span><span class="nv">db</span><span class="si">}</span>/dbCAN2
</span></span><span class="line"><span class="cl"><span class="c1">## ä¸‹è½½åºåˆ—å’Œæè¿°</span>
</span></span><span class="line"><span class="cl">wget -c http://bcb.unl.edu/dbCAN2/download/CAZyDB.07312020.fa
</span></span><span class="line"><span class="cl">wget -c http://bcb.unl.edu/dbCAN2/download/Databases/CAZyDB.07302020.fam-activities.txt
</span></span><span class="line"><span class="cl"><span class="c1">## å¤‡ç”¨æ•°æ®åº“ä¸‹è½½åœ°å€å¹¶è§£å‹ </span>
</span></span><span class="line"><span class="cl"><span class="c1">#wget -c http://210.75.224.110/db/dbcan2/CAZyDB.07312020.fa.gz</span>
</span></span><span class="line"><span class="cl"><span class="c1">#gunzip CAZyDB.07312020.fa.gz</span>
</span></span><span class="line"><span class="cl"><span class="c1">## diamondå»ºç´¢å¼•ï¼Œ800Mï¼Œ1m</span>
</span></span><span class="line"><span class="cl">diamond --version <span class="c1">## 0.8.22/2.0.5</span>
</span></span><span class="line"><span class="cl"><span class="nb">time</span> diamond makedb <span class="se">\
</span></span></span><span class="line"><span class="cl"><span class="se"></span>  --in CAZyDB.07312020.fa <span class="se">\
</span></span></span><span class="line"><span class="cl"><span class="se"></span>  --db CAZyDB.07312020
</span></span></code></pre></td></tr></table>
</div>
</div><div class="highlight"><div class="chroma">
<table class="lntable"><tr><td class="lntd">
<pre tabindex="0" class="chroma"><code><span class="lnt">1
</span><span class="lnt">2
</span><span class="lnt">3
</span><span class="lnt">4
</span><span class="lnt">5
</span><span class="lnt">6
</span><span class="lnt">7
</span><span class="lnt">8
</span></code></pre></td>
<td class="lntd">
<pre tabindex="0" class="chroma"><code class="language-bash" data-lang="bash"><span class="line"><span class="cl"><span class="c1">## æ¯”å¯¹CAZyæ•°æ®åº“, ç”¨æ—¶2~18m; åŠ --sensitiveæ›´å…¨ä½†æ…¢è‡³1h</span>
</span></span><span class="line"><span class="cl">mkdir -p temp/dbcan2
</span></span><span class="line"><span class="cl">diamond blastp   <span class="se">\
</span></span></span><span class="line"><span class="cl"><span class="se"></span>	--db ~/db/dbcan2/CAZyDB.07312020  <span class="se">\
</span></span></span><span class="line"><span class="cl"><span class="se"></span>	--query NR/protein.fa   <span class="se">\
</span></span></span><span class="line"><span class="cl"><span class="se"></span>	--threads <span class="m">64</span> -e 1e-5 --outfmt <span class="m">6</span> <span class="se">\
</span></span></span><span class="line"><span class="cl"><span class="se"></span>	--max-target-seqs <span class="m">1</span> --quiet <span class="se">\
</span></span></span><span class="line"><span class="cl"><span class="se"></span>	--out temp/dbcan2/gene_diamond.f6
</span></span></code></pre></td></tr></table>
</div>
</div><h4 id="3-argscard">3 ARGsï¼ˆCARDï¼‰</h4>
<p>## Github: <a class="link" href="https://github.com/arpcard/rgi"  target="_blank" rel="noopener"
    >https://github.com/arpcard/rgi</a></p>
<div class="highlight"><div class="chroma">
<table class="lntable"><tr><td class="lntd">
<pre tabindex="0" class="chroma"><code><span class="lnt"> 1
</span><span class="lnt"> 2
</span><span class="lnt"> 3
</span><span class="lnt"> 4
</span><span class="lnt"> 5
</span><span class="lnt"> 6
</span><span class="lnt"> 7
</span><span class="lnt"> 8
</span><span class="lnt"> 9
</span><span class="lnt">10
</span><span class="lnt">11
</span><span class="lnt">12
</span><span class="lnt">13
</span><span class="lnt">14
</span><span class="lnt">15
</span><span class="lnt">16
</span><span class="lnt">17
</span><span class="lnt">18
</span><span class="lnt">19
</span><span class="lnt">20
</span><span class="lnt">21
</span></code></pre></td>
<td class="lntd">
<pre tabindex="0" class="chroma"><code class="language-bash" data-lang="bash"><span class="line"><span class="cl"><span class="cp">#!/bin/bash
</span></span></span><span class="line"><span class="cl"><span class="cp"></span><span class="c1">#SBATCH --job-name=rgi</span>
</span></span><span class="line"><span class="cl"><span class="c1">#SBATCH --output=/share/home/jianglab/pengchen/work/asthma/%x_%a.out</span>
</span></span><span class="line"><span class="cl"><span class="c1">#SBATCH --error=/share/home/jianglab/pengchen/work/asthma/%x_%a.err</span>
</span></span><span class="line"><span class="cl"><span class="c1">#SBATCH --cpus-per-task=32</span>
</span></span><span class="line"><span class="cl"><span class="c1">#SBATCH --partition=cpu</span>
</span></span><span class="line"><span class="cl">
</span></span><span class="line"><span class="cl"><span class="nb">echo</span> start: <span class="sb">`</span>date +<span class="s1">&#39;%Y-%m-%d %T&#39;</span><span class="sb">`</span>
</span></span><span class="line"><span class="cl"><span class="nv">start</span><span class="o">=</span><span class="sb">`</span>date +%s<span class="sb">`</span>
</span></span><span class="line"><span class="cl"><span class="c1">##############</span>
</span></span><span class="line"><span class="cl">mkdir -p card
</span></span><span class="line"><span class="cl">
</span></span><span class="line"><span class="cl">rgi main --input_sequence ~/work/asthma/temp/protein.fa <span class="se">\
</span></span></span><span class="line"><span class="cl"><span class="se"></span>--output_file card/protein <span class="se">\
</span></span></span><span class="line"><span class="cl"><span class="se"></span>--input_type protein --num_threads <span class="m">32</span> <span class="se">\
</span></span></span><span class="line"><span class="cl"><span class="se"></span>--clean --alignment_tool DIAMOND
</span></span><span class="line"><span class="cl">
</span></span><span class="line"><span class="cl"><span class="c1">##############</span>
</span></span><span class="line"><span class="cl"><span class="nb">echo</span> end: <span class="sb">`</span>date +<span class="s1">&#39;%Y-%m-%d %T&#39;</span><span class="sb">`</span>
</span></span><span class="line"><span class="cl"><span class="nv">end</span><span class="o">=</span><span class="sb">`</span>date +%s<span class="sb">`</span>
</span></span><span class="line"><span class="cl"><span class="nb">echo</span> TIME:<span class="sb">`</span>expr <span class="nv">$end</span> - <span class="nv">$start</span><span class="sb">`</span>s
</span></span></code></pre></td></tr></table>
</div>
</div><h4 id="4-æ¯’åŠ›å› å­vfdb">4 æ¯’åŠ›å› å­VFDB</h4>
<p>å®˜ç½‘åœ°å€ï¼š<a class="link" href="http://www.mgc.ac.cn/VFs/"  target="_blank" rel="noopener"
    >http://www.mgc.ac.cn/VFs/</a> åœ¨å®˜ç½‘ä¸‹è½½æ•°æ®åº“æ—¶ï¼Œå¸¦æœ‰setA çš„åº“ä¸ºVFDBæ•°æ®åº“æ ¸å¿ƒåº“(set A)ï¼Œè€ŒsetBä¸ºå…¨åº“(setB), å…¶ä¸­setAä»…åŒ…å«ç»å®éªŒéªŒè¯è¿‡çš„æ¯’åŠ›åŸºå› ï¼Œè€ŒsetBåˆ™åœ¨setAçš„åŸºç¡€ä¸Šå¢åŠ äº†é¢„æµ‹çš„æ¯’åŠ›åŸºå› ï¼Œé€‰æ‹©å¥½æ•°æ®åº“åï¼Œç›´æ¥ç”¨blast/diamondå³å¯å®Œæˆæ³¨é‡Šã€‚</p>
<div class="highlight"><div class="chroma">
<table class="lntable"><tr><td class="lntd">
<pre tabindex="0" class="chroma"><code><span class="lnt">1
</span><span class="lnt">2
</span><span class="lnt">3
</span><span class="lnt">4
</span><span class="lnt">5
</span><span class="lnt">6
</span><span class="lnt">7
</span></code></pre></td>
<td class="lntd">
<pre tabindex="0" class="chroma"><code class="language-bash" data-lang="bash"><span class="line"><span class="cl">mkdir -p temp/vfdb
</span></span><span class="line"><span class="cl">diamond blastp   <span class="se">\
</span></span></span><span class="line"><span class="cl"><span class="se"></span>	--db ~/db/VFDB/VFDB_setB_pro  <span class="se">\
</span></span></span><span class="line"><span class="cl"><span class="se"></span>	--query NR/protein.fa   <span class="se">\
</span></span></span><span class="line"><span class="cl"><span class="se"></span>	--threads <span class="m">32</span> -e 1e-5 --outfmt <span class="m">6</span> <span class="se">\
</span></span></span><span class="line"><span class="cl"><span class="se"></span>	--max-target-seqs <span class="m">1</span> --quiet <span class="se">\
</span></span></span><span class="line"><span class="cl"><span class="se"></span>	--out temp/vfdb/gene_diamond.f6
</span></span></code></pre></td></tr></table>
</div>
</div><h4 id="5-å…¶ä»–å„ç§æ•°æ®åº“">5 å…¶ä»–å„ç§æ•°æ®åº“</h4>
<h3 id="åŠŸèƒ½æ³¨é‡Šåˆå¹¶">åŠŸèƒ½æ³¨é‡Šåˆå¹¶</h3>
<p>å†™ä¸€ä¸ªpythonè„šæœ¬ï¼Œå°†è¡¨1ï¼ˆåŸºå› -åŠŸèƒ½çš„å¯¹åº”è¡¨ï¼‰ä¸è¡¨2ï¼ˆåŸºå› ä¸°åº¦è¡¨ï¼‰åˆå¹¶ï¼Œå³ä¸åŒåŸºå› å¯èƒ½æ³¨é‡Šåˆ°ç›¸åŒåŠŸèƒ½ï¼ŒæŠŠå®ƒä»¬çš„ä¸°åº¦åŠ åœ¨ä¸€èµ·å¾—åˆ°æ–°è¡¨3ï¼ˆåŠŸèƒ½ä¸°åº¦è¡¨ï¼‰</p>
<h2 id="binning">binning</h2>
<p>å®åŸºå› ç»„binningæ˜¯æŒ‡å°†ä¸åŒçš„åºåˆ—é›†åˆï¼ˆå¦‚metagenomeåºåˆ—é›†åˆï¼‰æ ¹æ®å®ƒä»¬çš„ç‰©ç§å½’ç±»åˆ°ä¸åŒçš„binsä¸­ï¼Œä»¥ä¾¿è¿›ä¸€æ­¥ç ”ç©¶å®ƒä»¬çš„ç»„æˆå’ŒåŠŸèƒ½ã€‚è¿™ä¸ªè¿‡ç¨‹å¯ä»¥å°†ç±»ä¼¼çš„åºåˆ—ç»„åˆåœ¨ä¸€èµ·ï¼Œå½¢æˆä»£è¡¨ä¸åŒç‰©ç§æˆ–åŸºå› ç»„çš„binsï¼Œä»¥ä¾¿è¿›è¡Œåç»­åˆ†æï¼Œå¦‚ç‰©ç§æ³¨é‡Šã€åŸºå› ç»„ç»„è£…ç­‰ã€‚</p>
<p>ä»¥ä¸‹æ˜¯å¸¸ç”¨çš„å®åŸºå› ç»„binningæ–¹æ³•ï¼š</p>
<ol>
<li>
<p>åŸºäºèšç±»çš„æ–¹æ³•ï¼šè¯¥æ–¹æ³•ä½¿ç”¨åºåˆ—èšç±»å°†ç›¸ä¼¼åºåˆ—åˆ†åˆ°åŒä¸€ä¸ªbinä¸­ã€‚ä¸€èˆ¬æ¥è¯´ï¼Œèšç±»ç®—æ³•å¯åˆ†ä¸ºä¸¤ç±»ï¼šæ— ç›‘ç£èšç±»ï¼ˆå¦‚k-meansã€DBSCANç­‰ï¼‰å’Œæœ‰ç›‘ç£èšç±»ï¼ˆå¦‚CAMIã€MyCCç­‰ï¼‰ã€‚</p>
</li>
<li>
<p>åŸºäºç»„è£…çš„æ–¹æ³•ï¼šè¯¥æ–¹æ³•ä½¿ç”¨de novoç»„è£…æ¥å°†ç›¸ä¼¼åºåˆ—ç»„è£…æˆè¿ç»­çš„åºåˆ—ï¼Œå†æ ¹æ®è¿™äº›åºåˆ—çš„åŸºå› ç»„ä¿¡æ¯æ¥å°†å…¶åˆ†ç±»åˆ°ä¸åŒçš„binsä¸­ã€‚è¿™ç§æ–¹æ³•çš„ä¼˜ç‚¹æ˜¯å¯ä»¥æ›´å¥½åœ°å¤„ç†é‡å¤åºåˆ—ï¼Œç¼ºç‚¹æ˜¯éœ€è¦å¤§é‡çš„è®¡ç®—èµ„æºå’Œæ—¶é—´ã€‚</p>
</li>
<li>
<p>åŸºäºåˆ†ç±»å™¨çš„æ–¹æ³•ï¼šè¯¥æ–¹æ³•ä½¿ç”¨æœºå™¨å­¦ä¹ åˆ†ç±»å™¨æ¥å°†åºåˆ—åˆ†é…åˆ°ä¸åŒçš„binsä¸­ã€‚è¿™ç§æ–¹æ³•çš„ä¼˜ç‚¹æ˜¯å¯ä»¥è‡ªåŠ¨å­¦ä¹ ç‰¹å¾å¹¶åœ¨å¤„ç†å¤§è§„æ¨¡æ•°æ®æ—¶æ•ˆç‡é«˜ï¼Œç¼ºç‚¹æ˜¯éœ€è¦å…ˆå»ºç«‹ä¸€ä¸ªåˆ†ç±»å™¨å¹¶è¿›è¡Œè®­ç»ƒã€‚</p>
</li>
</ol>
<p>åœ¨è¿›è¡Œå®åŸºå› ç»„binningæ—¶ï¼Œé€šå¸¸éœ€è¦ä½¿ç”¨å¤šä¸ªæ–¹æ³•è¿›è¡Œæ¯”è¾ƒï¼Œä»¥é€‰æ‹©æœ€é€‚åˆæ•°æ®é›†çš„æ–¹æ³•ã€‚å¯ä»¥ä½¿ç”¨ä¸€äº›æµè¡Œçš„å·¥å…·æ¥è¿›è¡Œbinningï¼Œå¦‚MetaBATã€MaxBinã€CONCOCTå’ŒMEGANç­‰ã€‚è¿™äº›å·¥å…·é€šå¸¸åŒ…å«å„ç§binningæ–¹æ³•ï¼Œå¯ä»¥æ ¹æ®æ•°æ®é›†å’Œåˆ†æç›®çš„é€‰æ‹©é€‚åˆçš„æ–¹æ³•ã€‚</p>
<p>ç¯‡å¹…é™åˆ¶ï¼Œå…·ä½“çš„æ–¹æ³•æ”¾åœ¨å¦ä¸€ç¯‡é‡Œé¢è®²è§£å§ã€‚</p>
<h2 id="reference">Reference</h2>
<div id="refs" class="references csl-bib-body">
<div id="ref-liuPracticalGuideAmplicon2021" class="csl-entry">
<p><span class="csl-left-margin">1. </span><span class="csl-right-inline">Y.-X. Liu, Y. Qin, T. Chen, M. Lu, X. Qian, X. Guo, Y. Bai, <a class="link" href="https://doi.org/10.1007/s13238-020-00724-8"  target="_blank" rel="noopener"
    >A practical guide to amplicon and metagenomic analysis of microbiome data</a>. <em>Protein &amp; Cell</em>. <strong>12</strong>, 315â€“330 (2021).</span></p>
</div>
</div>

</section>


    <footer class="article-footer">
    
    <section class="article-tags">
        
            <a href="/tags/shell/">shell</a>
        
            <a href="/tags/%E7%BC%96%E7%A8%8B/">ç¼–ç¨‹</a>
        
    </section>


    
    <section class="article-copyright">
        <svg xmlns="http://www.w3.org/2000/svg" class="icon icon-tabler icon-tabler-copyright" width="24" height="24" viewBox="0 0 24 24" stroke-width="2" stroke="currentColor" fill="none" stroke-linecap="round" stroke-linejoin="round">
  <path stroke="none" d="M0 0h24v24H0z"/>
  <circle cx="12" cy="12" r="9" />
  <path d="M14.5 9a3.5 4 0 1 0 0 6" />
</svg>



        <span>Licensed under CC BY-NC-SA 4.0</span>
    </section>
    </footer>


    
        <link 
                rel="stylesheet" 
                href="https://cdn.jsdelivr.net/npm/katex@0.15.6/dist/katex.min.css"integrity="sha256-J&#43;iAE0sgH8QSz9hpcDxXIftnj65JEZgNhGcgReTTK9s="crossorigin="anonymous"
            ><script 
                src="https://cdn.jsdelivr.net/npm/katex@0.15.6/dist/katex.min.js"integrity="sha256-InsNdER1b2xUewP&#43;pKCUJpkhiqwHgqiPXDlIk7GzBu4="crossorigin="anonymous"
                defer
                >
            </script><script 
                src="https://cdn.jsdelivr.net/npm/katex@0.15.6/dist/contrib/auto-render.min.js"integrity="sha256-y39Mpg7V3D4lhBX4x6O0bUqTV4pSrfgwEfGKfxkOdgI="crossorigin="anonymous"
                defer
                >
            </script><script>
    window.addEventListener("DOMContentLoaded", () => {
        renderMathInElement(document.querySelector(`.article-content`), {
            delimiters: [
                { left: "$$", right: "$$", display: true },
                { left: "$", right: "$", display: false },
                { left: "\\(", right: "\\)", display: false },
                { left: "\\[", right: "\\]", display: true }
            ],
            ignoredClasses: ["gist"]
        });})
</script>
    
</article>

    

    

<aside class="related-content--wrapper">
    <h2 class="section-title">Related content</h2>
    <div class="related-content">
        <div class="flex article-list--tile">
            
                
<article class="has-image">
    <a href="/p/reporter-score/">
        
        
            <div class="article-image">
                <img src="/p/reporter-score/images/misuse.d0024bb6a2d1c513c17b8a0329a1bfbc_huc67289fcd675657fb391b94803a3cc8a_1235442_250x150_fill_box_smart1_3.png" 
                        width="250" 
                        height="150" 
                        loading="lazy"
                        alt="Featured image of post Reporter Score å¾®ç”Ÿç‰©åŠŸèƒ½å¯Œé›†åˆ†æ"
                        data-key="reporter-score" 
                        data-hash="md5-0AJLtqLRxRPBe4oDKaG/vA==">
                
            </div>
        

        <div class="article-details">
            <h2 class="article-title">Reporter Score å¾®ç”Ÿç‰©åŠŸèƒ½å¯Œé›†åˆ†æ</h2>
        </div>
    </a>
</article>

            
                
<article class="">
    <a href="/p/batch-effect/">
        
        

        <div class="article-details">
            <h2 class="article-title">ç»„å­¦æ•°æ®å»é™¤æ‰¹æ¬¡æ•ˆåº”ï¼ˆbatch effectï¼‰</h2>
        </div>
    </a>
</article>

            
                
<article class="">
    <a href="/p/mmseqs2/">
        
        

        <div class="article-details">
            <h2 class="article-title">Mmseqs2çš„åŸºç¡€ä½¿ç”¨</h2>
        </div>
    </a>
</article>

            
        </div>
    </div>
</aside>

     
    
        
    <script src="https://utteranc.es/client.js" 
        repo="Asa12138/pc_blog"
        issue-term="pathname"
        
        label="none"
        
        crossorigin="anonymous"
        async
        >
</script>

<style>
    .utterances {
        max-width: unset;
    }
</style>

<script>
    let utterancesLoaded = false;

    function setUtterancesTheme(theme) {
        let utterances = document.querySelector('.utterances iframe');
        if (utterances) {
            utterances.contentWindow.postMessage(
                {
                    type: 'set-theme',
                    theme: `github-${theme}`
                },
                'https://utteranc.es'
            );
        }
    }

    addEventListener('message', event => {
        if (event.origin !== 'https://utteranc.es') return;

        
        utterancesLoaded = true;
        setUtterancesTheme(document.documentElement.dataset.scheme)
    });

    window.addEventListener('onColorSchemeChange', (e) => {
        if (!utterancesLoaded) return;
        setUtterancesTheme(e.detail)
    })
</script>


    

    <footer class="site-footer">
    <section class="copyright">
        &copy; 
        
            2022 - 
        
        2023 Asa&#39;s blog
    </section>
    
    <section class="powerby">
        
            Email: pengchen2001@zju.edu.cn <br/>
        Built with <a href="https://gohugo.io/" target="_blank" rel="noopener">Hugo</a> <br />
        Theme <b><a href="https://github.com/CaiJimmy/hugo-theme-stack" target="_blank" rel="noopener" data-version="3.16.0">Stack</a></b> designed by <a href="https://jimmycai.com" target="_blank" rel="noopener">Jimmy</a>
    </section>
</footer>


    
<div class="pswp" tabindex="-1" role="dialog" aria-hidden="true">

    
    <div class="pswp__bg"></div>

    
    <div class="pswp__scroll-wrap">

        
        <div class="pswp__container">
            <div class="pswp__item"></div>
            <div class="pswp__item"></div>
            <div class="pswp__item"></div>
        </div>

        
        <div class="pswp__ui pswp__ui--hidden">

            <div class="pswp__top-bar">

                

                <div class="pswp__counter"></div>

                <button class="pswp__button pswp__button--close" title="Close (Esc)"></button>

                <button class="pswp__button pswp__button--share" title="Share"></button>

                <button class="pswp__button pswp__button--fs" title="Toggle fullscreen"></button>

                <button class="pswp__button pswp__button--zoom" title="Zoom in/out"></button>

                
                
                <div class="pswp__preloader">
                    <div class="pswp__preloader__icn">
                        <div class="pswp__preloader__cut">
                            <div class="pswp__preloader__donut"></div>
                        </div>
                    </div>
                </div>
            </div>

            <div class="pswp__share-modal pswp__share-modal--hidden pswp__single-tap">
                <div class="pswp__share-tooltip"></div>
            </div>

            <button class="pswp__button pswp__button--arrow--left" title="Previous (arrow left)">
            </button>

            <button class="pswp__button pswp__button--arrow--right" title="Next (arrow right)">
            </button>

            <div class="pswp__caption">
                <div class="pswp__caption__center"></div>
            </div>

        </div>

    </div>

</div><script 
                src="https://cdn.jsdelivr.net/npm/photoswipe@4.1.3/dist/photoswipe.min.js"integrity="sha256-ePwmChbbvXbsO02lbM3HoHbSHTHFAeChekF1xKJdleo="crossorigin="anonymous"
                defer
                >
            </script><script 
                src="https://cdn.jsdelivr.net/npm/photoswipe@4.1.3/dist/photoswipe-ui-default.min.js"integrity="sha256-UKkzOn/w1mBxRmLLGrSeyB4e1xbrp4xylgAWb3M42pU="crossorigin="anonymous"
                defer
                >
            </script><link 
                rel="stylesheet" 
                href="https://cdn.jsdelivr.net/npm/photoswipe@4.1.3/dist/default-skin/default-skin.min.css"crossorigin="anonymous"
            ><link 
                rel="stylesheet" 
                href="https://cdn.jsdelivr.net/npm/photoswipe@4.1.3/dist/photoswipe.min.css"crossorigin="anonymous"
            >

            </main>
        </div>
        <script 
                src="https://cdn.jsdelivr.net/npm/node-vibrant@3.1.6/dist/vibrant.min.js"integrity="sha256-awcR2jno4kI5X0zL8ex0vi2z&#43;KMkF24hUW8WePSA9HM="crossorigin="anonymous"
                
                >
            </script><script type="text/javascript" src="/ts/main.js" defer></script>
<script>
    (function () {
        const customFont = document.createElement('link');
        customFont.href = "https://fonts.googleapis.com/css2?family=Lato:wght@300;400;700&display=swap";

        customFont.type = "text/css";
        customFont.rel = "stylesheet";

        document.head.appendChild(customFont);
    }());
</script>

    </body>
</html>
